{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "64b5995d-8918-4709-abcd-7042b738534e",
   "metadata": {},
   "source": [
    "# Prepare PD Summary Statistics\n",
    "* Project: Cross-ancestry PAR analysis\n",
    "* Version: Python/3.9\n",
    "* Last Updated: 08-JUL-2024"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23dc65fa-0991-4b08-bfc5-6a36125939e4",
   "metadata": {},
   "source": [
    "## Notebook Overview\n",
    "* Load list of 90 risk loci associated with PD from Nalls publication\n",
    "* Import summary statistics from ancestry-specific GWAS\n",
    "* Select top hits for each ancestry and generate dataset for calculations\n",
    "* Identify known risk variants, population-specific variants in summary stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ad5d80be-b641-4e52-b6a5-49f28092c2ae",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Import packages\n",
    "from datetime import datetime\n",
    "import os\n",
    "import glob\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import sys\n",
    "import openpyxl\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49debe24-ee93-4d41-b877-672c4f042d5a",
   "metadata": {},
   "source": [
    "### Load list of 90 risk loci from Nalls paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "23321edc-1729-4810-9321-e4b103bdeb20",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Locus</th>\n",
       "      <th>QTL Nominated Gene (nearest QTL)</th>\n",
       "      <th>Effect allele</th>\n",
       "      <th>Other allele</th>\n",
       "      <th>Effect allele frequency</th>\n",
       "      <th>Beta, all studies</th>\n",
       "      <th>SE, all studies</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq1, new studies</th>\n",
       "      <th>Beta, new studies</th>\n",
       "      <th>StdErr, new studies</th>\n",
       "      <th>P, new studies</th>\n",
       "      <th>I2, new studies</th>\n",
       "      <th>Passes pooled 23andMe QC</th>\n",
       "      <th>Known GWAS locus within 1MB</th>\n",
       "      <th>Failed final filtering and QC</th>\n",
       "      <th>Locus within 250KB</th>\n",
       "      <th>Locus Number</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs114138760</td>\n",
       "      <td>1</td>\n",
       "      <td>154898185</td>\n",
       "      <td>PMVK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.0112</td>\n",
       "      <td>0.2812</td>\n",
       "      <td>0.0478</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0109</td>\n",
       "      <td>0.1997</td>\n",
       "      <td>0.0843</td>\n",
       "      <td>1.779000e-02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs35749011</td>\n",
       "      <td>1</td>\n",
       "      <td>155135036</td>\n",
       "      <td>KRTCAP2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.0169</td>\n",
       "      <td>0.6068</td>\n",
       "      <td>0.0342</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0177</td>\n",
       "      <td>0.6798</td>\n",
       "      <td>0.0615</td>\n",
       "      <td>2.160000e-28</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs76763715</td>\n",
       "      <td>1</td>\n",
       "      <td>155205634</td>\n",
       "      <td>GBAP1</td>\n",
       "      <td>GBAP1</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.9953</td>\n",
       "      <td>-0.7467</td>\n",
       "      <td>0.0765</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9941</td>\n",
       "      <td>-0.6693</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>1.340000e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>F</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs6658353</td>\n",
       "      <td>1</td>\n",
       "      <td>161469054</td>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.5011</td>\n",
       "      <td>0.0650</td>\n",
       "      <td>0.0094</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5105</td>\n",
       "      <td>0.0662</td>\n",
       "      <td>0.0171</td>\n",
       "      <td>1.101000e-04</td>\n",
       "      <td>46.4</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs11578699</td>\n",
       "      <td>1</td>\n",
       "      <td>171719769</td>\n",
       "      <td>VAMP4</td>\n",
       "      <td>VAMP4</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.1949</td>\n",
       "      <td>-0.0704</td>\n",
       "      <td>0.0120</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1947</td>\n",
       "      <td>-0.0663</td>\n",
       "      <td>0.0226</td>\n",
       "      <td>3.313000e-03</td>\n",
       "      <td>13.4</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           SNP  Chromosome   Position    Locus  \\\n",
       "0  rs114138760           1  154898185     PMVK   \n",
       "1   rs35749011           1  155135036  KRTCAP2   \n",
       "2   rs76763715           1  155205634    GBAP1   \n",
       "3    rs6658353           1  161469054   FCGR2A   \n",
       "4   rs11578699           1  171719769    VAMP4   \n",
       "\n",
       "  QTL Nominated Gene (nearest QTL) Effect allele Other allele  \\\n",
       "0                              NaN             c            g   \n",
       "1                              NaN             a            g   \n",
       "2                            GBAP1             t            c   \n",
       "3                           FCGR2A             c            g   \n",
       "4                            VAMP4             t            c   \n",
       "\n",
       "   Effect allele frequency  Beta, all studies  SE, all studies  ...  \\\n",
       "0                   0.0112             0.2812           0.0478  ...   \n",
       "1                   0.0169             0.6068           0.0342  ...   \n",
       "2                   0.9953            -0.7467           0.0765  ...   \n",
       "3                   0.5011             0.0650           0.0094  ...   \n",
       "4                   0.1949            -0.0704           0.0120  ...   \n",
       "\n",
       "   Freq1, new studies  Beta, new studies  StdErr, new studies  P, new studies  \\\n",
       "0              0.0109             0.1997               0.0843    1.779000e-02   \n",
       "1              0.0177             0.6798               0.0615    2.160000e-28   \n",
       "2              0.9941            -0.6693               0.1269    1.340000e-07   \n",
       "3              0.5105             0.0662               0.0171    1.101000e-04   \n",
       "4              0.1947            -0.0663               0.0226    3.313000e-03   \n",
       "\n",
       "   I2, new studies  Passes pooled 23andMe QC  Known GWAS locus within 1MB  \\\n",
       "0              0.0                         T                            1   \n",
       "1              0.0                         T                            1   \n",
       "2              0.0                         F                            1   \n",
       "3             46.4                         T                            0   \n",
       "4             13.4                         T                            0   \n",
       "\n",
       "   Failed final filtering and QC  Locus within 250KB  Locus Number  \n",
       "0                              0                   1             1  \n",
       "1                              0                   1             1  \n",
       "2                              0                   1             1  \n",
       "3                              0                   0             2  \n",
       "4                              0                   0             3  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_nalls = pd.read_excel('{WORK_DIR}/90risk_loci.xlsx').rename(columns={'CHR':'Chromosome','BP':'Position', 'Nearest Gene':'Locus'})\n",
    "df_nalls.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35c3eff9-0d8d-4b5f-8fd6-b292df836bfa",
   "metadata": {},
   "source": [
    "### Summary statistics for each ancestry"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e4171513-f349-4059-9ec3-fd58588c0401",
   "metadata": {},
   "source": [
    "#### European from Nalls et al. 2019"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d3ae1bf4-2100-432e-a840-05fe8022c250",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Allele1</th>\n",
       "      <th>Allele2</th>\n",
       "      <th>Freq1</th>\n",
       "      <th>FreqSE</th>\n",
       "      <th>MinFreq</th>\n",
       "      <th>MaxFreq</th>\n",
       "      <th>Effect</th>\n",
       "      <th>StdErr</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Direction</th>\n",
       "      <th>HetISq</th>\n",
       "      <th>HetChiSq</th>\n",
       "      <th>HetDf</th>\n",
       "      <th>HetPVal</th>\n",
       "      <th>freqSpan</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Chr</th>\n",
       "      <th>BP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:98240868</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.5665</td>\n",
       "      <td>0.0067</td>\n",
       "      <td>0.5525</td>\n",
       "      <td>0.5943</td>\n",
       "      <td>0.0110</td>\n",
       "      <td>0.0095</td>\n",
       "      <td>0.2476</td>\n",
       "      <td>+++++++-++----+-+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.140</td>\n",
       "      <td>16</td>\n",
       "      <td>0.5144</td>\n",
       "      <td>0.0418</td>\n",
       "      <td>rs7899632</td>\n",
       "      <td>10</td>\n",
       "      <td>98240868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:98240888</td>\n",
       "      <td>a</td>\n",
       "      <td>c</td>\n",
       "      <td>0.7953</td>\n",
       "      <td>0.0080</td>\n",
       "      <td>0.7208</td>\n",
       "      <td>0.8111</td>\n",
       "      <td>-0.0091</td>\n",
       "      <td>0.0116</td>\n",
       "      <td>0.4295</td>\n",
       "      <td>-+++-------+-+-+-</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.625</td>\n",
       "      <td>16</td>\n",
       "      <td>0.6999</td>\n",
       "      <td>0.0903</td>\n",
       "      <td>rs61875309</td>\n",
       "      <td>10</td>\n",
       "      <td>98240888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:98242110</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.0140</td>\n",
       "      <td>0.0017</td>\n",
       "      <td>0.0044</td>\n",
       "      <td>0.0178</td>\n",
       "      <td>-0.0152</td>\n",
       "      <td>0.0649</td>\n",
       "      <td>0.8147</td>\n",
       "      <td>+??+--?++??+-++--</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.126</td>\n",
       "      <td>11</td>\n",
       "      <td>0.7019</td>\n",
       "      <td>0.0134</td>\n",
       "      <td>rs150203744</td>\n",
       "      <td>10</td>\n",
       "      <td>98242110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:98242642</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.0018</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>0.0014</td>\n",
       "      <td>0.0043</td>\n",
       "      <td>-0.1331</td>\n",
       "      <td>0.1778</td>\n",
       "      <td>0.4541</td>\n",
       "      <td>-?+--?+??+??-?+??</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.328</td>\n",
       "      <td>7</td>\n",
       "      <td>0.7413</td>\n",
       "      <td>0.0029</td>\n",
       "      <td>rs8181398</td>\n",
       "      <td>10</td>\n",
       "      <td>98242642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:98242707</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.9868</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>0.9792</td>\n",
       "      <td>0.9912</td>\n",
       "      <td>0.0347</td>\n",
       "      <td>0.0742</td>\n",
       "      <td>0.6396</td>\n",
       "      <td>+???--?++??+---++</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.593</td>\n",
       "      <td>10</td>\n",
       "      <td>0.5711</td>\n",
       "      <td>0.0120</td>\n",
       "      <td>rs111551711</td>\n",
       "      <td>10</td>\n",
       "      <td>98242707</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       MarkerName Allele1 Allele2   Freq1  FreqSE  MinFreq  MaxFreq  Effect  \\\n",
       "0  chr10:98240868       a       g  0.5665  0.0067   0.5525   0.5943  0.0110   \n",
       "1  chr10:98240888       a       c  0.7953  0.0080   0.7208   0.8111 -0.0091   \n",
       "2  chr10:98242110       t       c  0.0140  0.0017   0.0044   0.0178 -0.0152   \n",
       "3  chr10:98242642       a       g  0.0018  0.0005   0.0014   0.0043 -0.1331   \n",
       "4  chr10:98242707       t       c  0.9868  0.0020   0.9792   0.9912  0.0347   \n",
       "\n",
       "   StdErr  P-value          Direction  HetISq  HetChiSq  HetDf  HetPVal  \\\n",
       "0  0.0095   0.2476  +++++++-++----+-+     0.0    15.140     16   0.5144   \n",
       "1  0.0116   0.4295  -+++-------+-+-+-     0.0    12.625     16   0.6999   \n",
       "2  0.0649   0.8147  +??+--?++??+-++--     0.0     8.126     11   0.7019   \n",
       "3  0.1778   0.4541  -?+--?+??+??-?+??     0.0     4.328      7   0.7413   \n",
       "4  0.0742   0.6396  +???--?++??+---++     0.0     8.593     10   0.5711   \n",
       "\n",
       "   freqSpan          SNP  Chr        BP  \n",
       "0    0.0418    rs7899632   10  98240868  \n",
       "1    0.0903   rs61875309   10  98240888  \n",
       "2    0.0134  rs150203744   10  98242110  \n",
       "3    0.0029    rs8181398   10  98242642  \n",
       "4    0.0120  rs111551711   10  98242707  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_eur = pd.read_csv('{WORK_DIR}/summary_stats/META5_all_with_rsid_hg38.txt', sep='\\t', header=0).rename(columns={\n",
    "    'ID':'SNP'})\n",
    "df_eur.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b4838d65-4ce0-4d94-863b-fd2f26d81a05",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>ancestry</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMVK</td>\n",
       "      <td>rs114138760</td>\n",
       "      <td>1</td>\n",
       "      <td>154925709</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.2812</td>\n",
       "      <td>4.193000e-09</td>\n",
       "      <td>0.0112</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KRTCAP2</td>\n",
       "      <td>rs35749011</td>\n",
       "      <td>1</td>\n",
       "      <td>155162560</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.6068</td>\n",
       "      <td>1.720000e-70</td>\n",
       "      <td>0.0169</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GBAP1</td>\n",
       "      <td>rs76763715</td>\n",
       "      <td>1</td>\n",
       "      <td>155235843</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.7467</td>\n",
       "      <td>1.592000e-22</td>\n",
       "      <td>0.9953</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>rs6658353</td>\n",
       "      <td>1</td>\n",
       "      <td>161499264</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0650</td>\n",
       "      <td>6.099000e-12</td>\n",
       "      <td>0.5011</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VAMP4</td>\n",
       "      <td>rs11578699</td>\n",
       "      <td>1</td>\n",
       "      <td>171750629</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.0704</td>\n",
       "      <td>4.468000e-09</td>\n",
       "      <td>0.1949</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>RIT2</td>\n",
       "      <td>rs12456492</td>\n",
       "      <td>18</td>\n",
       "      <td>43093415</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.0983</td>\n",
       "      <td>3.798000e-23</td>\n",
       "      <td>0.6816</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>MEX3C</td>\n",
       "      <td>rs8087969</td>\n",
       "      <td>18</td>\n",
       "      <td>51157219</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.0578</td>\n",
       "      <td>1.412000e-08</td>\n",
       "      <td>0.5496</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>SPPL2B</td>\n",
       "      <td>rs55818311</td>\n",
       "      <td>19</td>\n",
       "      <td>2341049</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.0696</td>\n",
       "      <td>4.180000e-10</td>\n",
       "      <td>0.6937</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>CRLS1</td>\n",
       "      <td>rs77351827</td>\n",
       "      <td>20</td>\n",
       "      <td>6025395</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0802</td>\n",
       "      <td>8.867000e-09</td>\n",
       "      <td>0.1275</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>DYRK1A</td>\n",
       "      <td>rs2248244</td>\n",
       "      <td>21</td>\n",
       "      <td>37480059</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0714</td>\n",
       "      <td>2.743000e-11</td>\n",
       "      <td>0.2828</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>90 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Locus          SNP  Chromosome   Position effect_allele other_allele  \\\n",
       "0      PMVK  rs114138760           1  154925709             C            G   \n",
       "1   KRTCAP2   rs35749011           1  155162560             A            G   \n",
       "2     GBAP1   rs76763715           1  155235843             T            C   \n",
       "3    FCGR2A    rs6658353           1  161499264             C            G   \n",
       "4     VAMP4   rs11578699           1  171750629             T            C   \n",
       "..      ...          ...         ...        ...           ...          ...   \n",
       "85     RIT2   rs12456492          18   43093415             A            G   \n",
       "86    MEX3C    rs8087969          18   51157219             T            G   \n",
       "87   SPPL2B   rs55818311          19    2341049             T            C   \n",
       "88    CRLS1   rs77351827          20    6025395             T            C   \n",
       "89   DYRK1A    rs2248244          21   37480059             A            G   \n",
       "\n",
       "      beta       p_value  effect_allele_frequency  ancestry  \n",
       "0   0.2812  4.193000e-09                   0.0112  European  \n",
       "1   0.6068  1.720000e-70                   0.0169  European  \n",
       "2  -0.7467  1.592000e-22                   0.9953  European  \n",
       "3   0.0650  6.099000e-12                   0.5011  European  \n",
       "4  -0.0704  4.468000e-09                   0.1949  European  \n",
       "..     ...           ...                      ...       ...  \n",
       "85 -0.0983  3.798000e-23                   0.6816  European  \n",
       "86 -0.0578  1.412000e-08                   0.5496  European  \n",
       "87 -0.0696  4.180000e-10                   0.6937  European  \n",
       "88  0.0802  8.867000e-09                   0.1275  European  \n",
       "89  0.0714  2.743000e-11                   0.2828  European  \n",
       "\n",
       "[90 rows x 10 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge summary stats with 90 risk loci from Nalls\n",
    "eur = pd.merge(df_nalls, df_eur, how='inner', on='SNP')\n",
    "\n",
    "# Add a column for ancestry\n",
    "eur['ancestry'] = 'European'\n",
    "\n",
    "# Subset and rename columns\n",
    "eur = eur[['Locus','SNP','Chr','BP','Allele1','Allele2','Effect','P-value','Freq1','ancestry']].rename(columns={\n",
    "    'ID':'SNP','Chr':'Chromosome','BP':'Position','Allele1':'effect_allele','Allele2':'other_allele',\n",
    "    'Effect':'beta','P-value':'p_value','Freq1':'effect_allele_frequency'})\n",
    "\n",
    "# Change alleles to be uppercase\n",
    "eur['effect_allele'] = eur['effect_allele'].str.upper()\n",
    "eur['other_allele'] = eur['other_allele'].str.upper()\n",
    "\n",
    "# Filter p < 0.05\n",
    "eur = eur[eur['p_value'] < 0.05]\n",
    "eur"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c395a06-76ae-434d-b5ea-669408c45771",
   "metadata": {},
   "source": [
    "#### African/African Admixed from Rizig et al. 2023"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5ea4559b-fbf1-484e-b7c9-ffca23d5fce7",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>p_value</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>direction</th>\n",
       "      <th>HetISq</th>\n",
       "      <th>HetChiSq</th>\n",
       "      <th>HetDf</th>\n",
       "      <th>HetPVal</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>66861</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.1072</td>\n",
       "      <td>0.4023</td>\n",
       "      <td>0.0724</td>\n",
       "      <td>0.7899</td>\n",
       "      <td>chr1:66861:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>??-?</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>rs28375825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>80346</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.4608</td>\n",
       "      <td>0.3456</td>\n",
       "      <td>0.8338</td>\n",
       "      <td>0.1824</td>\n",
       "      <td>chr1:80346:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>??+?</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>rs376665626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>595259</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.0456</td>\n",
       "      <td>0.4097</td>\n",
       "      <td>0.0597</td>\n",
       "      <td>0.9114</td>\n",
       "      <td>chr1:595259:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>??-?</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>rs201764041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>664938</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.1109</td>\n",
       "      <td>0.2264</td>\n",
       "      <td>0.9390</td>\n",
       "      <td>0.6242</td>\n",
       "      <td>chr1:664938:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>???-</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>rs536144132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>665098</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.0315</td>\n",
       "      <td>0.3587</td>\n",
       "      <td>0.1262</td>\n",
       "      <td>0.9301</td>\n",
       "      <td>chr1:665098:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>??-?</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>rs114979547</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  base_pair_location effect_allele other_allele    beta  \\\n",
       "0          1               66861             T            C -0.1072   \n",
       "1          1               80346             C            G  0.4608   \n",
       "2          1              595259             A            G -0.0456   \n",
       "3          1              664938             A            G -0.1109   \n",
       "4          1              665098             A            G -0.0315   \n",
       "\n",
       "   standard_error  effect_allele_frequency  p_value       variant_id  \\\n",
       "0          0.4023                   0.0724   0.7899   chr1:66861:C:T   \n",
       "1          0.3456                   0.8338   0.1824   chr1:80346:C:G   \n",
       "2          0.4097                   0.0597   0.9114  chr1:595259:G:A   \n",
       "3          0.2264                   0.9390   0.6242  chr1:664938:A:G   \n",
       "4          0.3587                   0.1262   0.9301  chr1:665098:G:A   \n",
       "\n",
       "  ref_allele direction  HetISq  HetChiSq  HetDf  HetPVal          SNP  \n",
       "0          C      ??-?     0.0       0.0      0      1.0   rs28375825  \n",
       "1          C      ??+?     0.0       0.0      0      1.0  rs376665626  \n",
       "2          G      ??-?     0.0       0.0      0      1.0  rs201764041  \n",
       "3          A      ???-     0.0       0.0      0      1.0  rs536144132  \n",
       "4          G      ??-?     0.0       0.0      0      1.0  rs114979547  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_afr = pd.read_csv('{WORK_DIR}/summary_stats/AFR_AAC_metaGWAS_MAF0.05_hg38_noindels_full_with23andMe.tab', sep='\\t', header=0, low_memory=False).rename(columns={\n",
    "    'rsid':'SNP'})\n",
    "df_afr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4dd34712-8fb4-41f4-882b-87bc0b5f3cb8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_441737/3325075928.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  afr_gba['Locus'] = 'GBA1'\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>ancestry</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>LINC00693</td>\n",
       "      <td>rs6808178</td>\n",
       "      <td>3</td>\n",
       "      <td>28664199</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.2227</td>\n",
       "      <td>9.219000e-03</td>\n",
       "      <td>0.1211</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MCCC1</td>\n",
       "      <td>rs10513789</td>\n",
       "      <td>3</td>\n",
       "      <td>183042285</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1727</td>\n",
       "      <td>6.314000e-03</td>\n",
       "      <td>0.8122</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>LCORL</td>\n",
       "      <td>rs34025766</td>\n",
       "      <td>4</td>\n",
       "      <td>17967188</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.1061</td>\n",
       "      <td>3.149000e-02</td>\n",
       "      <td>0.4289</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SNCA</td>\n",
       "      <td>rs356182</td>\n",
       "      <td>4</td>\n",
       "      <td>89704960</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.2188</td>\n",
       "      <td>2.481000e-05</td>\n",
       "      <td>0.6818</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>LOC100131289</td>\n",
       "      <td>rs4140646</td>\n",
       "      <td>6</td>\n",
       "      <td>27771022</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1077</td>\n",
       "      <td>4.796000e-02</td>\n",
       "      <td>0.2516</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>RIMS1</td>\n",
       "      <td>rs12528068</td>\n",
       "      <td>6</td>\n",
       "      <td>71778059</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.1037</td>\n",
       "      <td>3.324000e-02</td>\n",
       "      <td>0.4097</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>FYN</td>\n",
       "      <td>rs997368</td>\n",
       "      <td>6</td>\n",
       "      <td>111922088</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.2066</td>\n",
       "      <td>2.744000e-04</td>\n",
       "      <td>0.3057</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>rs13294100</td>\n",
       "      <td>9</td>\n",
       "      <td>17579692</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.1849</td>\n",
       "      <td>2.815000e-03</td>\n",
       "      <td>0.7672</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GBF1</td>\n",
       "      <td>rs10748818</td>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1870</td>\n",
       "      <td>1.233000e-02</td>\n",
       "      <td>0.8696</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>IGSF9B</td>\n",
       "      <td>rs3802920</td>\n",
       "      <td>11</td>\n",
       "      <td>133917106</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.3455</td>\n",
       "      <td>4.670000e-03</td>\n",
       "      <td>0.0729</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>MBNL2</td>\n",
       "      <td>rs4771268</td>\n",
       "      <td>13</td>\n",
       "      <td>97212767</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.1841</td>\n",
       "      <td>4.975000e-03</td>\n",
       "      <td>0.1578</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>VPS13C</td>\n",
       "      <td>rs2251086</td>\n",
       "      <td>15</td>\n",
       "      <td>61705186</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.1192</td>\n",
       "      <td>1.957000e-02</td>\n",
       "      <td>0.3778</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>RIT2</td>\n",
       "      <td>rs12456492</td>\n",
       "      <td>18</td>\n",
       "      <td>43093415</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.1282</td>\n",
       "      <td>1.423000e-02</td>\n",
       "      <td>0.7090</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393146</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>rs3115534</td>\n",
       "      <td>1</td>\n",
       "      <td>155235878</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.4494</td>\n",
       "      <td>2.397000e-14</td>\n",
       "      <td>0.7816</td>\n",
       "      <td>African/African Admixed</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Locus         SNP Chromosome   Position effect_allele  \\\n",
       "10         LINC00693   rs6808178          3   28664199             T   \n",
       "15             MCCC1  rs10513789          3  183042285             T   \n",
       "18             LCORL  rs34025766          4   17967188             A   \n",
       "22              SNCA    rs356182          4   89704960             A   \n",
       "29      LOC100131289   rs4140646          6   27771022             A   \n",
       "31             RIMS1  rs12528068          6   71778059             T   \n",
       "32               FYN    rs997368          6  111922088             A   \n",
       "39            SH3GL2  rs13294100          9   17579692             T   \n",
       "43              GBF1  rs10748818         10  102255522             A   \n",
       "47            IGSF9B   rs3802920         11  133917106             T   \n",
       "53             MBNL2   rs4771268         13   97212767             T   \n",
       "58            VPS13C   rs2251086         15   61705186             T   \n",
       "72              RIT2  rs12456492         18   43093415             A   \n",
       "393146          GBA1   rs3115534          1  155235878             T   \n",
       "\n",
       "       other_allele    beta       p_value  effect_allele_frequency  \\\n",
       "10                C  0.2227  9.219000e-03                   0.1211   \n",
       "15                G  0.1727  6.314000e-03                   0.8122   \n",
       "18                T -0.1061  3.149000e-02                   0.4289   \n",
       "22                G -0.2188  2.481000e-05                   0.6818   \n",
       "29                G  0.1077  4.796000e-02                   0.2516   \n",
       "31                C  0.1037  3.324000e-02                   0.4097   \n",
       "32                G  0.2066  2.744000e-04                   0.3057   \n",
       "39                G -0.1849  2.815000e-03                   0.7672   \n",
       "43                G  0.1870  1.233000e-02                   0.8696   \n",
       "47                G  0.3455  4.670000e-03                   0.0729   \n",
       "53                C  0.1841  4.975000e-03                   0.1578   \n",
       "58                C -0.1192  1.957000e-02                   0.3778   \n",
       "72                G -0.1282  1.423000e-02                   0.7090   \n",
       "393146            G -0.4494  2.397000e-14                   0.7816   \n",
       "\n",
       "                       ancestry  \n",
       "10      African/African Admixed  \n",
       "15      African/African Admixed  \n",
       "18      African/African Admixed  \n",
       "22      African/African Admixed  \n",
       "29      African/African Admixed  \n",
       "31      African/African Admixed  \n",
       "32      African/African Admixed  \n",
       "39      African/African Admixed  \n",
       "43      African/African Admixed  \n",
       "47      African/African Admixed  \n",
       "53      African/African Admixed  \n",
       "58      African/African Admixed  \n",
       "72      African/African Admixed  \n",
       "393146  African/African Admixed  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge summary stats with 90 risk loci from Nalls\n",
    "afr = pd.merge(df_nalls, df_afr, how='inner', on='SNP')\n",
    "\n",
    "# A/AA GBA variant not in df_nalls\n",
    "afr_gba = df_afr[df_afr.SNP == 'rs3115534']\n",
    "afr_gba['Locus'] = 'GBA1'\n",
    "afr = pd.concat([afr, afr_gba])\n",
    "\n",
    "# Add a column for ancestry\n",
    "afr['ancestry'] = 'African/African Admixed'\n",
    "\n",
    "# Subset and rename columns\n",
    "afr = afr[['Locus','SNP','chromosome','base_pair_location','effect_allele','other_allele','beta','p_value',\n",
    "              'effect_allele_frequency','ancestry']].rename(columns={'chromosome':'Chromosome',\n",
    "                                                                     'base_pair_location':'Position'})\n",
    "\n",
    "# Get African-specific GBA1 variant\n",
    "afr_gba = df_afr[df_afr.SNP == 'rs3115534']\n",
    "\n",
    "# Filter p < 0.05\n",
    "afr = afr[afr['p_value'] < 0.05]\n",
    "afr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ff8b7e1b-3145-442a-b247-93070d122eb8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# checking for SNPs reviewers might ask for:\n",
    "# LRRK2: R1441G (rs33939927), R1628P (rs33949390), G2385R (rs34778348), G2019S (rs34637584)\n",
    "list_check_snps = ['rs33939927','rs33949390','rs34778348','rs34637584']\n",
    "check_snps_eur = df_eur[df_eur['SNP'].isin(list_check_snps)]\n",
    "check_snps_afr = df_afr[df_afr['SNP'].isin(list_check_snps)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26f98be7-b08b-4d12-b117-ff554a9459f2",
   "metadata": {},
   "source": [
    "#### East Asian from Foo et al. 2020"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a8a8edc9-1a81-45fa-95f0-7419e0351ad6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MARKERNAME</th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>EA</th>\n",
       "      <th>NEA</th>\n",
       "      <th>EAF</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>OR</th>\n",
       "      <th>OR_95U</th>\n",
       "      <th>OR_95L</th>\n",
       "      <th>N</th>\n",
       "      <th>NMISS</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:794332</td>\n",
       "      <td>1</td>\n",
       "      <td>794332</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.139423</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>0.033560</td>\n",
       "      <td>1.001301</td>\n",
       "      <td>1.069378</td>\n",
       "      <td>0.937558</td>\n",
       "      <td>31575</td>\n",
       "      <td>21168.408424</td>\n",
       "      <td>0.9691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:832359</td>\n",
       "      <td>1</td>\n",
       "      <td>832359</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.031250</td>\n",
       "      <td>-0.0932</td>\n",
       "      <td>0.070600</td>\n",
       "      <td>0.911011</td>\n",
       "      <td>1.046213</td>\n",
       "      <td>0.793282</td>\n",
       "      <td>31575</td>\n",
       "      <td>21168.408424</td>\n",
       "      <td>0.1868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:834056</td>\n",
       "      <td>1</td>\n",
       "      <td>834056</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.028846</td>\n",
       "      <td>-0.0766</td>\n",
       "      <td>0.077458</td>\n",
       "      <td>0.926260</td>\n",
       "      <td>1.078119</td>\n",
       "      <td>0.795792</td>\n",
       "      <td>31575</td>\n",
       "      <td>21168.408424</td>\n",
       "      <td>0.3227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:834263</td>\n",
       "      <td>1</td>\n",
       "      <td>834263</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.028846</td>\n",
       "      <td>-0.0846</td>\n",
       "      <td>0.079177</td>\n",
       "      <td>0.918880</td>\n",
       "      <td>1.073138</td>\n",
       "      <td>0.786795</td>\n",
       "      <td>31575</td>\n",
       "      <td>21168.408424</td>\n",
       "      <td>0.2853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:834956</td>\n",
       "      <td>1</td>\n",
       "      <td>834956</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.026442</td>\n",
       "      <td>-0.0704</td>\n",
       "      <td>0.077861</td>\n",
       "      <td>0.932021</td>\n",
       "      <td>1.085680</td>\n",
       "      <td>0.800109</td>\n",
       "      <td>31575</td>\n",
       "      <td>21168.408424</td>\n",
       "      <td>0.3659</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    MARKERNAME  CHROMOSOME  POSITION EA NEA       EAF    BETA        SE  \\\n",
       "0  chr1:794332           1    794332  A   G  0.139423  0.0013  0.033560   \n",
       "1  chr1:832359           1    832359  T   C  0.031250 -0.0932  0.070600   \n",
       "2  chr1:834056           1    834056  C   A  0.028846 -0.0766  0.077458   \n",
       "3  chr1:834263           1    834263  T   C  0.028846 -0.0846  0.079177   \n",
       "4  chr1:834956           1    834956  A   G  0.026442 -0.0704  0.077861   \n",
       "\n",
       "         OR    OR_95U    OR_95L      N         NMISS       P  \n",
       "0  1.001301  1.069378  0.937558  31575  21168.408424  0.9691  \n",
       "1  0.911011  1.046213  0.793282  31575  21168.408424  0.1868  \n",
       "2  0.926260  1.078119  0.795792  31575  21168.408424  0.3227  \n",
       "3  0.918880  1.073138  0.786795  31575  21168.408424  0.2853  \n",
       "4  0.932021  1.085680  0.800109  31575  21168.408424  0.3659  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ea = pd.read_csv('{WORK_DIR}/summary_stats/EAS.final.txt', sep='\\t', header=0, low_memory=False)\n",
    "df_ea.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "db8edb8b-e8c2-4d53-98de-8c8911c81f3e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MARKERNAME</th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>EA</th>\n",
       "      <th>NEA</th>\n",
       "      <th>EAF</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>OR</th>\n",
       "      <th>OR_95U</th>\n",
       "      <th>OR_95L</th>\n",
       "      <th>N</th>\n",
       "      <th>NMISS</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3151430</th>\n",
       "      <td>chr12:40757328</td>\n",
       "      <td>12</td>\n",
       "      <td>40757328</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.036058</td>\n",
       "      <td>0.6634</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.941382</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31575</td>\n",
       "      <td>21168.408424</td>\n",
       "      <td>4.680000e-24</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             MARKERNAME  CHROMOSOME  POSITION EA NEA       EAF    BETA  SE  \\\n",
       "3151430  chr12:40757328          12  40757328  A   G  0.036058  0.6634 NaN   \n",
       "\n",
       "               OR  OR_95U  OR_95L      N         NMISS             P  \n",
       "3151430  1.941382     NaN     NaN  31575  21168.408424  4.680000e-24  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check for LRRK2 G2385R (rs34778348)\n",
    "df_ea[df_ea.MARKERNAME == 'chr12:40757328']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "190bd6d9-2c82-4d58-8095-9ac139167723",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_441737/2393798005.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  ea_snps['SNP'] = ea_snps['MARKERNAME'].map({'chr1:155210185':'rs146532106','chr5:75599208':'rs246814',\n",
      "/tmp/ipykernel_441737/2393798005.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  ea_snps['Locus'] = ea_snps['MARKERNAME'].map({'chr1:155210185':'GBA1','chr5:75599208':'SV2C',\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>ancestry</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RAB29</td>\n",
       "      <td>rs11557080</td>\n",
       "      <td>1</td>\n",
       "      <td>205737739</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.2126</td>\n",
       "      <td>2.910000e-20</td>\n",
       "      <td>0.519231</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ITPKB</td>\n",
       "      <td>rs4653767</td>\n",
       "      <td>1</td>\n",
       "      <td>226916078</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.1305</td>\n",
       "      <td>3.180000e-07</td>\n",
       "      <td>0.264423</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MED12L</td>\n",
       "      <td>rs11707416</td>\n",
       "      <td>3</td>\n",
       "      <td>151108965</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.0645</td>\n",
       "      <td>4.794000e-02</td>\n",
       "      <td>0.120192</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MCCC1</td>\n",
       "      <td>rs10513789</td>\n",
       "      <td>3</td>\n",
       "      <td>182760073</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.1901</td>\n",
       "      <td>1.100000e-16</td>\n",
       "      <td>0.608173</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>TMEM175</td>\n",
       "      <td>rs34311866</td>\n",
       "      <td>4</td>\n",
       "      <td>951947</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.1520</td>\n",
       "      <td>7.030000e-06</td>\n",
       "      <td>0.127404</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>BST1</td>\n",
       "      <td>rs4698412</td>\n",
       "      <td>4</td>\n",
       "      <td>15737348</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1105</td>\n",
       "      <td>1.510000e-06</td>\n",
       "      <td>0.360577</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>LCORL</td>\n",
       "      <td>rs34025766</td>\n",
       "      <td>4</td>\n",
       "      <td>17968811</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.0854</td>\n",
       "      <td>8.808000e-03</td>\n",
       "      <td>0.144231</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>rs6825004</td>\n",
       "      <td>4</td>\n",
       "      <td>77110365</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.1192</td>\n",
       "      <td>4.380000e-07</td>\n",
       "      <td>0.379808</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>CLCN3</td>\n",
       "      <td>rs62333164</td>\n",
       "      <td>4</td>\n",
       "      <td>170583157</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.1028</td>\n",
       "      <td>3.004000e-02</td>\n",
       "      <td>0.060096</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>C5orf24</td>\n",
       "      <td>rs11950533</td>\n",
       "      <td>5</td>\n",
       "      <td>134199105</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.0557</td>\n",
       "      <td>3.602000e-02</td>\n",
       "      <td>0.293269</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>LOC100131289</td>\n",
       "      <td>rs4140646</td>\n",
       "      <td>6</td>\n",
       "      <td>27738801</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0875</td>\n",
       "      <td>1.424000e-03</td>\n",
       "      <td>0.170673</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>HLA-DRB5</td>\n",
       "      <td>rs112485576</td>\n",
       "      <td>6</td>\n",
       "      <td>32578772</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.1605</td>\n",
       "      <td>2.972000e-04</td>\n",
       "      <td>0.069711</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>RIMS1</td>\n",
       "      <td>rs12528068</td>\n",
       "      <td>6</td>\n",
       "      <td>72487762</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.1299</td>\n",
       "      <td>2.193000e-03</td>\n",
       "      <td>0.060096</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>FYN</td>\n",
       "      <td>rs997368</td>\n",
       "      <td>6</td>\n",
       "      <td>112243291</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.1087</td>\n",
       "      <td>5.570000e-06</td>\n",
       "      <td>0.360577</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>RPS12</td>\n",
       "      <td>rs75859381</td>\n",
       "      <td>6</td>\n",
       "      <td>133210361</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.2195</td>\n",
       "      <td>4.104000e-04</td>\n",
       "      <td>0.031250</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GPNMB</td>\n",
       "      <td>rs199351</td>\n",
       "      <td>7</td>\n",
       "      <td>23300049</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.0708</td>\n",
       "      <td>3.291000e-03</td>\n",
       "      <td>0.310096</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GS1-124K5.11</td>\n",
       "      <td>rs76949143</td>\n",
       "      <td>7</td>\n",
       "      <td>66009851</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.0820</td>\n",
       "      <td>1.051000e-02</td>\n",
       "      <td>0.153846</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>FGF20</td>\n",
       "      <td>rs620513</td>\n",
       "      <td>8</td>\n",
       "      <td>16697593</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.0911</td>\n",
       "      <td>9.200000e-05</td>\n",
       "      <td>0.444712</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>BIN3</td>\n",
       "      <td>rs2280104</td>\n",
       "      <td>8</td>\n",
       "      <td>22525980</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.0632</td>\n",
       "      <td>3.003000e-02</td>\n",
       "      <td>0.802885</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>rs13294100</td>\n",
       "      <td>9</td>\n",
       "      <td>17579690</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0658</td>\n",
       "      <td>4.580000e-03</td>\n",
       "      <td>0.487981</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>SH3GL2</td>\n",
       "      <td>rs10756907</td>\n",
       "      <td>9</td>\n",
       "      <td>17727065</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0791</td>\n",
       "      <td>7.791000e-04</td>\n",
       "      <td>0.408654</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>UBAP2</td>\n",
       "      <td>rs6476434</td>\n",
       "      <td>9</td>\n",
       "      <td>34046391</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.0977</td>\n",
       "      <td>1.647000e-03</td>\n",
       "      <td>0.834135</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>DLG2</td>\n",
       "      <td>rs12283611</td>\n",
       "      <td>11</td>\n",
       "      <td>83487277</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.1187</td>\n",
       "      <td>5.820000e-06</td>\n",
       "      <td>0.269231</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>HIP1R</td>\n",
       "      <td>rs10847864</td>\n",
       "      <td>12</td>\n",
       "      <td>123326598</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0765</td>\n",
       "      <td>1.420000e-03</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>FBRSL1</td>\n",
       "      <td>rs11610045</td>\n",
       "      <td>12</td>\n",
       "      <td>133063768</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0991</td>\n",
       "      <td>3.283000e-03</td>\n",
       "      <td>0.122596</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>RPS6KL1</td>\n",
       "      <td>rs3742785</td>\n",
       "      <td>14</td>\n",
       "      <td>75373034</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.0761</td>\n",
       "      <td>6.151000e-03</td>\n",
       "      <td>0.216346</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>VPS13C</td>\n",
       "      <td>rs2251086</td>\n",
       "      <td>15</td>\n",
       "      <td>61997385</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.1280</td>\n",
       "      <td>5.560000e-05</td>\n",
       "      <td>0.829327</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>SETD1A</td>\n",
       "      <td>rs11150601</td>\n",
       "      <td>16</td>\n",
       "      <td>30977799</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0920</td>\n",
       "      <td>2.580000e-02</td>\n",
       "      <td>0.949519</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>NOD2</td>\n",
       "      <td>rs6500328</td>\n",
       "      <td>16</td>\n",
       "      <td>50736656</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.0636</td>\n",
       "      <td>2.556000e-02</td>\n",
       "      <td>0.218750</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>CASC16</td>\n",
       "      <td>rs3104783</td>\n",
       "      <td>16</td>\n",
       "      <td>52636242</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0881</td>\n",
       "      <td>1.503000e-03</td>\n",
       "      <td>0.817308</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>UBTF</td>\n",
       "      <td>rs2269906</td>\n",
       "      <td>17</td>\n",
       "      <td>42294337</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.0745</td>\n",
       "      <td>5.800000e-03</td>\n",
       "      <td>0.764423</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>ASXL3</td>\n",
       "      <td>rs1941685</td>\n",
       "      <td>18</td>\n",
       "      <td>31304318</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1539</td>\n",
       "      <td>5.900000e-05</td>\n",
       "      <td>0.884615</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>MEX3C</td>\n",
       "      <td>rs8087969</td>\n",
       "      <td>18</td>\n",
       "      <td>48683589</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.0574</td>\n",
       "      <td>3.056000e-02</td>\n",
       "      <td>0.269231</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188029</th>\n",
       "      <td>GBA1</td>\n",
       "      <td>rs146532106</td>\n",
       "      <td>1</td>\n",
       "      <td>155210185</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.3120</td>\n",
       "      <td>1.960000e-06</td>\n",
       "      <td>0.021635</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1474405</th>\n",
       "      <td>SV2C</td>\n",
       "      <td>rs246814</td>\n",
       "      <td>5</td>\n",
       "      <td>75599208</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.2167</td>\n",
       "      <td>3.480000e-08</td>\n",
       "      <td>0.105769</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Locus          SNP  Chromosome   Position effect_allele  \\\n",
       "2               RAB29   rs11557080           1  205737739             A   \n",
       "3               ITPKB    rs4653767           1  226916078             C   \n",
       "10             MED12L   rs11707416           3  151108965             A   \n",
       "12              MCCC1   rs10513789           3  182760073             G   \n",
       "13            TMEM175   rs34311866           4     951947             C   \n",
       "14               BST1    rs4698412           4   15737348             A   \n",
       "15              LCORL   rs34025766           4   17968811             A   \n",
       "16             SCARB2    rs6825004           4   77110365             G   \n",
       "20              CLCN3   rs62333164           4  170583157             A   \n",
       "23            C5orf24   rs11950533           5  134199105             A   \n",
       "24       LOC100131289    rs4140646           6   27738801             A   \n",
       "25           HLA-DRB5  rs112485576           6   32578772             A   \n",
       "26              RIMS1   rs12528068           6   72487762             T   \n",
       "27                FYN     rs997368           6  112243291             G   \n",
       "28              RPS12   rs75859381           6  133210361             C   \n",
       "29              GPNMB     rs199351           7   23300049             C   \n",
       "30       GS1-124K5.11   rs76949143           7   66009851             A   \n",
       "31              FGF20     rs620513           8   16697593             T   \n",
       "32               BIN3    rs2280104           8   22525980             C   \n",
       "34             SH3GL2   rs13294100           9   17579690             G   \n",
       "35             SH3GL2   rs10756907           9   17727065             G   \n",
       "36              UBAP2    rs6476434           9   34046391             T   \n",
       "38               DLG2   rs12283611          11   83487277             A   \n",
       "41              HIP1R   rs10847864          12  123326598             T   \n",
       "42             FBRSL1   rs11610045          12  133063768             A   \n",
       "47            RPS6KL1    rs3742785          14   75373034             C   \n",
       "48             VPS13C    rs2251086          15   61997385             C   \n",
       "49             SETD1A   rs11150601          16   30977799             A   \n",
       "50               NOD2    rs6500328          16   50736656             G   \n",
       "51             CASC16    rs3104783          16   52636242             A   \n",
       "55               UBTF    rs2269906          17   42294337             C   \n",
       "60              ASXL3    rs1941685          18   31304318             T   \n",
       "61              MEX3C    rs8087969          18   48683589             G   \n",
       "188029           GBA1  rs146532106           1  155210185             C   \n",
       "1474405          SV2C     rs246814           5   75599208             T   \n",
       "\n",
       "        other_allele    beta       p_value  effect_allele_frequency  \\\n",
       "2                  G  0.2126  2.910000e-20                 0.519231   \n",
       "3                  T -0.1305  3.180000e-07                 0.264423   \n",
       "10                 T -0.0645  4.794000e-02                 0.120192   \n",
       "12                 T -0.1901  1.100000e-16                 0.608173   \n",
       "13                 T  0.1520  7.030000e-06                 0.127404   \n",
       "14                 G  0.1105  1.510000e-06                 0.360577   \n",
       "15                 T -0.0854  8.808000e-03                 0.144231   \n",
       "16                 C -0.1192  4.380000e-07                 0.379808   \n",
       "20                 G -0.1028  3.004000e-02                 0.060096   \n",
       "23                 C -0.0557  3.602000e-02                 0.293269   \n",
       "24                 G  0.0875  1.424000e-03                 0.170673   \n",
       "25                 C -0.1605  2.972000e-04                 0.069711   \n",
       "26                 C  0.1299  2.193000e-03                 0.060096   \n",
       "27                 A -0.1087  5.570000e-06                 0.360577   \n",
       "28                 T  0.2195  4.104000e-04                 0.031250   \n",
       "29                 A -0.0708  3.291000e-03                 0.310096   \n",
       "30                 T -0.0820  1.051000e-02                 0.153846   \n",
       "31                 G -0.0911  9.200000e-05                 0.444712   \n",
       "32                 T -0.0632  3.003000e-02                 0.802885   \n",
       "34                 T  0.0658  4.580000e-03                 0.487981   \n",
       "35                 A  0.0791  7.791000e-04                 0.408654   \n",
       "36                 C -0.0977  1.647000e-03                 0.834135   \n",
       "38                 C -0.1187  5.820000e-06                 0.269231   \n",
       "41                 G  0.0765  1.420000e-03                 0.500000   \n",
       "42                 G  0.0991  3.283000e-03                 0.122596   \n",
       "47                 A -0.0761  6.151000e-03                 0.216346   \n",
       "48                 T  0.1280  5.560000e-05                 0.829327   \n",
       "49                 G  0.0920  2.580000e-02                 0.949519   \n",
       "50                 A -0.0636  2.556000e-02                 0.218750   \n",
       "51                 C  0.0881  1.503000e-03                 0.817308   \n",
       "55                 A -0.0745  5.800000e-03                 0.764423   \n",
       "60                 G  0.1539  5.900000e-05                 0.884615   \n",
       "61                 T -0.0574  3.056000e-02                 0.269231   \n",
       "188029             G  0.3120  1.960000e-06                 0.021635   \n",
       "1474405            C  0.2167  3.480000e-08                 0.105769   \n",
       "\n",
       "           ancestry  \n",
       "2        East Asian  \n",
       "3        East Asian  \n",
       "10       East Asian  \n",
       "12       East Asian  \n",
       "13       East Asian  \n",
       "14       East Asian  \n",
       "15       East Asian  \n",
       "16       East Asian  \n",
       "20       East Asian  \n",
       "23       East Asian  \n",
       "24       East Asian  \n",
       "25       East Asian  \n",
       "26       East Asian  \n",
       "27       East Asian  \n",
       "28       East Asian  \n",
       "29       East Asian  \n",
       "30       East Asian  \n",
       "31       East Asian  \n",
       "32       East Asian  \n",
       "34       East Asian  \n",
       "35       East Asian  \n",
       "36       East Asian  \n",
       "38       East Asian  \n",
       "41       East Asian  \n",
       "42       East Asian  \n",
       "47       East Asian  \n",
       "48       East Asian  \n",
       "49       East Asian  \n",
       "50       East Asian  \n",
       "51       East Asian  \n",
       "55       East Asian  \n",
       "60       East Asian  \n",
       "61       East Asian  \n",
       "188029   East Asian  \n",
       "1474405  East Asian  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Rename columns and merge with risk loci\n",
    "df_ea = df_ea.rename(columns={'CHROMOSOME':'Chromosome','POSITION':'Position'})\n",
    "ea = pd.merge(df_nalls, df_ea, how='inner', on=['Chromosome','Position'])\n",
    "\n",
    "# Add in reported top GWAS hits from Foo\n",
    "list_ea_snps = ['chr1:155210185','chr5:75599208','chr7:70750493']\n",
    "ea_snps = df_ea[df_ea['MARKERNAME'].isin(list_ea_snps)]\n",
    "\n",
    "# Fill in missing rsID's and locus names\n",
    "ea_snps['SNP'] = ea_snps['MARKERNAME'].map({'chr1:155210185':'rs146532106','chr5:75599208':'rs246814',\n",
    "                                            'chr7:70750493':'rs9638616'})\n",
    "ea_snps['Locus'] = ea_snps['MARKERNAME'].map({'chr1:155210185':'GBA1','chr5:75599208':'SV2C',\n",
    "                                              'chr7:70750493':'WBSCR17'})\n",
    "ea = pd.concat([ea,ea_snps])\n",
    "\n",
    "# Add a column for ancestry\n",
    "ea['ancestry'] = 'East Asian'\n",
    "\n",
    "# Subset and rename columns\n",
    "ea = ea[['Locus','SNP','Chromosome','Position','EA','NEA','BETA','P','EAF','ancestry']].rename(columns={\n",
    "    'EA':'effect_allele','NEA':'other_allele','BETA':'beta',\n",
    "    'P':'p_value','EAF':'effect_allele_frequency'})\n",
    "\n",
    "# Filter p < 0.05\n",
    "ea = ea[ea['p_value'] < 0.05]\n",
    "ea"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5188b71d-8cf4-4482-a26a-e12c1908ecf3",
   "metadata": {},
   "source": [
    "#### Latino from Loesch et al. 2021"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "04f6f8ea-1bf9-4eac-9a1b-bb3edc662ed4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MARKERNAME</th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>EA</th>\n",
       "      <th>NEA</th>\n",
       "      <th>EAF</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>OR</th>\n",
       "      <th>OR_95U</th>\n",
       "      <th>OR_95L</th>\n",
       "      <th>N</th>\n",
       "      <th>NMISS</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:662622</td>\n",
       "      <td>1</td>\n",
       "      <td>662622</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.065474</td>\n",
       "      <td>-0.187575</td>\n",
       "      <td>0.210099</td>\n",
       "      <td>0.828967</td>\n",
       "      <td>1.251344</td>\n",
       "      <td>0.549158</td>\n",
       "      <td>1481</td>\n",
       "      <td>1487.855711</td>\n",
       "      <td>0.371967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:666249</td>\n",
       "      <td>1</td>\n",
       "      <td>666249</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.073317</td>\n",
       "      <td>0.266337</td>\n",
       "      <td>0.180918</td>\n",
       "      <td>1.305175</td>\n",
       "      <td>1.860669</td>\n",
       "      <td>0.915521</td>\n",
       "      <td>1481</td>\n",
       "      <td>1487.855711</td>\n",
       "      <td>0.140982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:668394</td>\n",
       "      <td>1</td>\n",
       "      <td>668394</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>0.018808</td>\n",
       "      <td>-0.171700</td>\n",
       "      <td>0.491800</td>\n",
       "      <td>0.842232</td>\n",
       "      <td>2.208309</td>\n",
       "      <td>0.321220</td>\n",
       "      <td>1481</td>\n",
       "      <td>1487.855711</td>\n",
       "      <td>0.726995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:676118</td>\n",
       "      <td>1</td>\n",
       "      <td>676118</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.038932</td>\n",
       "      <td>0.044314</td>\n",
       "      <td>0.282878</td>\n",
       "      <td>1.045310</td>\n",
       "      <td>1.819850</td>\n",
       "      <td>0.600420</td>\n",
       "      <td>1481</td>\n",
       "      <td>1487.855711</td>\n",
       "      <td>0.875518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:693625</td>\n",
       "      <td>1</td>\n",
       "      <td>693625</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.021886</td>\n",
       "      <td>-0.310839</td>\n",
       "      <td>0.358697</td>\n",
       "      <td>0.732832</td>\n",
       "      <td>1.480244</td>\n",
       "      <td>0.362806</td>\n",
       "      <td>1481</td>\n",
       "      <td>1487.855711</td>\n",
       "      <td>0.386173</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    MARKERNAME CHROMOSOME  POSITION EA NEA       EAF      BETA        SE  \\\n",
       "0  chr1:662622          1    662622  A   G  0.065474 -0.187575  0.210099   \n",
       "1  chr1:666249          1    666249  T   C  0.073317  0.266337  0.180918   \n",
       "2  chr1:668394          1    668394  A  AG  0.018808 -0.171700  0.491800   \n",
       "3  chr1:676118          1    676118  T   C  0.038932  0.044314  0.282878   \n",
       "4  chr1:693625          1    693625  C   T  0.021886 -0.310839  0.358697   \n",
       "\n",
       "         OR    OR_95U    OR_95L     N        NMISS         P  \n",
       "0  0.828967  1.251344  0.549158  1481  1487.855711  0.371967  \n",
       "1  1.305175  1.860669  0.915521  1481  1487.855711  0.140982  \n",
       "2  0.842232  2.208309  0.321220  1481  1487.855711  0.726995  \n",
       "3  1.045310  1.819850  0.600420  1481  1487.855711  0.875518  \n",
       "4  0.732832  1.480244  0.362806  1481  1487.855711  0.386173  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_lat = pd.read_csv('{WORK_DIR}/summary_stats/AMR.final.txt', sep='\\t', header=0, low_memory=False)\n",
    "df_lat.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ef5eacd2-0be0-468f-80ae-4c828e081c89",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# GBA stats are in 23andMe data\n",
    "df_lat23 = pd.read_csv('{WORK_DIR}/summary_stats/23AMR.formatted.txt.gz', sep='\\t', header=0, low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "8b0b62e1-a696-4aac-a251-021bf975b74b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>ancestry</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>KCNS3</td>\n",
       "      <td>rs76116224</td>\n",
       "      <td>2</td>\n",
       "      <td>18147848</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.359278</td>\n",
       "      <td>2.684601e-02</td>\n",
       "      <td>0.073345</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KCNIP3</td>\n",
       "      <td>rs2042477</td>\n",
       "      <td>2</td>\n",
       "      <td>96000943</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.195749</td>\n",
       "      <td>2.921862e-02</td>\n",
       "      <td>0.614394</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>LINC00693</td>\n",
       "      <td>rs6808178</td>\n",
       "      <td>3</td>\n",
       "      <td>28705690</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.228445</td>\n",
       "      <td>3.913275e-02</td>\n",
       "      <td>0.810031</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>KPNA1</td>\n",
       "      <td>rs55961674</td>\n",
       "      <td>3</td>\n",
       "      <td>122196892</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.272241</td>\n",
       "      <td>4.138190e-02</td>\n",
       "      <td>0.118806</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SNCA</td>\n",
       "      <td>rs356182</td>\n",
       "      <td>4</td>\n",
       "      <td>90626111</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.460053</td>\n",
       "      <td>2.484767e-08</td>\n",
       "      <td>0.555666</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ELOVL7</td>\n",
       "      <td>rs1867598</td>\n",
       "      <td>5</td>\n",
       "      <td>60137959</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.292834</td>\n",
       "      <td>2.627049e-02</td>\n",
       "      <td>0.107321</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>BIN3</td>\n",
       "      <td>rs2280104</td>\n",
       "      <td>8</td>\n",
       "      <td>22525980</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.214305</td>\n",
       "      <td>2.515152e-02</td>\n",
       "      <td>0.758349</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>ITGA8</td>\n",
       "      <td>rs896435</td>\n",
       "      <td>10</td>\n",
       "      <td>15557406</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.192646</td>\n",
       "      <td>2.817975e-02</td>\n",
       "      <td>0.673742</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>VPS13C</td>\n",
       "      <td>rs2251086</td>\n",
       "      <td>15</td>\n",
       "      <td>61997385</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.236127</td>\n",
       "      <td>4.489795e-02</td>\n",
       "      <td>0.855848</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>SETD1A</td>\n",
       "      <td>rs11150601</td>\n",
       "      <td>16</td>\n",
       "      <td>30977799</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.278549</td>\n",
       "      <td>9.806977e-04</td>\n",
       "      <td>0.558559</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>CRHR1</td>\n",
       "      <td>rs117615688</td>\n",
       "      <td>17</td>\n",
       "      <td>43798308</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.740730</td>\n",
       "      <td>2.291157e-04</td>\n",
       "      <td>0.045562</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Locus          SNP  Chromosome   Position effect_allele other_allele  \\\n",
       "6       KCNS3   rs76116224           2   18147848             T            A   \n",
       "7      KCNIP3    rs2042477           2   96000943             T            A   \n",
       "11  LINC00693    rs6808178           3   28705690             C            T   \n",
       "13      KPNA1   rs55961674           3  122196892             T            C   \n",
       "24       SNCA     rs356182           4   90626111             A            G   \n",
       "27     ELOVL7    rs1867598           5   60137959             G            A   \n",
       "37       BIN3    rs2280104           8   22525980             C            T   \n",
       "42      ITGA8     rs896435          10   15557406             T            C   \n",
       "57     VPS13C    rs2251086          15   61997385             C            T   \n",
       "59     SETD1A   rs11150601          16   30977799             A            G   \n",
       "68      CRHR1  rs117615688          17   43798308             A            G   \n",
       "\n",
       "        beta       p_value  effect_allele_frequency ancestry  \n",
       "6  -0.359278  2.684601e-02                 0.073345   Latino  \n",
       "7   0.195749  2.921862e-02                 0.614394   Latino  \n",
       "11 -0.228445  3.913275e-02                 0.810031   Latino  \n",
       "13  0.272241  4.138190e-02                 0.118806   Latino  \n",
       "24 -0.460053  2.484767e-08                 0.555666   Latino  \n",
       "27  0.292834  2.627049e-02                 0.107321   Latino  \n",
       "37 -0.214305  2.515152e-02                 0.758349   Latino  \n",
       "42  0.192646  2.817975e-02                 0.673742   Latino  \n",
       "57  0.236127  4.489795e-02                 0.855848   Latino  \n",
       "59  0.278549  9.806977e-04                 0.558559   Latino  \n",
       "68 -0.740730  2.291157e-04                 0.045562   Latino  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get GBA1 from 23andMe datafame\n",
    "lat_gba = df_lat23[df_lat23.MARKERNAME == 'chr1:155205634']\n",
    "df_lat = pd.concat([df_lat,lat_gba])\n",
    "\n",
    "# Chromsome, position columns are objects -> convert to int before merging\n",
    "df_lat['CHROMOSOME'] = pd.to_numeric(df_lat['CHROMOSOME'], errors='coerce')\n",
    "# Some values were strings, so drop the na's \n",
    "df_lat = df_lat.dropna(subset=['CHROMOSOME'])\n",
    "df_lat['CHROMOSOME'] = df_lat['CHROMOSOME'].astype(int)\n",
    "df_lat['POSITION'] = pd.to_numeric(df_lat['POSITION'], errors='coerce')\n",
    "df_lat = df_lat.rename(columns={'CHROMOSOME':'Chromosome','POSITION':'Position'})\n",
    "\n",
    "# Merge with 90 risk loci\n",
    "lat = pd.merge(df_lat, df_nalls, how='inner', on=['Chromosome', 'Position'])\n",
    "\n",
    "# Add a column for ancestry\n",
    "lat['ancestry'] = 'Latino'\n",
    "\n",
    "# beta = ln(odds_ratio)\n",
    "lat['beta'] = np.log(lat['OR'])\n",
    "\n",
    "# Subset and rename columns\n",
    "lat = lat[['Locus','SNP','Chromosome','Position','EA','NEA','beta','P','EAF','ancestry']].rename(columns={\n",
    "    'EA':'effect_allele','NEA':'other_allele',\n",
    "    'P':'p_value','EAF':'effect_allele_frequency'})\n",
    "\n",
    "# Filter p < 0.05\n",
    "lat = lat[lat['p_value'] < 0.05]\n",
    "lat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "5a94577e-0d7c-413c-8d6b-ccaa62a1b152",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>ancestry</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMVK</td>\n",
       "      <td>rs114138760</td>\n",
       "      <td>1</td>\n",
       "      <td>154925709</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.281200</td>\n",
       "      <td>4.193000e-09</td>\n",
       "      <td>0.011200</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KRTCAP2</td>\n",
       "      <td>rs35749011</td>\n",
       "      <td>1</td>\n",
       "      <td>155162560</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.606800</td>\n",
       "      <td>1.720000e-70</td>\n",
       "      <td>0.016900</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GBAP1</td>\n",
       "      <td>rs76763715</td>\n",
       "      <td>1</td>\n",
       "      <td>155235843</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.746700</td>\n",
       "      <td>1.592000e-22</td>\n",
       "      <td>0.995300</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>rs6658353</td>\n",
       "      <td>1</td>\n",
       "      <td>161499264</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.065000</td>\n",
       "      <td>6.099000e-12</td>\n",
       "      <td>0.501100</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VAMP4</td>\n",
       "      <td>rs11578699</td>\n",
       "      <td>1</td>\n",
       "      <td>171750629</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.070400</td>\n",
       "      <td>4.468000e-09</td>\n",
       "      <td>0.194900</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>BIN3</td>\n",
       "      <td>rs2280104</td>\n",
       "      <td>8</td>\n",
       "      <td>22525980</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.214305</td>\n",
       "      <td>2.515152e-02</td>\n",
       "      <td>0.758349</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>ITGA8</td>\n",
       "      <td>rs896435</td>\n",
       "      <td>10</td>\n",
       "      <td>15557406</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.192646</td>\n",
       "      <td>2.817975e-02</td>\n",
       "      <td>0.673742</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>VPS13C</td>\n",
       "      <td>rs2251086</td>\n",
       "      <td>15</td>\n",
       "      <td>61997385</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.236127</td>\n",
       "      <td>4.489795e-02</td>\n",
       "      <td>0.855848</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>SETD1A</td>\n",
       "      <td>rs11150601</td>\n",
       "      <td>16</td>\n",
       "      <td>30977799</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.278549</td>\n",
       "      <td>9.806977e-04</td>\n",
       "      <td>0.558559</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>CRHR1</td>\n",
       "      <td>rs117615688</td>\n",
       "      <td>17</td>\n",
       "      <td>43798308</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.740730</td>\n",
       "      <td>2.291157e-04</td>\n",
       "      <td>0.045562</td>\n",
       "      <td>Latino</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>150 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Locus          SNP Chromosome   Position effect_allele other_allele  \\\n",
       "0      PMVK  rs114138760          1  154925709             C            G   \n",
       "1   KRTCAP2   rs35749011          1  155162560             A            G   \n",
       "2     GBAP1   rs76763715          1  155235843             T            C   \n",
       "3    FCGR2A    rs6658353          1  161499264             C            G   \n",
       "4     VAMP4   rs11578699          1  171750629             T            C   \n",
       "..      ...          ...        ...        ...           ...          ...   \n",
       "37     BIN3    rs2280104          8   22525980             C            T   \n",
       "42    ITGA8     rs896435         10   15557406             T            C   \n",
       "57   VPS13C    rs2251086         15   61997385             C            T   \n",
       "59   SETD1A   rs11150601         16   30977799             A            G   \n",
       "68    CRHR1  rs117615688         17   43798308             A            G   \n",
       "\n",
       "        beta       p_value  effect_allele_frequency  ancestry  \n",
       "0   0.281200  4.193000e-09                 0.011200  European  \n",
       "1   0.606800  1.720000e-70                 0.016900  European  \n",
       "2  -0.746700  1.592000e-22                 0.995300  European  \n",
       "3   0.065000  6.099000e-12                 0.501100  European  \n",
       "4  -0.070400  4.468000e-09                 0.194900  European  \n",
       "..       ...           ...                      ...       ...  \n",
       "37 -0.214305  2.515152e-02                 0.758349    Latino  \n",
       "42  0.192646  2.817975e-02                 0.673742    Latino  \n",
       "57  0.236127  4.489795e-02                 0.855848    Latino  \n",
       "59  0.278549  9.806977e-04                 0.558559    Latino  \n",
       "68 -0.740730  2.291157e-04                 0.045562    Latino  \n",
       "\n",
       "[150 rows x 10 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Concat dataframes\n",
    "df_out = pd.concat([eur,afr,ea,lat])\n",
    "df_out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "15b925a6-a658-4574-9388-203b26bc3ac0",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Get current date in YYYY-MM-DD format\n",
    "current_date = datetime.now().strftime('%Y-%m-%d')\n",
    "\n",
    "# Define the specific path to save the file\n",
    "save_path = '{WORK_DIR}/PD/processed_p_val_filtered/'\n",
    "\n",
    "# Define the filename with the current date, path\n",
    "filename = f'{current_date}_PD_PAR_data_for_analysis_pval.csv'\n",
    "full_path = os.path.join(save_path, filename)\n",
    "\n",
    "# Export the dataframe to CSV\n",
    "df_out.to_csv(full_path, index=False, header=True)\n",
    "\n",
    "print(f'DataFrame exported as {filename}')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fae1cafd-85a3-4b52-8a4f-117508ae3ead",
   "metadata": {},
   "source": [
    "### Select top hits for each ancestry and generate dataset for analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "58533d43-df8e-4f30-9b12-bb72336c516b",
   "metadata": {},
   "source": [
    "#### Based on previously identified risk loci\n",
    "Added on June 10th, 2024"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cee35e09-980c-4672-befb-20a64e7dfc49",
   "metadata": {},
   "source": [
    "##### Nalls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "68e5b257-92a9-425b-a3e9-160a36df9d1e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Locus</th>\n",
       "      <th>QTL Nominated Gene (nearest QTL)</th>\n",
       "      <th>Effect allele</th>\n",
       "      <th>Other allele</th>\n",
       "      <th>Effect allele frequency</th>\n",
       "      <th>Beta, all studies</th>\n",
       "      <th>SE, all studies</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq1, new studies</th>\n",
       "      <th>Beta, new studies</th>\n",
       "      <th>StdErr, new studies</th>\n",
       "      <th>P, new studies</th>\n",
       "      <th>I2, new studies</th>\n",
       "      <th>Passes pooled 23andMe QC</th>\n",
       "      <th>Known GWAS locus within 1MB</th>\n",
       "      <th>Failed final filtering and QC</th>\n",
       "      <th>Locus within 250KB</th>\n",
       "      <th>Locus Number</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs114138760</td>\n",
       "      <td>1</td>\n",
       "      <td>154898185</td>\n",
       "      <td>PMVK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.0112</td>\n",
       "      <td>0.2812</td>\n",
       "      <td>0.0478</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0109</td>\n",
       "      <td>0.1997</td>\n",
       "      <td>0.0843</td>\n",
       "      <td>1.779000e-02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs35749011</td>\n",
       "      <td>1</td>\n",
       "      <td>155135036</td>\n",
       "      <td>KRTCAP2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.0169</td>\n",
       "      <td>0.6068</td>\n",
       "      <td>0.0342</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0177</td>\n",
       "      <td>0.6798</td>\n",
       "      <td>0.0615</td>\n",
       "      <td>2.160000e-28</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs76763715</td>\n",
       "      <td>1</td>\n",
       "      <td>155205634</td>\n",
       "      <td>GBAP1</td>\n",
       "      <td>GBAP1</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.9953</td>\n",
       "      <td>-0.7467</td>\n",
       "      <td>0.0765</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9941</td>\n",
       "      <td>-0.6693</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>1.340000e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>F</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs6658353</td>\n",
       "      <td>1</td>\n",
       "      <td>161469054</td>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.5011</td>\n",
       "      <td>0.0650</td>\n",
       "      <td>0.0094</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5105</td>\n",
       "      <td>0.0662</td>\n",
       "      <td>0.0171</td>\n",
       "      <td>1.101000e-04</td>\n",
       "      <td>46.4</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs11578699</td>\n",
       "      <td>1</td>\n",
       "      <td>171719769</td>\n",
       "      <td>VAMP4</td>\n",
       "      <td>VAMP4</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.1949</td>\n",
       "      <td>-0.0704</td>\n",
       "      <td>0.0120</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1947</td>\n",
       "      <td>-0.0663</td>\n",
       "      <td>0.0226</td>\n",
       "      <td>3.313000e-03</td>\n",
       "      <td>13.4</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           SNP  Chromosome   Position    Locus  \\\n",
       "0  rs114138760           1  154898185     PMVK   \n",
       "1   rs35749011           1  155135036  KRTCAP2   \n",
       "2   rs76763715           1  155205634    GBAP1   \n",
       "3    rs6658353           1  161469054   FCGR2A   \n",
       "4   rs11578699           1  171719769    VAMP4   \n",
       "\n",
       "  QTL Nominated Gene (nearest QTL) Effect allele Other allele  \\\n",
       "0                              NaN             c            g   \n",
       "1                              NaN             a            g   \n",
       "2                            GBAP1             t            c   \n",
       "3                           FCGR2A             c            g   \n",
       "4                            VAMP4             t            c   \n",
       "\n",
       "   Effect allele frequency  Beta, all studies  SE, all studies  ...  \\\n",
       "0                   0.0112             0.2812           0.0478  ...   \n",
       "1                   0.0169             0.6068           0.0342  ...   \n",
       "2                   0.9953            -0.7467           0.0765  ...   \n",
       "3                   0.5011             0.0650           0.0094  ...   \n",
       "4                   0.1949            -0.0704           0.0120  ...   \n",
       "\n",
       "   Freq1, new studies  Beta, new studies  StdErr, new studies  P, new studies  \\\n",
       "0              0.0109             0.1997               0.0843    1.779000e-02   \n",
       "1              0.0177             0.6798               0.0615    2.160000e-28   \n",
       "2              0.9941            -0.6693               0.1269    1.340000e-07   \n",
       "3              0.5105             0.0662               0.0171    1.101000e-04   \n",
       "4              0.1947            -0.0663               0.0226    3.313000e-03   \n",
       "\n",
       "   I2, new studies  Passes pooled 23andMe QC  Known GWAS locus within 1MB  \\\n",
       "0              0.0                         T                            1   \n",
       "1              0.0                         T                            1   \n",
       "2              0.0                         F                            1   \n",
       "3             46.4                         T                            0   \n",
       "4             13.4                         T                            0   \n",
       "\n",
       "   Failed final filtering and QC  Locus within 250KB  Locus Number  \n",
       "0                              0                   1             1  \n",
       "1                              0                   1             1  \n",
       "2                              0                   1             1  \n",
       "3                              0                   0             2  \n",
       "4                              0                   0             3  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_nalls = pd.read_excel('{WORK_DIR}/90risk_loci.xlsx').rename(columns={'CHR':'Chromosome','BP':'Position', 'Nearest Gene':'Locus'})\n",
    "df_nalls.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "487737f2-b353-48fe-af1b-deba3c24e5ef",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Merge summary stats with 90 risk loci from Nalls\n",
    "eur_risk = pd.merge(eur, df_nalls, how='inner', on='SNP')\n",
    "eur_risk = eur_risk[['Locus', 'SNP', 'Chromosome_x', 'Position_x', 'effect_allele', 'other_allele',\n",
    "       'beta', 'p_value', 'effect_allele_frequency', 'ancestry']].rename(columns={'Chromosome_x':'Chromosome',\n",
    "                                                                                  'Position_x':'Position'})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f480a9e3-f8b0-40b7-9d08-1673b642796e",
   "metadata": {},
   "source": [
    "##### Rizig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a0fe8181-8b14-4f0d-b18c-489c78d84284",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Genome-wide significant SNPs identified in the African and African Admixed GWAS meta-analysis with frequency metrics\n",
    "df_rizig_gwsig = pd.read_excel('{WORK_DIR}/Rizig_Genomewide_signif.xlsx', sheet_name='Tab 2 ST2').rename(columns={'Gene.refGene':'Locus',\n",
    "                                                                                                       'rsID':'SNP'})\n",
    "\n",
    "# Add in LRRK2 for comparison (had a high p-val in summary stats, so was filtered out)\n",
    "list_afr_snps = ['rs76904798','rs34637584']\n",
    "afr_snps = df_afr[df_afr['rsid'].isin(list_afr_snps)]\n",
    "afr_snps = afr_snps[['rsid','chromosome','base_pair_location','effect_allele','other_allele','beta',\n",
    "                     'effect_allele_frequency','p_value','ancestry']].rename(columns={'chromosome':'Chromosome',\n",
    "                                                                                      'base_pair_location':'Position','rsid':'SNP'})\n",
    "afr_snps['Locus'] = 'LRRK2'\n",
    "\n",
    "# Merge summary stats with 90 risk loci from Nalls\n",
    "df_rizig = pd.concat([df_rizig_gwsig, df_nalls])\n",
    "afr_risk = pd.merge(afr, df_rizig, how='inner', on='SNP')\n",
    "afr_risk = afr_risk[['Locus', 'SNP', 'Chromosome_x', 'Position_x', 'effect_allele', 'other_allele',\n",
    "       'beta', 'p_value', 'effect_allele_frequency', 'ancestry']].rename(columns={'Chromosome_x':'Chromosome',\n",
    "                                                                                  'Position_x':'Position'})\n",
    "afr_risk = pd.concat([afr_risk,afr_snps])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd9014f9-ab6d-4c8b-9bfb-c7899fd44823",
   "metadata": {},
   "source": [
    "##### Foo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d15921b6-1712-44d5-a92c-df2245ff08c2",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Merge with 90 risk loci\n",
    "ea_risk = pd.merge(ea, df_nalls, how='inner', on=['Chromosome', 'Position'])\n",
    "\n",
    "# Add in reported top GWAS hits from Foo\n",
    "list_ea_snps = ['chr1:155210185','chr5:75599208','chr1:205756484','chr1:226846712','chr3:182735211','chr4:77101068','chr4:90682474','chr6:112151452',\n",
    "             'chr7:70750493','chr11:83510117','chr12:40387749','chr12:40757328','chr18:40678235']\n",
    "ea_snps = df_ea[df_ea['MARKERNAME'].isin(list_ea_snps)]\n",
    "\n",
    "# Fill in missing rsID's and locus names\n",
    "ea_snps['SNP'] = ea_snps['MARKERNAME'].map({'chr1:155210185':'rs146532106','chr5:75599208':'rs246814',\n",
    "                                            'chr1:205756484':'rs6679073','chr1:226846712':'rs16846351',\n",
    "                                            'chr3:182735211':'rs2292056','chr4:77101068':'rs3816248',\n",
    "                                            'chr4:90682474':'rs6826785','chr6:112151452':'rs1887316',\n",
    "                                            'chr7:70750493':'rs9638616','chr11:83510117':'rs12278023',\n",
    "                                            'chr12:40387749':'rs141336855','chr18:40678235':'rs4130047',\n",
    "                                            'chr12:40757328':'rs34778348'})\n",
    "ea_snps['Locus'] = ea_snps['MARKERNAME'].map({'chr1:155210185':'GBA','chr5:75599208':'SV2C',\n",
    "                                              'chr1:205756484':'PARK16','chr1:226846712':'ITPKB',\n",
    "                                              'chr3:182735211':'MCCC1','chr4:77101068':'FAM47E-SCARB2',\n",
    "                                              'chr4:90682474':'SNCA','chr6:112151452':'FYN',\n",
    "                                              'chr7:70750493':'WBSCR17','chr11:83510117':'DLG2',\n",
    "                                              'chr12:40387749':'LRRK2','chr18:40678235':'RIT2',\n",
    "                                              'chr12:40757328':'LRRK2'})\n",
    "\n",
    "# Subset and rename columns\n",
    "ea_snps = ea_snps[['Locus','SNP','CHROMOSOME','POSITION','EA','NEA','beta','P','EAF','ancestry']].rename(columns={\n",
    "    'CHROMOSOME':'Chromosome','POSITION':'Position','EA':'effect_allele','NEA':'other_allele',\n",
    "    'P':'p_value','EAF':'effect_allele_frequency'})\n",
    "\n",
    "# Combine with top hits\n",
    "ea_risk = pd.concat([ea_risk, ea_snps])\n",
    "ea_risk = ea_risk[['Locus', 'SNP', 'Chromosome', 'Position', 'effect_allele', 'other_allele',\n",
    "       'beta', 'p_value', 'effect_allele_frequency', 'ancestry']]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a782185e-6a4b-4258-b9c8-3c673a34727c",
   "metadata": {},
   "source": [
    "##### Loesch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4cbfd53a-1610-4fc1-8558-e9b6797d0f8f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Chromsome, position columns are objects -> convert to int before merging\n",
    "lat['Chromosome'] = pd.to_numeric(lat['Chromosome'], errors='coerce')\n",
    "# Some values were strings, so drop the na's \n",
    "lat = lat.dropna(subset=['Chromosome'])\n",
    "lat['Chromosome'] = lat['Chromosome'].astype(int)\n",
    "lat['Position'] = pd.to_numeric(lat['Position'], errors='coerce')\n",
    "\n",
    "# Merge with 90 risk loci\n",
    "lat_risk = pd.merge(lat, df_nalls, how='inner', on=['Chromosome', 'Position'])\n",
    "\n",
    "# Add in reported top GWAS hits from Loesch\n",
    "list_lat_snps = ['chr1:155205634','chr1:155135036','chr1:155206167','chr4:90626111','chr4:90626111','chr4:90744993','chr4:90471245',\n",
    "                'chr4:90819961','chr3:196357126','chr17:44051846','chr4:951947']\n",
    "lat_snps = df_lat[df_lat['MARKERNAME'].isin(list_lat_snps)]\n",
    "\n",
    "# Fill in missing rsID's and locus names\n",
    "lat_snps['SNP'] = lat_snps['MARKERNAME'].map({'chr1:155205634':'rs76763715','chr1:155135036':'rs35749011',\n",
    "                                              'chr1:155206167':'rs2230288','chr4:90626111':'rs356182',\n",
    "                                              'chr4:90643757':'rs356225','chr12:40734202':'rs34637584',\n",
    "                                              'chr4:90744993':'rs6830166','chr4:90471245':'rs2870004',\n",
    "                                              'chr4:90819961':'rs763443','chr3:196357126':'rs78820950',\n",
    "                                              'chr17:44051846':'rs1800547','chr4:951947':'rs34311866'})\n",
    "lat_snps['Locus'] = lat_snps['MARKERNAME'].map({'chr1:155205634':'GBA','chr1:155135036':'GBA',\n",
    "                                                'chr1:155206167':'GBA','chr4:90626111':'SNCA',\n",
    "                                                'chr4:90643757':'SNCA','chr12:40734202':'LRRK2',\n",
    "                                                'chr4:90744993':'SNCA','chr4:90471245':'SNCA',\n",
    "                                                'chr4:90819961':'MMRN1','chr3:196357126':'NRROS',\n",
    "                                                'chr17:44051846':'MAPT','chr4:951947':'TMEM175'})\n",
    "\n",
    "# Subset and rename columns\n",
    "lat_snps = lat_snps[['Locus','SNP','CHROMOSOME','POSITION','EA','NEA','beta','P','EAF','ancestry']].rename(columns={\n",
    "    'CHROMOSOME':'Chromosome','POSITION':'Position','EA':'effect_allele','NEA':'other_allele',\n",
    "    'P':'p_value','EAF':'effect_allele_frequency'})\n",
    "lat_snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2cfbd54f-6dde-4846-883c-bfcd7ecfed4d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Add in GBA data from 23andMe\n",
    "gba_snps = ['chr1:155206167','chr1:155135036','chr1:155205634']\n",
    "lat23_gba = df_lat23[df_lat23['MARKERNAME'].isin(gba_snps)]\n",
    "# Fill in missing locus, ancestry, beta, and rsID's\n",
    "lat23_gba['Locus'] = 'GBA'\n",
    "lat23_gba['ancestry'] = 'Latino'\n",
    "lat23_gba['beta'] = np.log(lat23_gba['OR'])\n",
    "lat23_gba['SNP'] = lat23_gba['MARKERNAME'].map({'chr1:155206167':'rs2230288','chr1:155135036':'rs35749011','chr1:155205634':'rs76763715'})\n",
    "lat23_gba"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5288ffbd-e52d-453b-b4d5-e00f1346e144",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Add in LRRK2 data from 23andMe\n",
    "lat23_lrrk2 = df_lat23[df_lat23['MARKERNAME'] == 'chr12:40610864']\n",
    "# Fill in missing locus, ancestry, beta, and rsID's\n",
    "lat23_lrrk2['Locus'] = 'LRRK2'\n",
    "lat23_lrrk2['ancestry'] = 'Latino'\n",
    "lat23_lrrk2['beta'] = np.log(lat23_lrrk2['OR'])\n",
    "lat23_lrrk2['SNP'] = 'rs11175620'\n",
    "lat23_lrrk2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "046f4e7c-922a-4b03-9022-8bed2b366ab5",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "lat23_snps = pd.concat([lat23_gba, lat23_lrrk2])\n",
    "lat23_snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7469e03c-03f0-4a87-bb5c-426df039cc92",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Subset and rename columns\n",
    "lat23_snps = lat23_snps[['Locus','SNP','CHROMOSOME','POSITION','EA','NEA','beta','P','EAF','ancestry']].rename(columns={\n",
    "    'CHROMOSOME':'Chromosome','POSITION':'Position','EA':'effect_allele','NEA':'other_allele',\n",
    "    'P':'p_value','EAF':'effect_allele_frequency'})\n",
    "lat23_snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52dc6021-eb26-4128-909b-32fac51bd1c9",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Concat all Latino data\n",
    "lat_risk = pd.concat([lat_risk, lat_snps, lat23_snps])\n",
    "lat_risk = lat_risk[['Locus', 'SNP', 'Chromosome', 'Position', 'effect_allele', 'other_allele',\n",
    "       'beta', 'p_value', 'effect_allele_frequency', 'ancestry']]\n",
    "lat_risk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "6fb7bab1-b3f3-4ae8-9db9-4af41cd1c643",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Concat all dataframes\n",
    "df_calcs = pd.concat([eur_risk, afr_risk, ea_risk, lat_risk])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46d82963-1546-4096-bbe4-a4c23590a463",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the specific path to save the file\n",
    "save_path = '{WORK_DIR}/PD/processed/'\n",
    "\n",
    "# Define the filename with the current date, path\n",
    "filename = f'{current_date}_PD_PAR_data_for_analysis.csv'\n",
    "full_path = os.path.join(save_path, filename)\n",
    "\n",
    "# Export the dataframe to CSV\n",
    "df_calcs.to_csv(full_path, index=False, header=True)\n",
    "\n",
    "print(f'DataFrame exported as {filename}')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90ac59f9-d968-4333-a8d4-b65974f7c926",
   "metadata": {},
   "source": [
    "## Checking additional SNPs of interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "a96ad69e-b030-4436-87d7-be4a1f054e4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# All pops\n",
    "list_check_all_pops = ['chr1:155236376','chr1:155236246','chr1:155235252','chr12:40310434','chr12:40340400','chr1:155206167','chr1:155206037','chr1:155205043','chr12:40704236','chr12:40734202']\n",
    "## Latino\n",
    "lat_snps = df_lat[df_lat['MARKERNAME'].isin(list_check_all_pops)]\n",
    "lat23_snps = df_lat23[df_lat23['MARKERNAME'].isin(list_check_all_pops)]\n",
    "## East Asian\n",
    "ea_snps = df_ea[df_ea['MARKERNAME'].isin(list_check_all_pops)]\n",
    "## European\n",
    "eur_snps = df_eur[df_eur['MarkerName'].isin(list_check_all_pops)]\n",
    "## African/African Admixed\n",
    "pattern = '|'.join(list_check_all_pops)\n",
    "afr_snps = df_afr[df_afr['variant_id'].str.contains(pattern, case=False, na=False)]\n",
    "\n",
    "# EAS only\n",
    "list_check_ea_only = ['chr12:40363526','chr12:40320043','chr12:40757328','chr12:40713845']\n",
    "ea_only_snps = df_ea[df_ea['MARKERNAME'].isin(list_check_ea_only)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "2c7e6d66-a9a3-4171-90c3-1210aed2c289",
   "metadata": {},
   "outputs": [],
   "source": [
    "lat23_snps = lat23_snps[['Chromosome','Position','effect_allele','other_allele','beta','p_value','effect_allele_frequency']]\n",
    "lat23_snps['ancestry'] = 'Latino'\n",
    "eur_snps = eur_snps[['SNP','Chromosome','Position','effect_allele','other_allele','beta','p_value','effect_allele_frequency']]\n",
    "eur_snps['ancestry'] = 'European'\n",
    "ea_only_snps = ea_only_snps[['Chromosome','Position','EA','NEA','BETA','P','EAF']].rename(columns={\n",
    "    'EA':'effect_allele','NEA':'other_allele',\n",
    "    'BETA':'beta','P':'p_value','EAF':'effect_allele_frequency'})\n",
    "ea_only_snps['ancestry'] = 'East Asian'\n",
    "check_snps_all = pd.concat([lat23_snps,eur_snps,ea_only_snps])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "9ceb7235-59dc-4210-8ea9-a832070545a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>ancestry</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>934915</th>\n",
       "      <td>1</td>\n",
       "      <td>155206037</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.583414</td>\n",
       "      <td>7.814800e-03</td>\n",
       "      <td>0.994590</td>\n",
       "      <td>Latino</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>934916</th>\n",
       "      <td>1</td>\n",
       "      <td>155206167</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.800830</td>\n",
       "      <td>9.571910e-06</td>\n",
       "      <td>0.005890</td>\n",
       "      <td>Latino</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1439210</th>\n",
       "      <td>12</td>\n",
       "      <td>40340400</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>2.428900</td>\n",
       "      <td>3.610000e-148</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>European</td>\n",
       "      <td>rs34637584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4191451</th>\n",
       "      <td>1</td>\n",
       "      <td>155236246</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.361900</td>\n",
       "      <td>1.420000e-07</td>\n",
       "      <td>0.010900</td>\n",
       "      <td>European</td>\n",
       "      <td>rs75548401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4191452</th>\n",
       "      <td>1</td>\n",
       "      <td>155236376</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.635700</td>\n",
       "      <td>9.125000e-48</td>\n",
       "      <td>0.016200</td>\n",
       "      <td>European</td>\n",
       "      <td>rs2230288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3151430</th>\n",
       "      <td>12</td>\n",
       "      <td>40757328</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.663400</td>\n",
       "      <td>4.680000e-24</td>\n",
       "      <td>0.036058</td>\n",
       "      <td>East Asian</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chromosome   Position effect_allele other_allele      beta  \\\n",
       "934915           1  155206037             G            A -0.583414   \n",
       "934916           1  155206167             T            C  0.800830   \n",
       "1439210         12   40340400             a            g  2.428900   \n",
       "4191451          1  155236246             a            g  0.361900   \n",
       "4191452          1  155236376             t            c  0.635700   \n",
       "3151430         12   40757328             A            G  0.663400   \n",
       "\n",
       "               p_value  effect_allele_frequency    ancestry         SNP  \n",
       "934915    7.814800e-03                 0.994590      Latino         NaN  \n",
       "934916    9.571910e-06                 0.005890      Latino         NaN  \n",
       "1439210  3.610000e-148                 0.001500    European  rs34637584  \n",
       "4191451   1.420000e-07                 0.010900    European  rs75548401  \n",
       "4191452   9.125000e-48                 0.016200    European   rs2230288  \n",
       "3151430   4.680000e-24                 0.036058  East Asian         NaN  "
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "check_snps_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "49c7d769-bdad-4c44-b113-8c0bb54f8eef",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Adding rsID, Locus, and Protein change for these variants\n",
    "check_snps_all['SNP'] = check_snps_all['Position'].map({155206037:'rs75548401',155206167:'rs2230288',\n",
    "                                              155236246:'rs75548401',155236376:'rs2230288',\n",
    "                                              40757328:'rs34778348',40340400:'rs34637584'})\n",
    "check_snps_all['Locus'] = check_snps_all['Position'].map({155206037:'GBA1',155206167:'GBA1',\n",
    "                                              155236246:'GBA1',155236376:'GBA1',\n",
    "                                              40757328:'LRRK2',40340400:'LRRK2'})\n",
    "check_snps_all['Protein change'] = check_snps_all['Position'].map({155206037:'T369M',155206167:'E326K',\n",
    "                                              155236246:'T369M',155236376:'E326K',\n",
    "                                              40757328:'G2385R',40340400:'G2019S'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "1f1be4cb-dcb6-4e62-a187-d1fce7631196",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>ancestry</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Protein change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>934915</th>\n",
       "      <td>1</td>\n",
       "      <td>155206037</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.583414</td>\n",
       "      <td>7.814800e-03</td>\n",
       "      <td>0.994590</td>\n",
       "      <td>Latino</td>\n",
       "      <td>rs75548401</td>\n",
       "      <td>GBA1</td>\n",
       "      <td>T369M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>934916</th>\n",
       "      <td>1</td>\n",
       "      <td>155206167</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.800830</td>\n",
       "      <td>9.571910e-06</td>\n",
       "      <td>0.005890</td>\n",
       "      <td>Latino</td>\n",
       "      <td>rs2230288</td>\n",
       "      <td>GBA1</td>\n",
       "      <td>E326K</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1439210</th>\n",
       "      <td>12</td>\n",
       "      <td>40340400</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2.428900</td>\n",
       "      <td>3.610000e-148</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>European</td>\n",
       "      <td>rs34637584</td>\n",
       "      <td>LRRK2</td>\n",
       "      <td>G2019S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4191451</th>\n",
       "      <td>1</td>\n",
       "      <td>155236246</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.361900</td>\n",
       "      <td>1.420000e-07</td>\n",
       "      <td>0.010900</td>\n",
       "      <td>European</td>\n",
       "      <td>rs75548401</td>\n",
       "      <td>GBA1</td>\n",
       "      <td>T369M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4191452</th>\n",
       "      <td>1</td>\n",
       "      <td>155236376</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.635700</td>\n",
       "      <td>9.125000e-48</td>\n",
       "      <td>0.016200</td>\n",
       "      <td>European</td>\n",
       "      <td>rs2230288</td>\n",
       "      <td>GBA1</td>\n",
       "      <td>E326K</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3151430</th>\n",
       "      <td>12</td>\n",
       "      <td>40757328</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.663400</td>\n",
       "      <td>4.680000e-24</td>\n",
       "      <td>0.036058</td>\n",
       "      <td>East Asian</td>\n",
       "      <td>rs34778348</td>\n",
       "      <td>LRRK2</td>\n",
       "      <td>G2385R</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chromosome   Position effect_allele other_allele      beta  \\\n",
       "934915           1  155206037             G            A -0.583414   \n",
       "934916           1  155206167             T            C  0.800830   \n",
       "1439210         12   40340400             A            G  2.428900   \n",
       "4191451          1  155236246             A            G  0.361900   \n",
       "4191452          1  155236376             T            C  0.635700   \n",
       "3151430         12   40757328             A            G  0.663400   \n",
       "\n",
       "               p_value  effect_allele_frequency    ancestry         SNP  \\\n",
       "934915    7.814800e-03                 0.994590      Latino  rs75548401   \n",
       "934916    9.571910e-06                 0.005890      Latino   rs2230288   \n",
       "1439210  3.610000e-148                 0.001500    European  rs34637584   \n",
       "4191451   1.420000e-07                 0.010900    European  rs75548401   \n",
       "4191452   9.125000e-48                 0.016200    European   rs2230288   \n",
       "3151430   4.680000e-24                 0.036058  East Asian  rs34778348   \n",
       "\n",
       "         Locus Protein change  \n",
       "934915    GBA1          T369M  \n",
       "934916    GBA1          E326K  \n",
       "1439210  LRRK2         G2019S  \n",
       "4191451   GBA1          T369M  \n",
       "4191452   GBA1          E326K  \n",
       "3151430  LRRK2         G2385R  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Some alleles are lowercase -> convert to uppercase\n",
    "check_snps_all['effect_allele'] = check_snps_all['effect_allele'].str.upper()\n",
    "check_snps_all['other_allele'] = check_snps_all['other_allele'].str.upper()\n",
    "check_snps_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "3fb2b183-2706-42d7-918b-b823880bfa8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Negative beta -> flipped alleles, new maf = 1 - maf\n",
    "check_snps_all['eaf_risk'] = check_snps_all['effect_allele_frequency'].where(check_snps_all['beta'] > 0, 1 - check_snps_all['effect_allele_frequency'])\n",
    "\n",
    "# Absolute value of beta\n",
    "check_snps_all['beta_abs'] = check_snps_all['beta'].abs()\n",
    "\n",
    "# Risk allele\n",
    "check_snps_all['risk_allele'] = np.where(check_snps_all['beta'] < 0, check_snps_all['other_allele'], check_snps_all['effect_allele'])\n",
    "\n",
    "# Calculate the new odds ratio with beta_abs\n",
    "check_snps_all['odds_ratio_new'] = np.exp(check_snps_all['beta_abs'])\n",
    "\n",
    "#Calculate the population attributable risk (PAR = MAF(OR - 1)/1 + MAF(OR - 1))\n",
    "check_snps_all['par'] = (check_snps_all['eaf_risk']*(check_snps_all['odds_ratio_new']-1)) / (1 + check_snps_all['eaf_risk']*(check_snps_all['odds_ratio_new']-1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "f9f3bc7e-52da-49e0-b0bf-c5ed1b952fd5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>ancestry</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Locus</th>\n",
       "      <th>Protein change</th>\n",
       "      <th>eaf_risk</th>\n",
       "      <th>beta_abs</th>\n",
       "      <th>risk_allele</th>\n",
       "      <th>odds_ratio_new</th>\n",
       "      <th>par</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>934915</th>\n",
       "      <td>1</td>\n",
       "      <td>155206037</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.583414</td>\n",
       "      <td>7.814800e-03</td>\n",
       "      <td>0.994590</td>\n",
       "      <td>Latino</td>\n",
       "      <td>rs75548401</td>\n",
       "      <td>GBA1</td>\n",
       "      <td>T369M</td>\n",
       "      <td>0.005410</td>\n",
       "      <td>0.583414</td>\n",
       "      <td>A</td>\n",
       "      <td>1.792146</td>\n",
       "      <td>0.004267</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>934916</th>\n",
       "      <td>1</td>\n",
       "      <td>155206167</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.800830</td>\n",
       "      <td>9.571910e-06</td>\n",
       "      <td>0.005890</td>\n",
       "      <td>Latino</td>\n",
       "      <td>rs2230288</td>\n",
       "      <td>GBA1</td>\n",
       "      <td>E326K</td>\n",
       "      <td>0.005890</td>\n",
       "      <td>0.800830</td>\n",
       "      <td>T</td>\n",
       "      <td>2.227389</td>\n",
       "      <td>0.007177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1439210</th>\n",
       "      <td>12</td>\n",
       "      <td>40340400</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>2.428900</td>\n",
       "      <td>3.610000e-148</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>European</td>\n",
       "      <td>rs34637584</td>\n",
       "      <td>LRRK2</td>\n",
       "      <td>G2019S</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>2.428900</td>\n",
       "      <td>A</td>\n",
       "      <td>11.346394</td>\n",
       "      <td>0.015282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4191451</th>\n",
       "      <td>1</td>\n",
       "      <td>155236246</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.361900</td>\n",
       "      <td>1.420000e-07</td>\n",
       "      <td>0.010900</td>\n",
       "      <td>European</td>\n",
       "      <td>rs75548401</td>\n",
       "      <td>GBA1</td>\n",
       "      <td>T369M</td>\n",
       "      <td>0.010900</td>\n",
       "      <td>0.361900</td>\n",
       "      <td>A</td>\n",
       "      <td>1.436055</td>\n",
       "      <td>0.004731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4191452</th>\n",
       "      <td>1</td>\n",
       "      <td>155236376</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.635700</td>\n",
       "      <td>9.125000e-48</td>\n",
       "      <td>0.016200</td>\n",
       "      <td>European</td>\n",
       "      <td>rs2230288</td>\n",
       "      <td>GBA1</td>\n",
       "      <td>E326K</td>\n",
       "      <td>0.016200</td>\n",
       "      <td>0.635700</td>\n",
       "      <td>T</td>\n",
       "      <td>1.888344</td>\n",
       "      <td>0.014187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3151430</th>\n",
       "      <td>12</td>\n",
       "      <td>40757328</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.663400</td>\n",
       "      <td>4.680000e-24</td>\n",
       "      <td>0.036058</td>\n",
       "      <td>East Asian</td>\n",
       "      <td>rs34778348</td>\n",
       "      <td>LRRK2</td>\n",
       "      <td>G2385R</td>\n",
       "      <td>0.036058</td>\n",
       "      <td>0.663400</td>\n",
       "      <td>A</td>\n",
       "      <td>1.941382</td>\n",
       "      <td>0.032830</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chromosome   Position effect_allele other_allele      beta  \\\n",
       "934915           1  155206037             G            A -0.583414   \n",
       "934916           1  155206167             T            C  0.800830   \n",
       "1439210         12   40340400             A            G  2.428900   \n",
       "4191451          1  155236246             A            G  0.361900   \n",
       "4191452          1  155236376             T            C  0.635700   \n",
       "3151430         12   40757328             A            G  0.663400   \n",
       "\n",
       "               p_value  effect_allele_frequency    ancestry         SNP  \\\n",
       "934915    7.814800e-03                 0.994590      Latino  rs75548401   \n",
       "934916    9.571910e-06                 0.005890      Latino   rs2230288   \n",
       "1439210  3.610000e-148                 0.001500    European  rs34637584   \n",
       "4191451   1.420000e-07                 0.010900    European  rs75548401   \n",
       "4191452   9.125000e-48                 0.016200    European   rs2230288   \n",
       "3151430   4.680000e-24                 0.036058  East Asian  rs34778348   \n",
       "\n",
       "         Locus Protein change  eaf_risk  beta_abs risk_allele  odds_ratio_new  \\\n",
       "934915    GBA1          T369M  0.005410  0.583414           A        1.792146   \n",
       "934916    GBA1          E326K  0.005890  0.800830           T        2.227389   \n",
       "1439210  LRRK2         G2019S  0.001500  2.428900           A       11.346394   \n",
       "4191451   GBA1          T369M  0.010900  0.361900           A        1.436055   \n",
       "4191452   GBA1          E326K  0.016200  0.635700           T        1.888344   \n",
       "3151430  LRRK2         G2385R  0.036058  0.663400           A        1.941382   \n",
       "\n",
       "              par  \n",
       "934915   0.004267  \n",
       "934916   0.007177  \n",
       "1439210  0.015282  \n",
       "4191451  0.004731  \n",
       "4191452  0.014187  \n",
       "3151430  0.032830  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "check_snps_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "8f4f6b06-4b3e-4bc1-beb3-f3eac9e19656",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the specific path to save the file\n",
    "save_path = '{WORK_DIR}/PD/known_variants/'\n",
    "\n",
    "# Define the filename with the current date, path\n",
    "filename = f'{current_date}_PD_PAR_lrrk2_gba1.csv'\n",
    "full_path = os.path.join(save_path, filename)\n",
    "\n",
    "# Export the dataframe to CSV\n",
    "check_snps_all.to_csv(full_path, index=False, header=True)\n",
    "\n",
    "print(f'DataFrame exported as {filename}')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python/3.9",
   "language": "python",
   "name": "py3.9"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
