{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "6337c24d-bfbd-456e-869f-d483e97c2576",
   "metadata": {},
   "source": [
    "# Prepare AD Summary Statistics\n",
    "* Project: Cross-ancestry PAR analysis\n",
    "* Version: Python/3.12\n",
    "* Status: COMPLETE\n",
    "* Last Updated: 20-OCT-2025"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f06ec6f-0abf-4c7a-b960-16b3c1da130f",
   "metadata": {},
   "source": [
    "## Notebook Overview\n",
    "* Import datasets for each ancestry\n",
    "* Clean and filter p < 0.05\n",
    "* Select top hits for each ancestry and generate dataset for calculations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bcc5956a-c934-4257-8bea-94347562ecbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import packages\n",
    "from datetime import datetime\n",
    "import os\n",
    "import glob\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import sys\n",
    "import openpyxl\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cbf9bf46-30b1-4406-bdd7-98b8de66febb",
   "metadata": {},
   "source": [
    "### European data from Bellenguez\n",
    "Annotated differently from top hits summary stats, so matching chromosome/position will not work. Will need to annotate using ANNOVAR before merging with top hits."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7ec9bcb2-30d3-4dd1-8176-97dd2b3e5020",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>n_cases</th>\n",
       "      <th>n_controls</th>\n",
       "      <th>het_isq</th>\n",
       "      <th>het_pvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr11:12541586:A:G</td>\n",
       "      <td>0.94860</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.3748</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>0.0106</td>\n",
       "      <td>36659</td>\n",
       "      <td>63137</td>\n",
       "      <td>32.7</td>\n",
       "      <td>0.1371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr8:102803998:G:A</td>\n",
       "      <td>0.19910</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0026</td>\n",
       "      <td>-0.1579</td>\n",
       "      <td>0.1230</td>\n",
       "      <td>32228</td>\n",
       "      <td>42115</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.7297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr13:55101557:T:C</td>\n",
       "      <td>0.56260</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.9856</td>\n",
       "      <td>-0.0252</td>\n",
       "      <td>0.0436</td>\n",
       "      <td>36227</td>\n",
       "      <td>62057</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.3599</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr7:17885109:C:G</td>\n",
       "      <td>0.01901</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.9989</td>\n",
       "      <td>0.5638</td>\n",
       "      <td>0.2404</td>\n",
       "      <td>23331</td>\n",
       "      <td>28992</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.6406</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:97729697:A:G</td>\n",
       "      <td>0.81340</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0183</td>\n",
       "      <td>-0.0126</td>\n",
       "      <td>0.0532</td>\n",
       "      <td>36131</td>\n",
       "      <td>56626</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.9542</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           MarkerName  p_value effect_allele other_allele  \\\n",
       "0  chr11:12541586:A:G  0.94860             A            G   \n",
       "1  chr8:102803998:G:A  0.19910             A            G   \n",
       "2  chr13:55101557:T:C  0.56260             T            C   \n",
       "3   chr7:17885109:C:G  0.01901             C            G   \n",
       "4   chr3:97729697:A:G  0.81340             A            G   \n",
       "\n",
       "   effect_allele_frequency    beta  standard_error  n_cases  n_controls  \\\n",
       "0                   0.3748  0.0007          0.0106    36659       63137   \n",
       "1                   0.0026 -0.1579          0.1230    32228       42115   \n",
       "2                   0.9856 -0.0252          0.0436    36227       62057   \n",
       "3                   0.9989  0.5638          0.2404    23331       28992   \n",
       "4                   0.0183 -0.0126          0.0532    36131       56626   \n",
       "\n",
       "   het_isq  het_pvalue  \n",
       "0     32.7      0.1371  \n",
       "1      0.0      0.7297  \n",
       "2      9.0      0.3599  \n",
       "3      0.0      0.6406  \n",
       "4      0.0      0.9542  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Without proxies\n",
    "df_eur = pd.read_csv('{WORK_DIR}/noProxy/EADB-minus-UKB_Nov2022.tsv.gz', sep='\\t', header=0)\n",
    "df_eur.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ac176ade-f727-4155-a1a2-60e535fb3048",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1591</th>\n",
       "      <td>6</td>\n",
       "      <td>32689933</td>\n",
       "      <td>32689933</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6824</th>\n",
       "      <td>19</td>\n",
       "      <td>44946776</td>\n",
       "      <td>44946776</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14557</th>\n",
       "      <td>11</td>\n",
       "      <td>60327769</td>\n",
       "      <td>60327769</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23540</th>\n",
       "      <td>11</td>\n",
       "      <td>86069406</td>\n",
       "      <td>86069406</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27778</th>\n",
       "      <td>19</td>\n",
       "      <td>44749457</td>\n",
       "      <td>44749457</td>\n",
       "      <td>ATAT</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20999262</th>\n",
       "      <td>19</td>\n",
       "      <td>44829875</td>\n",
       "      <td>44829875</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21069925</th>\n",
       "      <td>19</td>\n",
       "      <td>45134987</td>\n",
       "      <td>45134987</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21072633</th>\n",
       "      <td>8</td>\n",
       "      <td>27603975</td>\n",
       "      <td>27603975</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21085415</th>\n",
       "      <td>17</td>\n",
       "      <td>63476980</td>\n",
       "      <td>63476980</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21090306</th>\n",
       "      <td>2</td>\n",
       "      <td>127115046</td>\n",
       "      <td>127115046</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3035 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr      Start        End   Ref Alt\n",
       "1591       6   32689933   32689933     A   G\n",
       "6824      19   44946776   44946776     C   T\n",
       "14557     11   60327769   60327769     T   C\n",
       "23540     11   86069406   86069406     T   C\n",
       "27778     19   44749457   44749457  ATAT   A\n",
       "...       ..        ...        ...   ...  ..\n",
       "20999262  19   44829875   44829875     T   G\n",
       "21069925  19   45134987   45134987     C   T\n",
       "21072633   8   27603975   27603975     C   A\n",
       "21085415  17   63476980   63476980     C   T\n",
       "21090306   2  127115046  127115046     A   G\n",
       "\n",
       "[3035 rows x 5 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eur = df_eur[df_eur['p_value'] < 0.00000005].drop_duplicates(subset=['MarkerName'])\n",
    "\n",
    "# Split the MarkerName column into different columns\n",
    "eur[['Chromosome', 'Position', 'Ref', 'Alt']] = eur['MarkerName'].str.split(':', expand=True)\n",
    "eur['Chromosome'] = eur['Chromosome'].str.replace('chr', '')\n",
    "\n",
    "# Prep annovar input \n",
    "df_annovar_eur = pd.DataFrame()\n",
    "df_annovar_eur['Chr'] = eur['Chromosome']\n",
    "df_annovar_eur['Start'] = eur['Position']\n",
    "df_annovar_eur['End'] = eur['Position']\n",
    "df_annovar_eur['Ref'] = eur['Ref']\n",
    "df_annovar_eur['Alt'] = eur['Alt']\n",
    "df_annovar_eur"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "34d48184-fcb2-4dd3-9c29-a44ac6eff511",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Export ANNOVAR input file\n",
    "df_annovar_eur.to_csv('{WORK_DIR}/EUR_no_proxy_for_annovar.txt', sep=' ', index=False, header=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c501d6cc-5792-4d60-886b-4666cb3041ae",
   "metadata": {},
   "source": [
    "#### Run ANNOVAR to annotate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a9e25479-31d5-46a7-9ae1-8cbbb90b02a5",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[+] Loading annovar 2020-06-08 on cn4272 \n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile EUR_no_proxy_for_annovar.txt.refGene -exonsort -nofirstcodondel EUR_no_proxy_for_annovar.txt /fdb/annovar/current/hg38 -splicing 15>\n",
      "NOTICE: Output files are written to EUR_no_proxy_for_annovar.txt.refGene.variant_function, EUR_no_proxy_for_annovar.txt.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /fdb/annovar/current/hg38/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2936 unique variants in 2936 input lines\n",
      "NOTICE: Reading FASTA sequences from /fdb/annovar/current/hg38/hg38_refGeneMrna.fa ... Done with 108 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Variants with invalid input format are written to EUR_no_proxy_for_annovar.txt.refGene.invalid_input\n",
      "\n",
      "NOTICE: Running with system command <coding_change.pl  EUR_no_proxy_for_annovar.txt.refGene.exonic_variant_function.orig /fdb/annovar/current/hg38/hg38_refGene.txt /fdb/annovar/current/hg38/hg38_refGeneMrna.fa -alltranscript -out EUR_no_proxy_for_annovar.txt.refGene.fa -newevf EUR_no_proxy_for_annovar.txt.refGene.exonic_variant_function>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=avsnp150\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype avsnp150 -buildver hg38 -outfile EUR_no_proxy_for_annovar.txt EUR_no_proxy_for_annovar.txt /fdb/annovar/current/hg38>\n",
      "NOTICE: Output file with variants matching filtering criteria is written to EUR_no_proxy_for_annovar.txt.hg38_avsnp150_dropped, and output file with other variants is written to EUR_no_proxy_for_annovar.txt.hg38_avsnp150_filtered\n",
      "NOTICE: Processing next batch with 2936 unique variants in 3035 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28304406 and the number of bins to be scanned is 2633\n",
      "NOTICE: Scanning filter database /fdb/annovar/current/hg38/hg38_avsnp150.txt...Done\n",
      "NOTICE: Variants with invalid input format are written to EUR_no_proxy_for_annovar.txt.invalid_input\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=ljb26_all\n",
      "NOTICE: Finished reading 25 column headers for '-dbtype ljb26_all'\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype ljb26_all -buildver hg38 -outfile EUR_no_proxy_for_annovar.txt EUR_no_proxy_for_annovar.txt /fdb/annovar/current/hg38 -otherinfo>\n",
      "NOTICE: the --dbtype ljb26_all is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to EUR_no_proxy_for_annovar.txt.hg38_ljb26_all_dropped, and output file with other variants is written to EUR_no_proxy_for_annovar.txt.hg38_ljb26_all_filtered\n",
      "NOTICE: Processing next batch with 2936 unique variants in 3035 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 555944 and the number of bins to be scanned is 107\n",
      "NOTICE: Scanning filter database /fdb/annovar/current/hg38/hg38_ljb26_all.txt...Done\n",
      "NOTICE: Variants with invalid input format are written to EUR_no_proxy_for_annovar.txt.invalid_input\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=gnomad312_genome\n",
      "NOTICE: Finished reading 16 column headers for '-dbtype gnomad312_genome'\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype gnomad312_genome -buildver hg38 -outfile EUR_no_proxy_for_annovar.txt EUR_no_proxy_for_annovar.txt /fdb/annovar/current/hg38 -otherinfo>\n",
      "NOTICE: Output file with variants matching filtering criteria is written to EUR_no_proxy_for_annovar.txt.hg38_gnomad312_genome_dropped, and output file with other variants is written to EUR_no_proxy_for_annovar.txt.hg38_gnomad312_genome_filtered\n",
      "NOTICE: Processing next batch with 2936 unique variants in 3035 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 2858146 and the number of bins to be scanned is 1539\n",
      "NOTICE: Scanning filter database /fdb/annovar/current/hg38/hg38_gnomad312_genome.txt...Done\n",
      "NOTICE: Variants with invalid input format are written to EUR_no_proxy_for_annovar.txt.invalid_input\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to EUR_no_proxy_for_annovar.txt.hg38_multianno.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "module load annovar\n",
    "\n",
    "table_annovar.pl EUR_no_proxy_for_annovar.txt $ANNOVAR_DATA/hg38 \\\n",
    "-buildver hg38 \\\n",
    "-arg '-splicing 15',,, \\\n",
    "-remove \\\n",
    "-protocol refGene,avsnp150,ljb26_all,gnomad312_genome \\\n",
    "-operation g,f,f,f \\\n",
    "-nastring . \\\n",
    "-polish"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4556962e-139f-4aea-beec-4dc93d8a13e4",
   "metadata": {},
   "source": [
    "#### ANNOVAR output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0b576a2f-894c-4f29-9664-7876833436fa",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomad312_AF_afr</th>\n",
       "      <th>gnomad312_AF_ami</th>\n",
       "      <th>gnomad312_AF_amr</th>\n",
       "      <th>gnomad312_AF_asj</th>\n",
       "      <th>gnomad312_AF_eas</th>\n",
       "      <th>gnomad312_AF_fin</th>\n",
       "      <th>gnomad312_AF_mid</th>\n",
       "      <th>gnomad312_AF_nfe</th>\n",
       "      <th>gnomad312_AF_oth</th>\n",
       "      <th>gnomad312_AF_sas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6</td>\n",
       "      <td>32689933</td>\n",
       "      <td>32689933</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>HLA-DQB1;HLA-DQA2</td>\n",
       "      <td>dist=23276;dist=51458</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0449</td>\n",
       "      <td>0.1629</td>\n",
       "      <td>0.2200</td>\n",
       "      <td>0.2053</td>\n",
       "      <td>0.1295</td>\n",
       "      <td>0.1245</td>\n",
       "      <td>0.1384</td>\n",
       "      <td>0.1158</td>\n",
       "      <td>0.1116</td>\n",
       "      <td>0.1021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>19</td>\n",
       "      <td>44946776</td>\n",
       "      <td>44946776</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ncRNA_intronic</td>\n",
       "      <td>APOC4-APOC2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5836</td>\n",
       "      <td>0.4346</td>\n",
       "      <td>0.5655</td>\n",
       "      <td>0.4398</td>\n",
       "      <td>0.5610</td>\n",
       "      <td>0.4893</td>\n",
       "      <td>0.3861</td>\n",
       "      <td>0.4547</td>\n",
       "      <td>0.4909</td>\n",
       "      <td>0.5810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11</td>\n",
       "      <td>60327769</td>\n",
       "      <td>60327769</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MS4A6E</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2732</td>\n",
       "      <td>0.2852</td>\n",
       "      <td>0.3920</td>\n",
       "      <td>0.2998</td>\n",
       "      <td>0.4075</td>\n",
       "      <td>0.5789</td>\n",
       "      <td>0.2025</td>\n",
       "      <td>0.3583</td>\n",
       "      <td>0.3319</td>\n",
       "      <td>0.1916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>11</td>\n",
       "      <td>86069406</td>\n",
       "      <td>86069406</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PICALM</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8912</td>\n",
       "      <td>0.6527</td>\n",
       "      <td>0.7834</td>\n",
       "      <td>0.7597</td>\n",
       "      <td>0.6900</td>\n",
       "      <td>0.8276</td>\n",
       "      <td>0.7229</td>\n",
       "      <td>0.8061</td>\n",
       "      <td>0.7955</td>\n",
       "      <td>0.7417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>19</td>\n",
       "      <td>44749457</td>\n",
       "      <td>44749457</td>\n",
       "      <td>ATAT</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 52 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chr     Start       End   Ref Alt    Func.refGene       Gene.refGene  \\\n",
       "0    6  32689933  32689933     A   G      intergenic  HLA-DQB1;HLA-DQA2   \n",
       "1   19  44946776  44946776     C   T  ncRNA_intronic        APOC4-APOC2   \n",
       "2   11  60327769  60327769     T   C        intronic             MS4A6E   \n",
       "3   11  86069406  86069406     T   C        intronic             PICALM   \n",
       "4   19  44749457  44749457  ATAT   A               .                  .   \n",
       "\n",
       "      GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  ...  \\\n",
       "0  dist=23276;dist=51458                  .                .  ...   \n",
       "1                      .                  .                .  ...   \n",
       "2                      .                  .                .  ...   \n",
       "3                      .                  .                .  ...   \n",
       "4                      .                  .                .  ...   \n",
       "\n",
       "  gnomad312_AF_afr gnomad312_AF_ami gnomad312_AF_amr gnomad312_AF_asj  \\\n",
       "0           0.0449           0.1629           0.2200           0.2053   \n",
       "1           0.5836           0.4346           0.5655           0.4398   \n",
       "2           0.2732           0.2852           0.3920           0.2998   \n",
       "3           0.8912           0.6527           0.7834           0.7597   \n",
       "4                .                .                .                .   \n",
       "\n",
       "  gnomad312_AF_eas gnomad312_AF_fin gnomad312_AF_mid gnomad312_AF_nfe  \\\n",
       "0           0.1295           0.1245           0.1384           0.1158   \n",
       "1           0.5610           0.4893           0.3861           0.4547   \n",
       "2           0.4075           0.5789           0.2025           0.3583   \n",
       "3           0.6900           0.8276           0.7229           0.8061   \n",
       "4                .                .                .                .   \n",
       "\n",
       "  gnomad312_AF_oth gnomad312_AF_sas  \n",
       "0           0.1116           0.1021  \n",
       "1           0.4909           0.5810  \n",
       "2           0.3319           0.1916  \n",
       "3           0.7955           0.7417  \n",
       "4                .                .  \n",
       "\n",
       "[5 rows x 52 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read in ANNOVAR output\n",
    "annovar_output_eur = pd.read_csv('{WORK_DIR}/AD/EUR_no_proxy_for_annovar.txt.hg38_multianno.txt', sep=\"\\t\")\n",
    "annovar_output_eur.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7ed0f9a1-b862-45f2-8fa1-c3cb00f05bfc",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>avsnp150</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6:32689933:A:G</td>\n",
       "      <td>rs9275207</td>\n",
       "      <td>6</td>\n",
       "      <td>32689933</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>HLA-DQB1;HLA-DQA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19:44946776:C:T</td>\n",
       "      <td>rs9304644</td>\n",
       "      <td>19</td>\n",
       "      <td>44946776</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ncRNA_intronic</td>\n",
       "      <td>APOC4-APOC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr11:60327769:T:C</td>\n",
       "      <td>rs12803929</td>\n",
       "      <td>11</td>\n",
       "      <td>60327769</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MS4A6E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr11:86069406:T:C</td>\n",
       "      <td>rs3016327</td>\n",
       "      <td>11</td>\n",
       "      <td>86069406</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PICALM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19:44749457:ATAT:A</td>\n",
       "      <td>.</td>\n",
       "      <td>19</td>\n",
       "      <td>44749457</td>\n",
       "      <td>ATAT</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              MarkerName    avsnp150  Chr     Start   Ref Alt    Func.refGene  \\\n",
       "0      chr6:32689933:A:G   rs9275207    6  32689933     A   G      intergenic   \n",
       "1     chr19:44946776:C:T   rs9304644   19  44946776     C   T  ncRNA_intronic   \n",
       "2     chr11:60327769:T:C  rs12803929   11  60327769     T   C        intronic   \n",
       "3     chr11:86069406:T:C   rs3016327   11  86069406     T   C        intronic   \n",
       "4  chr19:44749457:ATAT:A           .   19  44749457  ATAT   A               .   \n",
       "\n",
       "        Gene.refGene  \n",
       "0  HLA-DQB1;HLA-DQA2  \n",
       "1        APOC4-APOC2  \n",
       "2             MS4A6E  \n",
       "3             PICALM  \n",
       "4                  .  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Format and clean columns\n",
    "annovar_output_eur['MarkerName'] = 'chr' + annovar_output_eur['Chr'].astype(str) + ':' + annovar_output_eur['Start'].astype(str) + ':' + annovar_output_eur['Ref'].astype(str) + ':' + annovar_output_eur['Alt'].astype(str)\n",
    "\n",
    "annovar_output_formatted_eur = annovar_output_eur[['MarkerName', 'avsnp150', 'Chr', 'Start', 'Ref', 'Alt', 'Func.refGene', 'Gene.refGene']]\n",
    "annovar_output_formatted_eur.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e9817085-919d-4618-9caf-55beb30e8bcf",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['MarkerName', 'p_value', 'effect_allele', 'other_allele',\n",
       "       'effect_allele_frequency', 'beta', 'standard_error', 'n_cases',\n",
       "       'n_controls', 'het_isq', 'het_pvalue', 'Chromosome', 'Position',\n",
       "       'Ref_x', 'Alt_x', 'avsnp150', 'Chr', 'Start', 'Ref_y', 'Alt_y',\n",
       "       'Func.refGene', 'Gene.refGene'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge annotated output with original stats\n",
    "merge_stats_eur = pd.merge(eur, annovar_output_formatted_eur, how = 'left', on='MarkerName')\n",
    "merge_stats_eur.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b01cf92c-4651-4171-9fb1-78709207e372",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HLA-DQB1;HLA-DQA2</td>\n",
       "      <td>rs9275207</td>\n",
       "      <td>6</td>\n",
       "      <td>32689933</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0997</td>\n",
       "      <td>1.296000e-08</td>\n",
       "      <td>0.9022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>APOC4-APOC2</td>\n",
       "      <td>rs9304644</td>\n",
       "      <td>19</td>\n",
       "      <td>44946776</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.1022</td>\n",
       "      <td>4.801000e-23</td>\n",
       "      <td>0.4508</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MS4A6E</td>\n",
       "      <td>rs12803929</td>\n",
       "      <td>11</td>\n",
       "      <td>60327769</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.0588</td>\n",
       "      <td>3.016000e-08</td>\n",
       "      <td>0.6467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PICALM</td>\n",
       "      <td>rs3016327</td>\n",
       "      <td>11</td>\n",
       "      <td>86069406</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.1028</td>\n",
       "      <td>1.578000e-15</td>\n",
       "      <td>0.1956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>19</td>\n",
       "      <td>44749457</td>\n",
       "      <td>A</td>\n",
       "      <td>ATAT</td>\n",
       "      <td>-0.1291</td>\n",
       "      <td>6.791000e-26</td>\n",
       "      <td>0.2867</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Locus         SNP Chromosome  Position effect_allele  \\\n",
       "0  HLA-DQB1;HLA-DQA2   rs9275207          6  32689933             A   \n",
       "1        APOC4-APOC2   rs9304644         19  44946776             T   \n",
       "2             MS4A6E  rs12803929         11  60327769             T   \n",
       "3             PICALM   rs3016327         11  86069406             T   \n",
       "4                  .           .         19  44749457             A   \n",
       "\n",
       "  other_allele    beta       p_value  effect_allele_frequency  \n",
       "0            G  0.0997  1.296000e-08                   0.9022  \n",
       "1            C  0.1022  4.801000e-23                   0.4508  \n",
       "2            C -0.0588  3.016000e-08                   0.6467  \n",
       "3            C -0.1028  1.578000e-15                   0.1956  \n",
       "4         ATAT -0.1291  6.791000e-26                   0.2867  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Subset columns\n",
    "merge_stats_eur_subset = merge_stats_eur[['Gene.refGene','avsnp150','Chromosome','Position','effect_allele','other_allele','beta','p_value','effect_allele_frequency']].rename(columns={\n",
    "    'Gene.refGene':'Locus','avsnp150':'SNP'})\n",
    "merge_stats_eur_subset.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "adb0f1de-f92e-4549-8069-5a0a8d283c42",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Add a column for ancestry\n",
    "merge_stats_eur_subset['ancestry'] = 'European'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e81381c7-1ef7-4717-9534-8cca1e636802",
   "metadata": {},
   "source": [
    "### European *APOE* data from Kunkle 2019"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "737bc9c8-8ffd-4b15-a8cf-112f21942dd4",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2805535/3944048298.py:1: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "  df_k = pd.read_csv('/data/CARD/AD/summary_stats/Kunkle_2019/Kunkle_etal_2019_IGAP_Summary_statistics.with_allelefreqs.txt', delim_whitespace=True, low_memory=False)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Effect_allele</th>\n",
       "      <th>Non_Effect_allele</th>\n",
       "      <th>Beta</th>\n",
       "      <th>SE</th>\n",
       "      <th>Pvalue</th>\n",
       "      <th>Effect_allele_freq</th>\n",
       "      <th>Test_MarkerName</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>100000012</td>\n",
       "      <td>rs10875231</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.0026</td>\n",
       "      <td>0.0168</td>\n",
       "      <td>0.8758</td>\n",
       "      <td>0.2406</td>\n",
       "      <td>1:100000012:G_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>100000827</td>\n",
       "      <td>rs6678176</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0008</td>\n",
       "      <td>0.0156</td>\n",
       "      <td>0.9574</td>\n",
       "      <td>0.2926</td>\n",
       "      <td>1:100000827:C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>100000843</td>\n",
       "      <td>rs78286437</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.0136</td>\n",
       "      <td>0.0330</td>\n",
       "      <td>0.6792</td>\n",
       "      <td>0.9477</td>\n",
       "      <td>1:100000843:C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>100000989</td>\n",
       "      <td>chr1:100000989:I</td>\n",
       "      <td>A</td>\n",
       "      <td>ATC</td>\n",
       "      <td>-0.0099</td>\n",
       "      <td>0.0343</td>\n",
       "      <td>0.7731</td>\n",
       "      <td>0.9511</td>\n",
       "      <td>1:100000989:A_ATC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>100001138</td>\n",
       "      <td>rs144406489</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.0061</td>\n",
       "      <td>0.0612</td>\n",
       "      <td>0.9204</td>\n",
       "      <td>0.9769</td>\n",
       "      <td>1:100001138:A_G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chromosome   Position        MarkerName Effect_allele Non_Effect_allele  \\\n",
       "0           1  100000012        rs10875231             T                 G   \n",
       "1           1  100000827         rs6678176             T                 C   \n",
       "2           1  100000843        rs78286437             T                 C   \n",
       "3           1  100000989  chr1:100000989:I             A               ATC   \n",
       "4           1  100001138       rs144406489             A                 G   \n",
       "\n",
       "     Beta      SE  Pvalue  Effect_allele_freq    Test_MarkerName  \n",
       "0 -0.0026  0.0168  0.8758              0.2406    1:100000012:G_T  \n",
       "1  0.0008  0.0156  0.9574              0.2926    1:100000827:C_T  \n",
       "2 -0.0136  0.0330  0.6792              0.9477    1:100000843:C_T  \n",
       "3 -0.0099  0.0343  0.7731              0.9511  1:100000989:A_ATC  \n",
       "4 -0.0061  0.0612  0.9204              0.9769    1:100001138:A_G  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_k = pd.read_csv('{WORK_DIR}/Kunkle_etal_2019_IGAP_Summary_statistics.with_allelefreqs.txt', delim_whitespace=True, low_memory=False)\n",
    "df_k.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "74ab1b8d-1f7e-4fac-bd3a-a2f4163062ec",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2805535/3810606804.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_eur_apoe['ancestry'] = 'European'\n",
      "/tmp/ipykernel_2805535/3810606804.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_eur_apoe['Locus'] = 'APOE'\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>ancestry</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10828822</th>\n",
       "      <td>APOE</td>\n",
       "      <td>rs429358</td>\n",
       "      <td>19</td>\n",
       "      <td>45411941</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-1.2017</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.7841</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10828823</th>\n",
       "      <td>APOE</td>\n",
       "      <td>rs7412</td>\n",
       "      <td>19</td>\n",
       "      <td>45412079</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.4673</td>\n",
       "      <td>6.401000e-53</td>\n",
       "      <td>0.0736</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Locus       SNP  Chromosome  Position effect_allele other_allele  \\\n",
       "10828822  APOE  rs429358          19  45411941             T            C   \n",
       "10828823  APOE    rs7412          19  45412079             T            C   \n",
       "\n",
       "            beta       p_value  effect_allele_frequency  ancestry  \n",
       "10828822 -1.2017  0.000000e+00                   0.7841  European  \n",
       "10828823 -0.4673  6.401000e-53                   0.0736  European  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Clean up dataframe\n",
    "df_eur_apoe = df_k[(df_k.MarkerName == 'rs429358') | (df_k.MarkerName == 'rs7412')]\n",
    "df_eur_apoe['ancestry'] = 'European'\n",
    "df_eur_apoe['Locus'] = 'APOE'\n",
    "\n",
    "df_eur_apoe = df_eur_apoe[['Locus', 'MarkerName', 'Chromosome', 'Position', 'Effect_allele', 'Non_Effect_allele', 'Beta', 'Pvalue',\n",
    "                          'Effect_allele_freq', 'ancestry']].rename(columns={\n",
    "    'MarkerName':'SNP','Non_Effect_allele':'other_allele','Effect_allele':'effect_allele','Beta':'beta',\n",
    "    'Pvalue':'p_value','Effect_allele_freq':'effect_allele_frequency'})\n",
    "df_eur_apoe"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b7bd127f-18b4-43e9-8e53-c3869abae5d7",
   "metadata": {},
   "source": [
    "### Black/African American data from Kunkle"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "54bc8c68-09b8-4728-a2d0-5fa1c97dde70",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Pos</th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Effect_allele</th>\n",
       "      <th>Non_Effect_allele</th>\n",
       "      <th>Beta</th>\n",
       "      <th>SE</th>\n",
       "      <th>Pvalue</th>\n",
       "      <th>Effect_allele_Freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>100000012</td>\n",
       "      <td>10:100000012</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1734</td>\n",
       "      <td>0.1515</td>\n",
       "      <td>0.2524</td>\n",
       "      <td>0.0180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10</td>\n",
       "      <td>100000122</td>\n",
       "      <td>10:100000122</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.3347</td>\n",
       "      <td>0.2294</td>\n",
       "      <td>0.1445</td>\n",
       "      <td>0.0083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>100000354</td>\n",
       "      <td>10:100000354</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.6226</td>\n",
       "      <td>1.0170</td>\n",
       "      <td>0.5404</td>\n",
       "      <td>0.9989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>100000588</td>\n",
       "      <td>10:100000588</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.0061</td>\n",
       "      <td>0.1033</td>\n",
       "      <td>0.9531</td>\n",
       "      <td>0.9603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>100000625</td>\n",
       "      <td>10:100000625</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0059</td>\n",
       "      <td>0.0415</td>\n",
       "      <td>0.8873</td>\n",
       "      <td>0.7182</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chr        Pos    MarkerName Effect_allele Non_Effect_allele    Beta  \\\n",
       "0   10  100000012  10:100000012             A                 G  0.1734   \n",
       "1   10  100000122  10:100000122             A                 T  0.3347   \n",
       "2   10  100000354  10:100000354             T                 C  0.6226   \n",
       "3   10  100000588  10:100000588             T                 C -0.0061   \n",
       "4   10  100000625  10:100000625             A                 G  0.0059   \n",
       "\n",
       "       SE  Pvalue  Effect_allele_Freq  \n",
       "0  0.1515  0.2524              0.0180  \n",
       "1  0.2294  0.1445              0.0083  \n",
       "2  1.0170  0.5404              0.9989  \n",
       "3  0.1033  0.9531              0.9603  \n",
       "4  0.0415  0.8873              0.7182  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_afr = pd.read_csv('{WORK_DIR}/Kunkle2020_ADGC_AA_META_Model1_SummaryStats.withAlleleFreqs_REFORMATTED.txt', sep='\\t', header=0, low_memory=False)\n",
    "df_afr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "26bea69c-f76a-4df9-a94f-e36008186313",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Add a column for ancestry\n",
    "df_afr['ancestry'] = 'Black/African American'\n",
    "\n",
    "# Subset and rename columns\n",
    "afr = df_afr[['Chr','Pos','Effect_allele','Non_Effect_allele','Beta','Pvalue','Effect_allele_Freq','ancestry']].rename(columns={\n",
    "    'Chr':'Chromosome','Pos':'Position','Effect_allele':'effect_allele','Non_Effect_allele':'other_allele',\n",
    "    'Beta':'beta','Pvalue':'p_value','Effect_allele_Freq':'effect_allele_frequency'})\n",
    "\n",
    "# Filter p < 0.05\n",
    "#afr = afr[afr['p_value'] < 0.05]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6fe377cf-eee5-4d55-8e13-46a317820bca",
   "metadata": {},
   "source": [
    "### East Asian data from Shigemizu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "961d44a5-3c17-4089-8935-bd9386346e4a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>NMISS</th>\n",
       "      <th>NMISS_A</th>\n",
       "      <th>NMISS_U</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>Info_NCGG</th>\n",
       "      <th>Info_Niigata</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>rs3094315</td>\n",
       "      <td>752566</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>8035</td>\n",
       "      <td>3961</td>\n",
       "      <td>4074</td>\n",
       "      <td>0.1471</td>\n",
       "      <td>0.1523</td>\n",
       "      <td>0.9691</td>\n",
       "      <td>0.04579</td>\n",
       "      <td>0.8860</td>\n",
       "      <td>1.060</td>\n",
       "      <td>-0.6845</td>\n",
       "      <td>0.4937</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>rs3115860</td>\n",
       "      <td>753405</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>7936</td>\n",
       "      <td>3921</td>\n",
       "      <td>4015</td>\n",
       "      <td>0.1489</td>\n",
       "      <td>0.1549</td>\n",
       "      <td>0.9668</td>\n",
       "      <td>0.04571</td>\n",
       "      <td>0.8840</td>\n",
       "      <td>1.057</td>\n",
       "      <td>-0.7385</td>\n",
       "      <td>0.4602</td>\n",
       "      <td>0.966</td>\n",
       "      <td>0.989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>rs2073813</td>\n",
       "      <td>753541</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>7723</td>\n",
       "      <td>3797</td>\n",
       "      <td>3926</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.3835</td>\n",
       "      <td>0.9759</td>\n",
       "      <td>0.03514</td>\n",
       "      <td>0.9110</td>\n",
       "      <td>1.045</td>\n",
       "      <td>-0.6944</td>\n",
       "      <td>0.4874</td>\n",
       "      <td>0.969</td>\n",
       "      <td>0.979</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs3131969</td>\n",
       "      <td>754182</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>7724</td>\n",
       "      <td>3795</td>\n",
       "      <td>3929</td>\n",
       "      <td>0.3740</td>\n",
       "      <td>0.3837</td>\n",
       "      <td>0.9746</td>\n",
       "      <td>0.03514</td>\n",
       "      <td>0.9097</td>\n",
       "      <td>1.044</td>\n",
       "      <td>-0.7322</td>\n",
       "      <td>0.4640</td>\n",
       "      <td>0.969</td>\n",
       "      <td>0.978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs3131968</td>\n",
       "      <td>754192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>7724</td>\n",
       "      <td>3795</td>\n",
       "      <td>3929</td>\n",
       "      <td>0.3740</td>\n",
       "      <td>0.3837</td>\n",
       "      <td>0.9746</td>\n",
       "      <td>0.03514</td>\n",
       "      <td>0.9097</td>\n",
       "      <td>1.044</td>\n",
       "      <td>-0.7322</td>\n",
       "      <td>0.4640</td>\n",
       "      <td>0.969</td>\n",
       "      <td>0.978</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR        SNP      BP A1 A2  NMISS  NMISS_A  NMISS_U   MAF_A   MAF_U  \\\n",
       "0    1  rs3094315  752566  G  A   8035     3961     4074  0.1471  0.1523   \n",
       "1    1  rs3115860  753405  C  A   7936     3921     4015  0.1489  0.1549   \n",
       "2    1  rs2073813  753541  A  G   7723     3797     3926  0.3741  0.3835   \n",
       "3    1  rs3131969  754182  A  G   7724     3795     3929  0.3740  0.3837   \n",
       "4    1  rs3131968  754192  A  G   7724     3795     3929  0.3740  0.3837   \n",
       "\n",
       "       OR       SE     L95    U95    STAT       P  Info_NCGG  Info_Niigata  \n",
       "0  0.9691  0.04579  0.8860  1.060 -0.6845  0.4937      1.000         0.999  \n",
       "1  0.9668  0.04571  0.8840  1.057 -0.7385  0.4602      0.966         0.989  \n",
       "2  0.9759  0.03514  0.9110  1.045 -0.6944  0.4874      0.969         0.979  \n",
       "3  0.9746  0.03514  0.9097  1.044 -0.7322  0.4640      0.969         0.978  \n",
       "4  0.9746  0.03514  0.9097  1.044 -0.7322  0.4640      0.969         0.978  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ea = pd.read_csv('{WORK_DIR}/NCGG_AD_GWAS2.txt', sep='\\t', header=0, low_memory=False)\n",
    "df_ea.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "21d28169-1eb8-422f-9ae1-8e92988c9006",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Add a column for ancestry\n",
    "df_ea['ancestry'] = 'East Asian'\n",
    "\n",
    "# Subset and rename columns\n",
    "ea = df_ea[['CHR','BP','SNP','A1','A2','OR','P','MAF_A','ancestry']].rename(columns={\n",
    "    'CHR':'Chromosome','BP':'Position','A1':'effect_allele','A2':'other_allele',\n",
    "    'P':'p_value','MAF_A':'effect_allele_frequency'})\n",
    "\n",
    "# Filter p < 0.05\n",
    "#ea = ea[ea['p_value'] < 0.05]\n",
    "\n",
    "# beta = ln(odds_ratio)\n",
    "ea['beta'] = np.log(ea['OR'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e5b0aacd-f000-4911-9de1-946458fd1990",
   "metadata": {},
   "source": [
    "### Caribbean/Hispanic data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "bab8cdb9-8e0c-4a91-a7f6-7a7a5b7c42cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MARKERNAME</th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>EA</th>\n",
       "      <th>NEA</th>\n",
       "      <th>EAF</th>\n",
       "      <th>OR</th>\n",
       "      <th>OR_95U</th>\n",
       "      <th>OR_95L</th>\n",
       "      <th>N</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:48824</td>\n",
       "      <td>1</td>\n",
       "      <td>48824</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.017666</td>\n",
       "      <td>0.795203</td>\n",
       "      <td>1.407862</td>\n",
       "      <td>0.449155</td>\n",
       "      <td>2240</td>\n",
       "      <td>0.431702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:54490</td>\n",
       "      <td>1</td>\n",
       "      <td>54490</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.043157</td>\n",
       "      <td>0.853678</td>\n",
       "      <td>1.255188</td>\n",
       "      <td>0.580603</td>\n",
       "      <td>2240</td>\n",
       "      <td>0.421182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:60351</td>\n",
       "      <td>1</td>\n",
       "      <td>60351</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.024821</td>\n",
       "      <td>0.864445</td>\n",
       "      <td>1.395185</td>\n",
       "      <td>0.535603</td>\n",
       "      <td>2240</td>\n",
       "      <td>0.550887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:64931</td>\n",
       "      <td>1</td>\n",
       "      <td>64931</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.024598</td>\n",
       "      <td>0.821548</td>\n",
       "      <td>1.334293</td>\n",
       "      <td>0.505842</td>\n",
       "      <td>2240</td>\n",
       "      <td>0.426951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:66861</td>\n",
       "      <td>1</td>\n",
       "      <td>66861</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.044946</td>\n",
       "      <td>0.868805</td>\n",
       "      <td>1.244424</td>\n",
       "      <td>0.606564</td>\n",
       "      <td>2240</td>\n",
       "      <td>0.442988</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  MARKERNAME  CHROMOSOME  POSITION EA NEA       EAF        OR    OR_95U  \\\n",
       "0    1:48824           1     48824  C   T  0.017666  0.795203  1.407862   \n",
       "1    1:54490           1     54490  A   G  0.043157  0.853678  1.255188   \n",
       "2    1:60351           1     60351  G   A  0.024821  0.864445  1.395185   \n",
       "3    1:64931           1     64931  A   G  0.024598  0.821548  1.334293   \n",
       "4    1:66861           1     66861  T   C  0.044946  0.868805  1.244424   \n",
       "\n",
       "     OR_95L     N         P  \n",
       "0  0.449155  2240  0.431702  \n",
       "1  0.580603  2240  0.421182  \n",
       "2  0.535603  2240  0.550887  \n",
       "3  0.505842  2240  0.426951  \n",
       "4  0.606564  2240  0.442988  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ch = pd.read_csv('{WORK_DIR}/CarHisp_for_MRMEGA.no_multiAllelics_indels.MAF_0.01.txt', sep='\\t', header=0, low_memory=False)\n",
    "df_ch.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "ddcf7a89-65c3-4950-9ff1-ffa81f41bb5e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Add a column for ancestry\n",
    "df_ch['ancestry'] = 'Latino'\n",
    "\n",
    "# beta = ln(odds_ratio)\n",
    "df_ch['beta'] = np.log(df_ch['OR'])\n",
    "\n",
    "# Subset and rename columns\n",
    "ch = df_ch[['CHROMOSOME','POSITION','EA','NEA','beta','P','EAF','ancestry']].rename(columns={\n",
    "    'CHROMOSOME':'Chromosome','POSITION':'Position','EA':'effect_allele','NEA':'other_allele',\n",
    "    'P':'p_value','EAF':'effect_allele_frequency'})\n",
    "\n",
    "# Filter p < 0.05\n",
    "#ch = ch[ch['p_value'] < 0.05]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "07596161-e147-465d-8467-d684de83aed0",
   "metadata": {},
   "source": [
    "### Select top hits from each ancestry"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "c833f79a-e41f-474f-b036-acbf71e75589",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Effect allele, random effects</th>\n",
       "      <th>Reference allele, random effects</th>\n",
       "      <th>P, random effects</th>\n",
       "      <th>beta, random effects</th>\n",
       "      <th>SE, random effects</th>\n",
       "      <th>I2</th>\n",
       "      <th>...</th>\n",
       "      <th>SE, Caribbean Hispanic</th>\n",
       "      <th>P, Bellenguez et al. 2022 and FinnGen R6 (EXMORE)</th>\n",
       "      <th>Beta, Bellenguez et al. 2022 and FinnGen R6 (EXMORE)</th>\n",
       "      <th>SE, Bellenguez et al. 2022 and FinnGen R6 (EXMORE)</th>\n",
       "      <th>P, Kunkle et al. 2021</th>\n",
       "      <th>Beta, Kunkle et al. 2021</th>\n",
       "      <th>SE, Kunkle et al. 2021</th>\n",
       "      <th>P, Shigemizu et al. 2021</th>\n",
       "      <th>Beta, Shigemizu et al. 2021</th>\n",
       "      <th>SE, Shigemizu et al. 2021</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCA7</td>\n",
       "      <td>rs12151021</td>\n",
       "      <td>19</td>\n",
       "      <td>1050874</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>3.900000e-33</td>\n",
       "      <td>0.1028</td>\n",
       "      <td>0.008572</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.079176</td>\n",
       "      <td>3.990000e-32</td>\n",
       "      <td>0.1043</td>\n",
       "      <td>0.0088</td>\n",
       "      <td>0.1748</td>\n",
       "      <td>0.0536</td>\n",
       "      <td>0.0395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACE</td>\n",
       "      <td>rs4311</td>\n",
       "      <td>17</td>\n",
       "      <td>61560763</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2.190000e-16</td>\n",
       "      <td>-0.0620</td>\n",
       "      <td>0.007551</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.079062</td>\n",
       "      <td>3.110000e-16</td>\n",
       "      <td>-0.0630</td>\n",
       "      <td>0.0077</td>\n",
       "      <td>0.3329</td>\n",
       "      <td>-0.0410</td>\n",
       "      <td>0.0424</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ADAM10</td>\n",
       "      <td>rs653765</td>\n",
       "      <td>15</td>\n",
       "      <td>59042012</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2.570000e-12</td>\n",
       "      <td>0.0570</td>\n",
       "      <td>0.008144</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.050000e-12</td>\n",
       "      <td>0.0592</td>\n",
       "      <td>0.0085</td>\n",
       "      <td>0.7464</td>\n",
       "      <td>0.0133</td>\n",
       "      <td>0.0411</td>\n",
       "      <td>0.6829</td>\n",
       "      <td>0.020783</td>\n",
       "      <td>0.050324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ADAMTS4</td>\n",
       "      <td>rs11265557</td>\n",
       "      <td>1</td>\n",
       "      <td>161106354</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>2.450000e-09</td>\n",
       "      <td>-0.0480</td>\n",
       "      <td>0.008047</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.700000e-09</td>\n",
       "      <td>-0.0482</td>\n",
       "      <td>0.0084</td>\n",
       "      <td>0.9681</td>\n",
       "      <td>0.0026</td>\n",
       "      <td>0.0662</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ANK3</td>\n",
       "      <td>rs7068231</td>\n",
       "      <td>10</td>\n",
       "      <td>61784928</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>6.140000e-10</td>\n",
       "      <td>-0.0484</td>\n",
       "      <td>0.007823</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.078698</td>\n",
       "      <td>3.470000e-10</td>\n",
       "      <td>-0.0501</td>\n",
       "      <td>0.0080</td>\n",
       "      <td>0.7864</td>\n",
       "      <td>0.0120</td>\n",
       "      <td>0.0441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Locus         SNP  Chromosome   Position Effect allele, random effects  \\\n",
       "0    ABCA7  rs12151021          19    1050874                             A   \n",
       "1      ACE      rs4311          17   61560763                             T   \n",
       "2   ADAM10    rs653765          15   59042012                             T   \n",
       "3  ADAMTS4  rs11265557           1  161106354                             T   \n",
       "4     ANK3   rs7068231          10   61784928                             T   \n",
       "\n",
       "  Reference allele, random effects  P, random effects  beta, random effects  \\\n",
       "0                                G       3.900000e-33                0.1028   \n",
       "1                                C       2.190000e-16               -0.0620   \n",
       "2                                C       2.570000e-12                0.0570   \n",
       "3                                G       2.450000e-09               -0.0480   \n",
       "4                                G       6.140000e-10               -0.0484   \n",
       "\n",
       "   SE, random effects   I2  ...  SE, Caribbean Hispanic  \\\n",
       "0            0.008572  0.0  ...                0.079176   \n",
       "1            0.007551  0.0  ...                0.079062   \n",
       "2            0.008144  0.0  ...                     NaN   \n",
       "3            0.008047  0.0  ...                     NaN   \n",
       "4            0.007823  0.0  ...                0.078698   \n",
       "\n",
       "   P, Bellenguez et al. 2022 and FinnGen R6 (EXMORE)  \\\n",
       "0                                       3.990000e-32   \n",
       "1                                       3.110000e-16   \n",
       "2                                       3.050000e-12   \n",
       "3                                       9.700000e-09   \n",
       "4                                       3.470000e-10   \n",
       "\n",
       "   Beta, Bellenguez et al. 2022 and FinnGen R6 (EXMORE)  \\\n",
       "0                                             0.1043      \n",
       "1                                            -0.0630      \n",
       "2                                             0.0592      \n",
       "3                                            -0.0482      \n",
       "4                                            -0.0501      \n",
       "\n",
       "   SE, Bellenguez et al. 2022 and FinnGen R6 (EXMORE)  P, Kunkle et al. 2021  \\\n",
       "0                                             0.0088                  0.1748   \n",
       "1                                             0.0077                  0.3329   \n",
       "2                                             0.0085                  0.7464   \n",
       "3                                             0.0084                  0.9681   \n",
       "4                                             0.0080                  0.7864   \n",
       "\n",
       "   Beta, Kunkle et al. 2021  SE, Kunkle et al. 2021  P, Shigemizu et al. 2021  \\\n",
       "0                    0.0536                  0.0395                       NaN   \n",
       "1                   -0.0410                  0.0424                       NaN   \n",
       "2                    0.0133                  0.0411                    0.6829   \n",
       "3                    0.0026                  0.0662                       NaN   \n",
       "4                    0.0120                  0.0441                       NaN   \n",
       "\n",
       "   Beta, Shigemizu et al. 2021  SE, Shigemizu et al. 2021  \n",
       "0                          NaN                        NaN  \n",
       "1                          NaN                        NaN  \n",
       "2                     0.020783                   0.050324  \n",
       "3                          NaN                        NaN  \n",
       "4                          NaN                        NaN  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# top hits from multi-ancestry GWAS meta-analysis\n",
    "top_hits = pd.read_excel('{WORK_DIR}/AD_stats.xlsx', sheet_name='S2b').rename(columns={'SNP, random effects':'SNP',\n",
    "                                                                                                   'Chromosome, random effects':'Chromosome',\n",
    "                                                                                                   'Position, random effects':'Position'})\n",
    "top_hits.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f75d72e8-9bc6-494e-bb34-d52e0ba12ba5",
   "metadata": {},
   "source": [
    "#### European top hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "a4cc9ead-9c9b-465e-8a7d-b15813e34527",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Merge with AD top hits\n",
    "eur_top = pd.merge(top_hits, merge_stats_eur_subset, on='SNP', how='inner')\n",
    "eur_top = eur_top[['Locus_y','SNP','Chromosome_y','Position_y','effect_allele','other_allele','beta',\n",
    "                   'p_value','effect_allele_frequency','ancestry']].rename(columns={'Locus_y':'Locus',\n",
    "                                                                                    'Chromosome_y':'Chromosome',\n",
    "                                                                                    'Position_y':'Position'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "90348044-86d3-4989-8523-bfb2a8680e1e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>ancestry</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCA7</td>\n",
       "      <td>rs12151021</td>\n",
       "      <td>19</td>\n",
       "      <td>1050875</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1226</td>\n",
       "      <td>7.692000e-30</td>\n",
       "      <td>0.3391</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACE</td>\n",
       "      <td>rs4311</td>\n",
       "      <td>17</td>\n",
       "      <td>63483402</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.0620</td>\n",
       "      <td>8.077000e-10</td>\n",
       "      <td>0.4911</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ADAM10</td>\n",
       "      <td>rs653765</td>\n",
       "      <td>15</td>\n",
       "      <td>58749813</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0657</td>\n",
       "      <td>6.408000e-09</td>\n",
       "      <td>0.7223</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>APH1B</td>\n",
       "      <td>rs75763893</td>\n",
       "      <td>15</td>\n",
       "      <td>63279621</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.1171</td>\n",
       "      <td>4.584000e-15</td>\n",
       "      <td>0.1462</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBLC</td>\n",
       "      <td>rs80168591</td>\n",
       "      <td>19</td>\n",
       "      <td>44781370</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.8483</td>\n",
       "      <td>4.272000e-65</td>\n",
       "      <td>0.0145</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BIN1;CYP27C1</td>\n",
       "      <td>rs34779859</td>\n",
       "      <td>2</td>\n",
       "      <td>127135192</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1259</td>\n",
       "      <td>9.070000e-30</td>\n",
       "      <td>0.3013</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CASS4</td>\n",
       "      <td>rs113221226</td>\n",
       "      <td>20</td>\n",
       "      <td>56449045</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.1264</td>\n",
       "      <td>1.890000e-11</td>\n",
       "      <td>0.0807</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CD2AP</td>\n",
       "      <td>rs7738044</td>\n",
       "      <td>6</td>\n",
       "      <td>47501537</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.0680</td>\n",
       "      <td>2.102000e-09</td>\n",
       "      <td>0.7274</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CLU</td>\n",
       "      <td>rs1532278</td>\n",
       "      <td>8</td>\n",
       "      <td>27608798</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.1092</td>\n",
       "      <td>1.048000e-25</td>\n",
       "      <td>0.3768</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>LINC02059;MIR4280</td>\n",
       "      <td>rs58233521</td>\n",
       "      <td>5</td>\n",
       "      <td>87079937</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0737</td>\n",
       "      <td>2.660000e-08</td>\n",
       "      <td>0.1836</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CR1</td>\n",
       "      <td>rs10779336</td>\n",
       "      <td>1</td>\n",
       "      <td>207625371</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1373</td>\n",
       "      <td>1.753000e-27</td>\n",
       "      <td>0.1969</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CTSB</td>\n",
       "      <td>rs1065712</td>\n",
       "      <td>8</td>\n",
       "      <td>11844613</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1284</td>\n",
       "      <td>1.158000e-08</td>\n",
       "      <td>0.0557</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>DOC2A</td>\n",
       "      <td>rs1140239</td>\n",
       "      <td>16</td>\n",
       "      <td>30010081</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.0599</td>\n",
       "      <td>3.269000e-08</td>\n",
       "      <td>0.3682</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>LOC107984208;ECHDC3</td>\n",
       "      <td>rs7912495</td>\n",
       "      <td>10</td>\n",
       "      <td>11676714</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.0605</td>\n",
       "      <td>2.835000e-09</td>\n",
       "      <td>0.5326</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>EPHA1-AS1</td>\n",
       "      <td>rs11771145</td>\n",
       "      <td>7</td>\n",
       "      <td>143413669</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.0748</td>\n",
       "      <td>1.810000e-12</td>\n",
       "      <td>0.3459</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HLA-DRB1;HLA-DQA1</td>\n",
       "      <td>rs35472547</td>\n",
       "      <td>6</td>\n",
       "      <td>32592593</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.0869</td>\n",
       "      <td>4.923000e-09</td>\n",
       "      <td>0.1479</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>INPP5D</td>\n",
       "      <td>rs36127492</td>\n",
       "      <td>2</td>\n",
       "      <td>233158323</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0562</td>\n",
       "      <td>3.682000e-08</td>\n",
       "      <td>0.4915</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>KANSL1</td>\n",
       "      <td>rs7225002</td>\n",
       "      <td>17</td>\n",
       "      <td>46111701</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0683</td>\n",
       "      <td>1.111000e-10</td>\n",
       "      <td>0.6016</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>LILRB5;LILRB2</td>\n",
       "      <td>rs12984029</td>\n",
       "      <td>19</td>\n",
       "      <td>54265512</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0688</td>\n",
       "      <td>1.482000e-08</td>\n",
       "      <td>0.2434</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MS4A6A</td>\n",
       "      <td>rs2278867</td>\n",
       "      <td>11</td>\n",
       "      <td>60175636</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0875</td>\n",
       "      <td>2.783000e-17</td>\n",
       "      <td>0.6034</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PICALM;EED</td>\n",
       "      <td>rs9787874</td>\n",
       "      <td>11</td>\n",
       "      <td>86149263</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1047</td>\n",
       "      <td>3.381000e-25</td>\n",
       "      <td>0.5291</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>RASGEF1C</td>\n",
       "      <td>rs113706587</td>\n",
       "      <td>5</td>\n",
       "      <td>180201150</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1016</td>\n",
       "      <td>4.840000e-10</td>\n",
       "      <td>0.1153</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>LOC100130950</td>\n",
       "      <td>rs57402520</td>\n",
       "      <td>17</td>\n",
       "      <td>5236512</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0878</td>\n",
       "      <td>1.839000e-08</td>\n",
       "      <td>0.1248</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SHARPIN</td>\n",
       "      <td>rs34173062</td>\n",
       "      <td>8</td>\n",
       "      <td>144103704</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1164</td>\n",
       "      <td>6.267000e-09</td>\n",
       "      <td>0.0869</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SLC24A4</td>\n",
       "      <td>rs36026988</td>\n",
       "      <td>14</td>\n",
       "      <td>92472038</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0802</td>\n",
       "      <td>1.567000e-10</td>\n",
       "      <td>0.7905</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PSMC3;RAPSN</td>\n",
       "      <td>rs4434960</td>\n",
       "      <td>11</td>\n",
       "      <td>47433000</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.0688</td>\n",
       "      <td>1.774000e-11</td>\n",
       "      <td>0.5930</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>TREML5P;TREM1</td>\n",
       "      <td>rs4714447</td>\n",
       "      <td>6</td>\n",
       "      <td>41272084</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0604</td>\n",
       "      <td>1.469000e-08</td>\n",
       "      <td>0.3444</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>TSPAN14</td>\n",
       "      <td>rs7922621</td>\n",
       "      <td>10</td>\n",
       "      <td>80491788</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0802</td>\n",
       "      <td>3.990000e-10</td>\n",
       "      <td>0.8017</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PMS2P1</td>\n",
       "      <td>rs7384878</td>\n",
       "      <td>7</td>\n",
       "      <td>100334426</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0695</td>\n",
       "      <td>3.351000e-10</td>\n",
       "      <td>0.6925</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10828822</th>\n",
       "      <td>APOE</td>\n",
       "      <td>rs429358</td>\n",
       "      <td>19</td>\n",
       "      <td>45411941</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-1.2017</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.7841</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10828823</th>\n",
       "      <td>APOE</td>\n",
       "      <td>rs7412</td>\n",
       "      <td>19</td>\n",
       "      <td>45412079</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.4673</td>\n",
       "      <td>6.401000e-53</td>\n",
       "      <td>0.0736</td>\n",
       "      <td>European</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Locus          SNP Chromosome   Position  \\\n",
       "0                       ABCA7   rs12151021         19    1050875   \n",
       "1                         ACE       rs4311         17   63483402   \n",
       "2                      ADAM10     rs653765         15   58749813   \n",
       "3                       APH1B   rs75763893         15   63279621   \n",
       "4                        CBLC   rs80168591         19   44781370   \n",
       "5                BIN1;CYP27C1   rs34779859          2  127135192   \n",
       "6                       CASS4  rs113221226         20   56449045   \n",
       "7                       CD2AP    rs7738044          6   47501537   \n",
       "8                         CLU    rs1532278          8   27608798   \n",
       "9           LINC02059;MIR4280   rs58233521          5   87079937   \n",
       "10                        CR1   rs10779336          1  207625371   \n",
       "11                       CTSB    rs1065712          8   11844613   \n",
       "12                      DOC2A    rs1140239         16   30010081   \n",
       "13        LOC107984208;ECHDC3    rs7912495         10   11676714   \n",
       "14                  EPHA1-AS1   rs11771145          7  143413669   \n",
       "15          HLA-DRB1;HLA-DQA1   rs35472547          6   32592593   \n",
       "16                     INPP5D   rs36127492          2  233158323   \n",
       "17                     KANSL1    rs7225002         17   46111701   \n",
       "18              LILRB5;LILRB2   rs12984029         19   54265512   \n",
       "19                     MS4A6A    rs2278867         11   60175636   \n",
       "20                 PICALM;EED    rs9787874         11   86149263   \n",
       "21                   RASGEF1C  rs113706587          5  180201150   \n",
       "22               LOC100130950   rs57402520         17    5236512   \n",
       "23                    SHARPIN   rs34173062          8  144103704   \n",
       "24                    SLC24A4   rs36026988         14   92472038   \n",
       "25                PSMC3;RAPSN    rs4434960         11   47433000   \n",
       "26              TREML5P;TREM1    rs4714447          6   41272084   \n",
       "27                    TSPAN14    rs7922621         10   80491788   \n",
       "28                     PMS2P1    rs7384878          7  100334426   \n",
       "10828822                 APOE     rs429358         19   45411941   \n",
       "10828823                 APOE       rs7412         19   45412079   \n",
       "\n",
       "         effect_allele other_allele    beta       p_value  \\\n",
       "0                    A            G  0.1226  7.692000e-30   \n",
       "1                    T            C -0.0620  8.077000e-10   \n",
       "2                    T            C  0.0657  6.408000e-09   \n",
       "3                    T            C  0.1171  4.584000e-15   \n",
       "4                    A            G  0.8483  4.272000e-65   \n",
       "5                    T            G  0.1259  9.070000e-30   \n",
       "6                    A            G -0.1264  1.890000e-11   \n",
       "7                    A            G -0.0680  2.102000e-09   \n",
       "8                    T            C -0.1092  1.048000e-25   \n",
       "9                    T            C  0.0737  2.660000e-08   \n",
       "10                   T            G  0.1373  1.753000e-27   \n",
       "11                   C            G  0.1284  1.158000e-08   \n",
       "12                   T            C -0.0599  3.269000e-08   \n",
       "13                   A            G -0.0605  2.835000e-09   \n",
       "14                   A            G -0.0748  1.810000e-12   \n",
       "15                   T            G -0.0869  4.923000e-09   \n",
       "16                   C            G  0.0562  3.682000e-08   \n",
       "17                   A            G  0.0683  1.111000e-10   \n",
       "18                   A            G  0.0688  1.482000e-08   \n",
       "19                   A            T  0.0875  2.783000e-17   \n",
       "20                   A            G  0.1047  3.381000e-25   \n",
       "21                   A            G  0.1016  4.840000e-10   \n",
       "22                   A            G  0.0878  1.839000e-08   \n",
       "23                   A            G  0.1164  6.267000e-09   \n",
       "24                   T            C  0.0802  1.567000e-10   \n",
       "25                   A            T -0.0688  1.774000e-11   \n",
       "26                   T            C  0.0604  1.469000e-08   \n",
       "27                   A            C  0.0802  3.990000e-10   \n",
       "28                   T            C  0.0695  3.351000e-10   \n",
       "10828822             T            C -1.2017  0.000000e+00   \n",
       "10828823             T            C -0.4673  6.401000e-53   \n",
       "\n",
       "          effect_allele_frequency  ancestry  \n",
       "0                          0.3391  European  \n",
       "1                          0.4911  European  \n",
       "2                          0.7223  European  \n",
       "3                          0.1462  European  \n",
       "4                          0.0145  European  \n",
       "5                          0.3013  European  \n",
       "6                          0.0807  European  \n",
       "7                          0.7274  European  \n",
       "8                          0.3768  European  \n",
       "9                          0.1836  European  \n",
       "10                         0.1969  European  \n",
       "11                         0.0557  European  \n",
       "12                         0.3682  European  \n",
       "13                         0.5326  European  \n",
       "14                         0.3459  European  \n",
       "15                         0.1479  European  \n",
       "16                         0.4915  European  \n",
       "17                         0.6016  European  \n",
       "18                         0.2434  European  \n",
       "19                         0.6034  European  \n",
       "20                         0.5291  European  \n",
       "21                         0.1153  European  \n",
       "22                         0.1248  European  \n",
       "23                         0.0869  European  \n",
       "24                         0.7905  European  \n",
       "25                         0.5930  European  \n",
       "26                         0.3444  European  \n",
       "27                         0.8017  European  \n",
       "28                         0.6925  European  \n",
       "10828822                   0.7841  European  \n",
       "10828823                   0.0736  European  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Add in APOE\n",
    "eur_final = pd.concat([eur_top, df_eur_apoe])\n",
    "\n",
    "# only keep rs429358 <> rs1081105, rs7412 <> rs1065853\n",
    "condition = eur_final['Locus'] == 'APOE'\n",
    "snps_to_keep = ['rs429358','rs7412']\n",
    "filtered_subset = eur_final.loc[condition & eur_final['SNP'].isin(snps_to_keep)]\n",
    "eur_final = pd.concat([eur_final[~condition], filtered_subset])\n",
    "eur_final"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2183c8b2-eb4d-4c19-a01a-c299b4024ae9",
   "metadata": {},
   "source": [
    "#### Black/African American top hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "6aae84be-228e-40f9-ac1a-ae94d667d4e8",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Merge with AD top hits\n",
    "afr_top = pd.merge(top_hits, afr, on=['Chromosome','Position'], how='inner')\n",
    "afr_top = afr_top[['Locus','SNP','Chromosome','Position','effect_allele','other_allele',\n",
    "                   'beta','p_value','effect_allele_frequency','ancestry']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "28af8ad5-acca-4a2a-825c-3c3d08e4fca4",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Add in APOE\n",
    "afr_apoe1 = afr[(afr['Chromosome'] == 19) & (afr['Position'] == 45411941)]\n",
    "afr_apoe2 = afr[(afr['Chromosome'] == 19) & (afr['Position'] == 45412079)]\n",
    "afr_apoe = pd.concat([afr_apoe1, afr_apoe2])\n",
    "afr_apoe['Locus'] = 'APOE'\n",
    "afr_apoe['SNP'] = afr_apoe['Position'].map({45411941: 'rs429358', 45412079: 'rs7412'})\n",
    "\n",
    "afr_final = pd.concat([afr_top, afr_apoe])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c43d57f-e720-4655-9703-87be23b87689",
   "metadata": {},
   "source": [
    "#### East Asian top hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "3c7d53ed-b204-48a8-80b3-0678791be082",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Merge with AD top hits\n",
    "ea_top = pd.merge(top_hits, ea, on='SNP', how='inner')\n",
    "ea_top = ea_top[['Locus','SNP','Chromosome_y','Position_y','effect_allele','other_allele',\n",
    "                   'beta','p_value','effect_allele_frequency','ancestry']].rename(columns={'Chromosome_y':'Chromosome',\n",
    "                                                                                           'Position_y':'Position'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "cfb7b8c1-48e0-43d4-be30-f0d50197e52a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Add in APOE\n",
    "ea_apoe = ea[(ea['SNP'] == 'rs429358') | (ea['SNP'] == 'rs7412')].drop(columns='OR')\n",
    "ea_apoe['Locus'] = 'APOE'\n",
    "ea_final = pd.concat([ea_top, ea_apoe])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc72179e-0140-480a-8b9c-7a313207a4a9",
   "metadata": {},
   "source": [
    "#### Latino top hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "87523ceb-065a-40fa-9d6d-133feefd45f8",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Merge with top hits\n",
    "ch_top = pd.merge(top_hits, ch, on=['Chromosome','Position'], how='inner')\n",
    "ch_top = ch_top[['Locus','SNP','Chromosome','Position','effect_allele','other_allele',\n",
    "                   'beta','p_value','effect_allele_frequency','ancestry']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "7f18653f-f9cf-4fe1-914c-6d65f8772ce0",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Add in APOE\n",
    "ch_apoe1 = ch[(ch['Chromosome'] == 19) & (ch['Position'] == 45411941)]\n",
    "ch_apoe2 = ch[(ch['Chromosome'] == 19) & (ch['Position'] == 45412079)]\n",
    "ch_apoe = pd.concat([ch_apoe1, ch_apoe2])\n",
    "ch_apoe['Locus'] = 'APOE'\n",
    "ch_apoe['SNP'] = ch_apoe['Position'].map({45411941: 'rs429358', 45412079: 'rs7412'})\n",
    "\n",
    "ch_final = pd.concat([ch_top, ch_apoe])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c7c23161-e2d5-467f-a79a-d68005ab6fda",
   "metadata": {},
   "source": [
    "### Generate dataset for analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "3e553776-34cb-4a9d-b9fd-996f904f9eec",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Concat all dataframes\n",
    "df_calcs = pd.concat([eur_final, afr_final, ea_final, ch_final])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "370bac27-cd4c-4a2f-b7b8-60b72c5efb04",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Current date in YYYY-MM-DD \n",
    "today = datetime.now().strftime('%Y-%m-%d')\n",
    "\n",
    "# Save path\n",
    "path = '{WORK_DIR}/AD/processed/'\n",
    "\n",
    "# Filename and path\n",
    "file = f'{path}_AD_PAR_data_for_analysis_pval.csv'\n",
    "full_path = os.path.join(path, file)\n",
    "\n",
    "# Export the dataframe to CSV\n",
    "df_calcs.to_csv(full_path, index=False, header=True)\n",
    "\n",
    "print(f'Data exported as {file}')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python/3.12",
   "language": "python",
   "name": "py3.12"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
