{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "a877c3d9-42dd-4cfc-b678-eb2345598d08",
   "metadata": {},
   "source": [
    "# Review of fine-mapping\n",
    "* Project: Cross-ancestry PAR\n",
    "* Version: python/3.10\n",
    "* Status: Complete\n",
    "* Last Updated: 13-FEB-2025"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "59c22439-36c7-4b2a-abeb-beedda14bbf0",
   "metadata": {},
   "source": [
    "## Notebook overview\n",
    "* Combine results of fine-mapping analysis\n",
    "* Add summary statistics needed for PAR calculations\n",
    "* Calculate PAR for variants with high posterior probability and save results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "78b2bf22-6a47-4e03-bcf5-c0cda10a3735",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/jonesln/.local/lib/python3.10/site-packages/pandas/core/computation/expressions.py:21: UserWarning: Pandas requires version '2.8.4' or newer of 'numexpr' (version '2.7.3' currently installed).\n",
      "  from pandas.core.computation.check import NUMEXPR_INSTALLED\n"
     ]
    }
   ],
   "source": [
    "# Import packages\n",
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import sys\n",
    "import openpyxl\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import glob"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1bb8ee94-18e4-4fcd-884f-750f34f3fed3",
   "metadata": {},
   "source": [
    "## Extract all fine map results and combine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "26f1c70d-c53e-4a95-9d19-404a657ed67f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define path and filename pattern\n",
    "dir = '{WORK_DIR}/PAR/'\n",
    "file_pattern = os.path.join(dir, '*_results_fine_map_*.csv')\n",
    "\n",
    "# List to store dataframes\n",
    "filtered_dfs = []\n",
    "\n",
    "files = glob.glob(file_pattern)\n",
    "if not files:\n",
    "    print(\"No files matched the pattern. Review your directory/file pattern.\")\n",
    "else:\n",
    "    print(f\"Found files: {files}\")\n",
    "\n",
    "# Loop over results files\n",
    "for file in glob.glob(file_pattern):\n",
    "    print(f\"Processing file: {file}\")\n",
    "    df = pd.read_csv(file)\n",
    "\n",
    "    # We only want SNPs with SNP.PP > 80%\n",
    "    filtered_df = df[df['SNP.PP'] > 0.8]\n",
    "\n",
    "    # Extract  gene, ancestry from filename\n",
    "    base_name = os.path.basename(file)\n",
    "    gene_name = base_name.split('_')[0]\n",
    "    ancestry = base_name.split('_')[-1].replace('.csv','')\n",
    "\n",
    "    # Columns for gene, ancestry\n",
    "    filtered_df['Gene'] = gene_name\n",
    "    filtered_df['Ancestry'] = ancestry\n",
    "\n",
    "    filtered_dfs.append(filtered_df)\n",
    "\n",
    "# Combine all filtered dataframes into one large dataframe for processing\n",
    "if filtered_dfs:\n",
    "    final_df = pd.concat(filtered_dfs, ignore_index=True)\n",
    "    # Save the combined dataframe to a new csv\n",
    "    final_df.to_csv('finemap_results_with_highpp.csv', index=False)\n",
    "    # Display the final dataframe\n",
    "    final_df.head(25)\n",
    "else:\n",
    "    print(\"No filtered dataframes to combine.\")\n",
    "\n",
    "# Sort by ancestry and posterior probability\n",
    "final_df = final_df.sort_values(by=['Ancestry', 'SNP.PP'], ascending=[True, False])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e9f7799-1f4f-4bdf-a712-93ce1daa190f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add a disease column\n",
    "disease_mapping = {\n",
    "    'Bellenguez': 'AD',\n",
    "    'Kunkle': 'AD',\n",
    "    'Shigemizu': 'AD',\n",
    "    'Lake': 'AD',\n",
    "    'Nalls': 'PD',\n",
    "    'afr': 'PD',\n",
    "    'Foo': 'PD',\n",
    "    'Loesch': 'PD'\n",
    "}\n",
    "\n",
    "# Match data source to disease\n",
    "final_df['Disease'] = final_df['Ancestry'].map(disease_mapping)\n",
    "\n",
    "# Map data source to ancestry\n",
    "ancestry_mapping = {\n",
    "    'Bellenguez': 'European',\n",
    "    'Nalls': 'European',\n",
    "    'Kunkle': 'African/African Admixed',\n",
    "    'afr': 'African/African Admixed',\n",
    "    'Foo': 'East Asian',\n",
    "    'Shigemizu': 'East Asian',\n",
    "    'Lake': 'Latino',\n",
    "    'Loesch': 'Latino'\n",
    "}\n",
    "\n",
    "# Replace ancestry values using the mapping\n",
    "final_df['Ancestry'] = final_df['Ancestry'].replace(ancestry_mapping)\n",
    "final_df = final_df.sort_values(by=['Ancestry', 'Disease'])\n",
    "\n",
    "# Rename column\n",
    "final_df = final_df.rename(columns={'SNP':'MarkerName'})\n",
    "\n",
    "# Export results to csv\n",
    "final_df.to_csv('{WORK_DIR}/PAR/results/finemap_results_with_80pp.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "125e88f5-e0cd-4cf2-9afe-4b0c7ea7e0e4",
   "metadata": {},
   "source": [
    "### Read in summary statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d552ef49-b44e-496d-84e1-72c373e5bde7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# AD files\n",
    "ad_eur = pd.read_csv('{WORK_DIR}/AD/summary_stats/Bellenguez_2022/noProxy/EADB-minus-UKB_Nov2022.tsv.gz', sep='\\t', header=0, low_memory=False)\n",
    "ad_afr = pd.read_csv('{WORK_DIR}/AD/summary_stats/diverse_ancestry/NG00100_Kunkle2021/Kunkle2020_ADGC_AA_META_Model1_SummaryStats.withAlleleFreqs_REFORMATTED.txt',sep='\\t', header=0, low_memory=False)\n",
    "ad_lat = pd.read_csv('{WORK_DIR}/trans_ethnic_AD/format_stats/CarHisp_for_MRMEGA.no_multiAllelics_indels.MAF_0.01.txt', sep='\\t', header=0, low_memory=False)\n",
    "ad_eas = pd.read_csv('{WORK_DIR}/AD/summary_stats/diverse_ancestry/Shigemizu_2021/NCGG_AD_GWAS2.txt', sep='\\t', header=0, low_memory=False)\n",
    "\n",
    "# PD files\n",
    "pd_eur = pd.read_csv('{WORK_DIR}/PD/summary_stats/META5_all_with_rsid_hg38.txt', sep='\\t', header=0).rename(columns={\n",
    "    'ID':'SNP'})\n",
    "pd_afr = pd.read_csv('{WORK_DIR}/PD/summary_stats/AFR_AAC_metaGWAS_MAF0.05_hg38_noindels_full_with23andMe.tab', sep='\\t', header=0, low_memory=False).rename(columns={\n",
    "    'rsid':'SNP'})\n",
    "pd_lat = pd.read_csv('{WORK_DIR}/Sumstat/per-cohort/AMR.final.txt', sep='\\t', header=0, low_memory=False)\n",
    "pd_eas = pd.read_csv('{WORK_DIR}/Sumstat/per-cohort/EAS.final.txt', sep='\\t', header=0, low_memory=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77909fb8-05a8-4266-b91d-fd11676689f3",
   "metadata": {},
   "source": [
    "### Prepare AD summary statistics for calculations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "3f296db3-0f0c-453b-9b3c-9fc4d1e66b69",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr11:12541586:A:G</td>\n",
       "      <td>11</td>\n",
       "      <td>12541586</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.3748</td>\n",
       "      <td>0.0007</td>\n",
       "      <td>0.94860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr8:102803998:G:A</td>\n",
       "      <td>8</td>\n",
       "      <td>102803998</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0026</td>\n",
       "      <td>-0.1579</td>\n",
       "      <td>0.19910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr13:55101557:T:C</td>\n",
       "      <td>13</td>\n",
       "      <td>55101557</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.9856</td>\n",
       "      <td>-0.0252</td>\n",
       "      <td>0.56260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr7:17885109:C:G</td>\n",
       "      <td>7</td>\n",
       "      <td>17885109</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.9989</td>\n",
       "      <td>0.5638</td>\n",
       "      <td>0.01901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr3:97729697:A:G</td>\n",
       "      <td>3</td>\n",
       "      <td>97729697</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0183</td>\n",
       "      <td>-0.0126</td>\n",
       "      <td>0.81340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101592</th>\n",
       "      <td>chr14:32118446:G:A</td>\n",
       "      <td>14</td>\n",
       "      <td>32118446</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>0.3314</td>\n",
       "      <td>0.17720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101593</th>\n",
       "      <td>chr1:160763389:C:A</td>\n",
       "      <td>1</td>\n",
       "      <td>160763389</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0006</td>\n",
       "      <td>0.2565</td>\n",
       "      <td>0.40880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101594</th>\n",
       "      <td>chr6:90702181:A:T</td>\n",
       "      <td>6</td>\n",
       "      <td>90702181</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.7871</td>\n",
       "      <td>-0.0151</td>\n",
       "      <td>0.23820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101595</th>\n",
       "      <td>chr9:17691435:C:T</td>\n",
       "      <td>9</td>\n",
       "      <td>17691435</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0376</td>\n",
       "      <td>-0.0175</td>\n",
       "      <td>0.55210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101596</th>\n",
       "      <td>chr18:22100264:G:A</td>\n",
       "      <td>18</td>\n",
       "      <td>22100264</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>-0.0013</td>\n",
       "      <td>0.99760</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21101597 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  MarkerName Chromosome   Position effect_allele other_allele  \\\n",
       "0         chr11:12541586:A:G         11   12541586             A            G   \n",
       "1         chr8:102803998:G:A          8  102803998             A            G   \n",
       "2         chr13:55101557:T:C         13   55101557             T            C   \n",
       "3          chr7:17885109:C:G          7   17885109             C            G   \n",
       "4          chr3:97729697:A:G          3   97729697             A            G   \n",
       "...                      ...        ...        ...           ...          ...   \n",
       "21101592  chr14:32118446:G:A         14   32118446             A            G   \n",
       "21101593  chr1:160763389:C:A          1  160763389             A            C   \n",
       "21101594   chr6:90702181:A:T          6   90702181             A            T   \n",
       "21101595   chr9:17691435:C:T          9   17691435             T            C   \n",
       "21101596  chr18:22100264:G:A         18   22100264             A            G   \n",
       "\n",
       "          effect_allele_frequency    beta  p_value  \n",
       "0                          0.3748  0.0007  0.94860  \n",
       "1                          0.0026 -0.1579  0.19910  \n",
       "2                          0.9856 -0.0252  0.56260  \n",
       "3                          0.9989  0.5638  0.01901  \n",
       "4                          0.0183 -0.0126  0.81340  \n",
       "...                           ...     ...      ...  \n",
       "21101592                   0.0011  0.3314  0.17720  \n",
       "21101593                   0.0006  0.2565  0.40880  \n",
       "21101594                   0.7871 -0.0151  0.23820  \n",
       "21101595                   0.0376 -0.0175  0.55210  \n",
       "21101596                   0.0004 -0.0013  0.99760  \n",
       "\n",
       "[21101597 rows x 8 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Split the 'MarkerName' column and extract 'Chromosome' and 'Position'\n",
    "ad_eur['Chromosome'] = ad_eur['MarkerName'].str.split(':').str[0].str.replace('chr', '', regex=False)\n",
    "ad_eur['Position'] = ad_eur['MarkerName'].str.split(':').str[1]\n",
    "\n",
    "# Rename columns\n",
    "ad_eur = ad_eur[['MarkerName','Chromosome','Position','effect_allele','other_allele','effect_allele_frequency','beta','p_value']]\n",
    "\n",
    "# Add ancestry column\n",
    "ad_eur['Ancestry'] = 'European'\n",
    "ad_eur.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9cf80401-04ea-464f-8256-424410f89f5d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:100000012</td>\n",
       "      <td>10</td>\n",
       "      <td>100000012</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0180</td>\n",
       "      <td>0.1734</td>\n",
       "      <td>0.2524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:100000122</td>\n",
       "      <td>10</td>\n",
       "      <td>100000122</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0083</td>\n",
       "      <td>0.3347</td>\n",
       "      <td>0.1445</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10:100000354</td>\n",
       "      <td>10</td>\n",
       "      <td>100000354</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.9989</td>\n",
       "      <td>0.6226</td>\n",
       "      <td>0.5404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10:100000588</td>\n",
       "      <td>10</td>\n",
       "      <td>100000588</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.9603</td>\n",
       "      <td>-0.0061</td>\n",
       "      <td>0.9531</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10:100000625</td>\n",
       "      <td>10</td>\n",
       "      <td>100000625</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.7182</td>\n",
       "      <td>0.0059</td>\n",
       "      <td>0.8873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27724500</th>\n",
       "      <td>9:99999366</td>\n",
       "      <td>9</td>\n",
       "      <td>99999366</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.9970</td>\n",
       "      <td>0.8480</td>\n",
       "      <td>0.1425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27724501</th>\n",
       "      <td>9:99999388</td>\n",
       "      <td>9</td>\n",
       "      <td>99999388</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.9736</td>\n",
       "      <td>0.1170</td>\n",
       "      <td>0.3396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27724502</th>\n",
       "      <td>9:99999468</td>\n",
       "      <td>9</td>\n",
       "      <td>99999468</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.8633</td>\n",
       "      <td>0.0523</td>\n",
       "      <td>0.3623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27724503</th>\n",
       "      <td>9:99999608</td>\n",
       "      <td>9</td>\n",
       "      <td>99999608</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0156</td>\n",
       "      <td>-0.1658</td>\n",
       "      <td>0.3139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27724504</th>\n",
       "      <td>9:99999952</td>\n",
       "      <td>9</td>\n",
       "      <td>99999952</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0033</td>\n",
       "      <td>0.3410</td>\n",
       "      <td>0.5191</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27724505 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            MarkerName  Chromosome   Position effect_allele other_allele  \\\n",
       "0         10:100000012          10  100000012             A            G   \n",
       "1         10:100000122          10  100000122             A            T   \n",
       "2         10:100000354          10  100000354             T            C   \n",
       "3         10:100000588          10  100000588             T            C   \n",
       "4         10:100000625          10  100000625             A            G   \n",
       "...                ...         ...        ...           ...          ...   \n",
       "27724500    9:99999366           9   99999366             T            G   \n",
       "27724501    9:99999388           9   99999388             A            C   \n",
       "27724502    9:99999468           9   99999468             A            G   \n",
       "27724503    9:99999608           9   99999608             T            C   \n",
       "27724504    9:99999952           9   99999952             T            C   \n",
       "\n",
       "          effect_allele_frequency    beta  p_value  \n",
       "0                          0.0180  0.1734   0.2524  \n",
       "1                          0.0083  0.3347   0.1445  \n",
       "2                          0.9989  0.6226   0.5404  \n",
       "3                          0.9603 -0.0061   0.9531  \n",
       "4                          0.7182  0.0059   0.8873  \n",
       "...                           ...     ...      ...  \n",
       "27724500                   0.9970  0.8480   0.1425  \n",
       "27724501                   0.9736  0.1170   0.3396  \n",
       "27724502                   0.8633  0.0523   0.3623  \n",
       "27724503                   0.0156 -0.1658   0.3139  \n",
       "27724504                   0.0033  0.3410   0.5191  \n",
       "\n",
       "[27724505 rows x 8 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Rename columns\n",
    "ad_afr = ad_afr.rename(columns={'Chr':'Chromosome','Pos':'Position','Effect_allele':'effect_allele','Non_Effect_allele':'other_allele','Beta':'beta','Pvalue':'p_value',\n",
    "                                'Effect_allele_Freq':'effect_allele_frequency'})\n",
    "ad_afr = ad_afr[['MarkerName','Chromosome','Position','effect_allele','other_allele','effect_allele_frequency','beta','p_value']]\n",
    "\n",
    "# Add ancestry column\n",
    "ad_afr['Ancestry'] = 'African/African Admixed'\n",
    "ad_afr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "41417fb2-fbf0-403e-955a-564e54c28c69",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:752566</td>\n",
       "      <td>1</td>\n",
       "      <td>752566</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.147100</td>\n",
       "      <td>-0.031387</td>\n",
       "      <td>0.4937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:753405</td>\n",
       "      <td>1</td>\n",
       "      <td>753405</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.148900</td>\n",
       "      <td>-0.033764</td>\n",
       "      <td>0.4602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:753541</td>\n",
       "      <td>1</td>\n",
       "      <td>753541</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.374100</td>\n",
       "      <td>-0.024395</td>\n",
       "      <td>0.4874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:754182</td>\n",
       "      <td>1</td>\n",
       "      <td>754182</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.374000</td>\n",
       "      <td>-0.025728</td>\n",
       "      <td>0.4640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:754192</td>\n",
       "      <td>1</td>\n",
       "      <td>754192</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.374000</td>\n",
       "      <td>-0.025728</td>\n",
       "      <td>0.4640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4852952</th>\n",
       "      <td>22:51185740</td>\n",
       "      <td>22</td>\n",
       "      <td>51185740</td>\n",
       "      <td>ACATTGT</td>\n",
       "      <td>A</td>\n",
       "      <td>0.008116</td>\n",
       "      <td>-0.229162</td>\n",
       "      <td>0.2024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4852953</th>\n",
       "      <td>22:51185743</td>\n",
       "      <td>22</td>\n",
       "      <td>51185743</td>\n",
       "      <td>A</td>\n",
       "      <td>AAGC</td>\n",
       "      <td>0.008116</td>\n",
       "      <td>-0.229162</td>\n",
       "      <td>0.2024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4852954</th>\n",
       "      <td>22:51189403</td>\n",
       "      <td>22</td>\n",
       "      <td>51189403</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.027350</td>\n",
       "      <td>-0.037702</td>\n",
       "      <td>0.7084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4852955</th>\n",
       "      <td>22:51208005</td>\n",
       "      <td>22</td>\n",
       "      <td>51208005</td>\n",
       "      <td>G</td>\n",
       "      <td>GTC</td>\n",
       "      <td>0.008122</td>\n",
       "      <td>-0.177812</td>\n",
       "      <td>0.3272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4852956</th>\n",
       "      <td>22:51222100</td>\n",
       "      <td>22</td>\n",
       "      <td>51222100</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.008257</td>\n",
       "      <td>-0.167827</td>\n",
       "      <td>0.3517</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4852957 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          MarkerName  Chromosome  Position effect_allele other_allele  \\\n",
       "0           1:752566           1    752566             G            A   \n",
       "1           1:753405           1    753405             C            A   \n",
       "2           1:753541           1    753541             A            G   \n",
       "3           1:754182           1    754182             A            G   \n",
       "4           1:754192           1    754192             A            G   \n",
       "...              ...         ...       ...           ...          ...   \n",
       "4852952  22:51185740          22  51185740       ACATTGT            A   \n",
       "4852953  22:51185743          22  51185743             A         AAGC   \n",
       "4852954  22:51189403          22  51189403             G            A   \n",
       "4852955  22:51208005          22  51208005             G          GTC   \n",
       "4852956  22:51222100          22  51222100             T            G   \n",
       "\n",
       "         effect_allele_frequency      beta  p_value  \n",
       "0                       0.147100 -0.031387   0.4937  \n",
       "1                       0.148900 -0.033764   0.4602  \n",
       "2                       0.374100 -0.024395   0.4874  \n",
       "3                       0.374000 -0.025728   0.4640  \n",
       "4                       0.374000 -0.025728   0.4640  \n",
       "...                          ...       ...      ...  \n",
       "4852952                 0.008116 -0.229162   0.2024  \n",
       "4852953                 0.008116 -0.229162   0.2024  \n",
       "4852954                 0.027350 -0.037702   0.7084  \n",
       "4852955                 0.008122 -0.177812   0.3272  \n",
       "4852956                 0.008257 -0.167827   0.3517  \n",
       "\n",
       "[4852957 rows x 8 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create MarkerName column\n",
    "ad_eas['MarkerName'] = ad_eas['CHR'].astype(str) + ':' + ad_eas['BP'].astype(str)\n",
    "\n",
    "# Rename columns\n",
    "ad_eas = ad_eas.rename(columns={'CHR':'Chromosome','BP':'Position','A1':'effect_allele','A2':'other_allele','P':'p_value',\n",
    "                                'MAF_A':'effect_allele_frequency'})\n",
    "# Calculate odds ratio\n",
    "ad_eas['beta'] = np.log(ad_eas['OR'])\n",
    "ad_eas = ad_eas[['MarkerName','Chromosome','Position','effect_allele','other_allele','effect_allele_frequency','beta','p_value']]\n",
    "\n",
    "# Add ancestry column\n",
    "ad_eas['Ancestry'] = 'East Asian'\n",
    "ad_eas.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "3a9ab833-a997-4eb0-bf72-a5f8bf1041b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:48824</td>\n",
       "      <td>1</td>\n",
       "      <td>48824</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.017666</td>\n",
       "      <td>-0.229158</td>\n",
       "      <td>0.431702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:54490</td>\n",
       "      <td>1</td>\n",
       "      <td>54490</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.043157</td>\n",
       "      <td>-0.158201</td>\n",
       "      <td>0.421182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:60351</td>\n",
       "      <td>1</td>\n",
       "      <td>60351</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.024821</td>\n",
       "      <td>-0.145668</td>\n",
       "      <td>0.550887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:64931</td>\n",
       "      <td>1</td>\n",
       "      <td>64931</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.024598</td>\n",
       "      <td>-0.196565</td>\n",
       "      <td>0.426951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:66861</td>\n",
       "      <td>1</td>\n",
       "      <td>66861</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.044946</td>\n",
       "      <td>-0.140637</td>\n",
       "      <td>0.442988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12466227</th>\n",
       "      <td>22:51242036</td>\n",
       "      <td>22</td>\n",
       "      <td>51242036</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.030188</td>\n",
       "      <td>0.092679</td>\n",
       "      <td>0.666683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12466228</th>\n",
       "      <td>22:51242137</td>\n",
       "      <td>22</td>\n",
       "      <td>51242137</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.013417</td>\n",
       "      <td>0.003723</td>\n",
       "      <td>0.992015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12466229</th>\n",
       "      <td>22:51242271</td>\n",
       "      <td>22</td>\n",
       "      <td>51242271</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.115385</td>\n",
       "      <td>0.184028</td>\n",
       "      <td>0.113002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12466230</th>\n",
       "      <td>22:51242557</td>\n",
       "      <td>22</td>\n",
       "      <td>51242557</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.063283</td>\n",
       "      <td>0.139475</td>\n",
       "      <td>0.364707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12466231</th>\n",
       "      <td>22:48118357</td>\n",
       "      <td>22</td>\n",
       "      <td>48118357</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.016100</td>\n",
       "      <td>0.117579</td>\n",
       "      <td>0.694706</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12466232 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           MarkerName  Chromosome  Position effect_allele other_allele  \\\n",
       "0             1:48824           1     48824             C            T   \n",
       "1             1:54490           1     54490             A            G   \n",
       "2             1:60351           1     60351             G            A   \n",
       "3             1:64931           1     64931             A            G   \n",
       "4             1:66861           1     66861             T            C   \n",
       "...               ...         ...       ...           ...          ...   \n",
       "12466227  22:51242036          22  51242036             C            G   \n",
       "12466228  22:51242137          22  51242137             A            T   \n",
       "12466229  22:51242271          22  51242271             G            C   \n",
       "12466230  22:51242557          22  51242557             T            A   \n",
       "12466231  22:48118357          22  48118357             A            G   \n",
       "\n",
       "          effect_allele_frequency      beta   p_value  \n",
       "0                        0.017666 -0.229158  0.431702  \n",
       "1                        0.043157 -0.158201  0.421182  \n",
       "2                        0.024821 -0.145668  0.550887  \n",
       "3                        0.024598 -0.196565  0.426951  \n",
       "4                        0.044946 -0.140637  0.442988  \n",
       "...                           ...       ...       ...  \n",
       "12466227                 0.030188  0.092679  0.666683  \n",
       "12466228                 0.013417  0.003723  0.992015  \n",
       "12466229                 0.115385  0.184028  0.113002  \n",
       "12466230                 0.063283  0.139475  0.364707  \n",
       "12466231                 0.016100  0.117579  0.694706  \n",
       "\n",
       "[12466232 rows x 8 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Rename columns\n",
    "ad_lat = ad_lat.rename(columns={'MARKERNAME':'MarkerName','CHROMOSOME':'Chromosome','POSITION':'Position','EA':'effect_allele','NEA':'other_allele','P':'p_value',\n",
    "                                'EAF':'effect_allele_frequency'})\n",
    "\n",
    "# Calculate odds ratio\n",
    "ad_lat['beta'] = np.log(ad_lat['OR'])\n",
    "ad_lat = ad_lat[['MarkerName','Chromosome','Position','effect_allele','other_allele','effect_allele_frequency','beta','p_value']]\n",
    "\n",
    "# Add ancestry column\n",
    "ad_lat['Ancestry'] = 'Latino'\n",
    "ad_lat.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c25a25d3-63ae-4f6f-9e9b-ec5db59489d9",
   "metadata": {},
   "source": [
    "### Prepare PD summary statistics for PAR calculations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b02a2802-0b35-42f9-a67b-4b550caa4f3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename columns\n",
    "pd_eur = pd_eur.rename(columns={'MarkerName':'MarkerName_old','SNP':'MarkerName','Allele1':'effect_allele','Allele2':'other_allele','Freq1':'effect_allele_frequency','Effect':'beta',\n",
    "                                'P-value':'p_value','Chr':'Chromosome','BP':'Position'})\n",
    "pd_eur = pd_eur[['MarkerName','Chromosome','Position','effect_allele','other_allele','effect_allele_frequency','beta','p_value']]\n",
    "\n",
    "# Uppercase for allele columns\n",
    "pd_eur['effect_allele'] = pd_eur['effect_allele'].str.upper()\n",
    "pd_eur['other_allele'] = pd_eur['other_allele'].str.upper()\n",
    "\n",
    "# Add ancestry column\n",
    "pd_eur['Ancestry'] = 'European'\n",
    "pd_eur.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "5dbdc3d8-f723-47cf-8b99-944f81e49a27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:66861:C:T</td>\n",
       "      <td>1</td>\n",
       "      <td>66861</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0724</td>\n",
       "      <td>-0.1072</td>\n",
       "      <td>0.789900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:80346:C:G</td>\n",
       "      <td>1</td>\n",
       "      <td>80346</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.8338</td>\n",
       "      <td>0.4608</td>\n",
       "      <td>0.182400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:595259:G:A</td>\n",
       "      <td>1</td>\n",
       "      <td>595259</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0597</td>\n",
       "      <td>-0.0456</td>\n",
       "      <td>0.911400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:664938:A:G</td>\n",
       "      <td>1</td>\n",
       "      <td>664938</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.9390</td>\n",
       "      <td>-0.1109</td>\n",
       "      <td>0.624200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:665098:G:A</td>\n",
       "      <td>1</td>\n",
       "      <td>665098</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1262</td>\n",
       "      <td>-0.0315</td>\n",
       "      <td>0.930100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9339045</th>\n",
       "      <td>chrX:156025429:T:C</td>\n",
       "      <td>X</td>\n",
       "      <td>156025429</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.9476</td>\n",
       "      <td>0.2412</td>\n",
       "      <td>0.291900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9339046</th>\n",
       "      <td>chrX:156025488:G:A</td>\n",
       "      <td>X</td>\n",
       "      <td>156025488</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0565</td>\n",
       "      <td>0.0024</td>\n",
       "      <td>0.992000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9339047</th>\n",
       "      <td>chrX:156030815:T:G</td>\n",
       "      <td>X</td>\n",
       "      <td>156030815</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.4469</td>\n",
       "      <td>0.2225</td>\n",
       "      <td>0.085630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9339048</th>\n",
       "      <td>chrX:156030858:G:T</td>\n",
       "      <td>X</td>\n",
       "      <td>156030858</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.4391</td>\n",
       "      <td>-0.4200</td>\n",
       "      <td>0.001933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9339049</th>\n",
       "      <td>chrY:5806136:T:G</td>\n",
       "      <td>Y</td>\n",
       "      <td>5806136</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.8867</td>\n",
       "      <td>-0.0042</td>\n",
       "      <td>0.977500</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9339050 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 MarkerName Chromosome   Position effect_allele other_allele  \\\n",
       "0            chr1:66861:C:T          1      66861             T            C   \n",
       "1            chr1:80346:C:G          1      80346             C            G   \n",
       "2           chr1:595259:G:A          1     595259             A            G   \n",
       "3           chr1:664938:A:G          1     664938             A            G   \n",
       "4           chr1:665098:G:A          1     665098             A            G   \n",
       "...                     ...        ...        ...           ...          ...   \n",
       "9339045  chrX:156025429:T:C          X  156025429             T            C   \n",
       "9339046  chrX:156025488:G:A          X  156025488             A            G   \n",
       "9339047  chrX:156030815:T:G          X  156030815             T            G   \n",
       "9339048  chrX:156030858:G:T          X  156030858             T            G   \n",
       "9339049    chrY:5806136:T:G          Y    5806136             T            G   \n",
       "\n",
       "         effect_allele_frequency    beta   p_value  \n",
       "0                         0.0724 -0.1072  0.789900  \n",
       "1                         0.8338  0.4608  0.182400  \n",
       "2                         0.0597 -0.0456  0.911400  \n",
       "3                         0.9390 -0.1109  0.624200  \n",
       "4                         0.1262 -0.0315  0.930100  \n",
       "...                          ...     ...       ...  \n",
       "9339045                   0.9476  0.2412  0.291900  \n",
       "9339046                   0.0565  0.0024  0.992000  \n",
       "9339047                   0.4469  0.2225  0.085630  \n",
       "9339048                   0.4391 -0.4200  0.001933  \n",
       "9339049                   0.8867 -0.0042  0.977500  \n",
       "\n",
       "[9339050 rows x 8 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Rename columns\n",
    "pd_afr  = pd_afr.rename(columns={'chromosome':'Chromosome','base_pair_location':'Position','variant_id':'MarkerName'})\n",
    "pd_afr = pd_afr[['MarkerName','Chromosome','Position','effect_allele','other_allele','effect_allele_frequency','beta','p_value']]\n",
    "\n",
    "# Add ancestry column\n",
    "pd_afr['Ancestry'] = 'African/African Admixed'\n",
    "pd_afr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "c05d1d0f-70ba-4f18-8036-16f5d0cb83dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:794332</td>\n",
       "      <td>1</td>\n",
       "      <td>794332</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.139423</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>0.96910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:832359</td>\n",
       "      <td>1</td>\n",
       "      <td>832359</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.031250</td>\n",
       "      <td>-0.0932</td>\n",
       "      <td>0.18680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:834056</td>\n",
       "      <td>1</td>\n",
       "      <td>834056</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.028846</td>\n",
       "      <td>-0.0766</td>\n",
       "      <td>0.32270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:834263</td>\n",
       "      <td>1</td>\n",
       "      <td>834263</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.028846</td>\n",
       "      <td>-0.0846</td>\n",
       "      <td>0.28530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:834956</td>\n",
       "      <td>1</td>\n",
       "      <td>834956</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.026442</td>\n",
       "      <td>-0.0704</td>\n",
       "      <td>0.36590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5629375</th>\n",
       "      <td>chr22:51158499</td>\n",
       "      <td>22</td>\n",
       "      <td>51158499</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.512019</td>\n",
       "      <td>-0.0029</td>\n",
       "      <td>0.90000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5629376</th>\n",
       "      <td>chr22:51164109</td>\n",
       "      <td>22</td>\n",
       "      <td>51164109</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.560096</td>\n",
       "      <td>0.0059</td>\n",
       "      <td>0.80760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5629377</th>\n",
       "      <td>chr22:51164115</td>\n",
       "      <td>22</td>\n",
       "      <td>51164115</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.562500</td>\n",
       "      <td>0.0030</td>\n",
       "      <td>0.90240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5629378</th>\n",
       "      <td>chr22:51181685</td>\n",
       "      <td>22</td>\n",
       "      <td>51181685</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.694712</td>\n",
       "      <td>0.0484</td>\n",
       "      <td>0.07563</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5629379</th>\n",
       "      <td>chr22:51181759</td>\n",
       "      <td>22</td>\n",
       "      <td>51181759</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.699519</td>\n",
       "      <td>0.0304</td>\n",
       "      <td>0.21050</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5629380 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             MarkerName  Chromosome  Position effect_allele other_allele  \\\n",
       "0           chr1:794332           1    794332             A            G   \n",
       "1           chr1:832359           1    832359             T            C   \n",
       "2           chr1:834056           1    834056             C            A   \n",
       "3           chr1:834263           1    834263             T            C   \n",
       "4           chr1:834956           1    834956             A            G   \n",
       "...                 ...         ...       ...           ...          ...   \n",
       "5629375  chr22:51158499          22  51158499             G            T   \n",
       "5629376  chr22:51164109          22  51164109             C            G   \n",
       "5629377  chr22:51164115          22  51164115             T            C   \n",
       "5629378  chr22:51181685          22  51181685             A            G   \n",
       "5629379  chr22:51181759          22  51181759             G            A   \n",
       "\n",
       "         effect_allele_frequency    beta  p_value  \n",
       "0                       0.139423  0.0013  0.96910  \n",
       "1                       0.031250 -0.0932  0.18680  \n",
       "2                       0.028846 -0.0766  0.32270  \n",
       "3                       0.028846 -0.0846  0.28530  \n",
       "4                       0.026442 -0.0704  0.36590  \n",
       "...                          ...     ...      ...  \n",
       "5629375                 0.512019 -0.0029  0.90000  \n",
       "5629376                 0.560096  0.0059  0.80760  \n",
       "5629377                 0.562500  0.0030  0.90240  \n",
       "5629378                 0.694712  0.0484  0.07563  \n",
       "5629379                 0.699519  0.0304  0.21050  \n",
       "\n",
       "[5629380 rows x 8 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Rename columns\n",
    "pd_eas = pd_eas.rename(columns={'CHROMOSOME':'Chromosome','POSITION':'Position','EA':'effect_allele','NEA':'other_allele','P':'p_value',\n",
    "                                'EAF':'effect_allele_frequency','BETA':'beta','MARKERNAME':'MarkerName'})\n",
    "pd_eas = pd_eas[['MarkerName','Chromosome','Position','effect_allele','other_allele','effect_allele_frequency','beta','p_value']]\n",
    "\n",
    "# Add ancestry column\n",
    "pd_eas['Ancestry'] = 'East Asian'\n",
    "pd_eas.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "aaa89aa8-a067-471e-8790-ada2d56bd5af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>beta</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:662622</td>\n",
       "      <td>1</td>\n",
       "      <td>662622</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.065474</td>\n",
       "      <td>-0.187575</td>\n",
       "      <td>0.371967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:666249</td>\n",
       "      <td>1</td>\n",
       "      <td>666249</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.073317</td>\n",
       "      <td>0.266337</td>\n",
       "      <td>0.140982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:668394</td>\n",
       "      <td>1</td>\n",
       "      <td>668394</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>0.018808</td>\n",
       "      <td>-0.171700</td>\n",
       "      <td>0.726995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:676118</td>\n",
       "      <td>1</td>\n",
       "      <td>676118</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.038932</td>\n",
       "      <td>0.044314</td>\n",
       "      <td>0.875518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:693625</td>\n",
       "      <td>1</td>\n",
       "      <td>693625</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.021886</td>\n",
       "      <td>-0.310839</td>\n",
       "      <td>0.386173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8668488</th>\n",
       "      <td>chr22:51235959</td>\n",
       "      <td>22</td>\n",
       "      <td>51235959</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.085745</td>\n",
       "      <td>-0.080132</td>\n",
       "      <td>0.643497</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8668489</th>\n",
       "      <td>chr22:51235979</td>\n",
       "      <td>22</td>\n",
       "      <td>51235979</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.281600</td>\n",
       "      <td>-0.046489</td>\n",
       "      <td>0.668448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8668490</th>\n",
       "      <td>chr22:51236013</td>\n",
       "      <td>22</td>\n",
       "      <td>51236013</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>0.230838</td>\n",
       "      <td>0.031699</td>\n",
       "      <td>0.778325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8668491</th>\n",
       "      <td>chr22:51237063</td>\n",
       "      <td>22</td>\n",
       "      <td>51237063</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.307650</td>\n",
       "      <td>0.125151</td>\n",
       "      <td>0.241345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8668492</th>\n",
       "      <td>chr22:51238249</td>\n",
       "      <td>22</td>\n",
       "      <td>51238249</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.050723</td>\n",
       "      <td>0.077172</td>\n",
       "      <td>0.721062</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8668493 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             MarkerName Chromosome  Position effect_allele other_allele  \\\n",
       "0           chr1:662622          1    662622             A            G   \n",
       "1           chr1:666249          1    666249             T            C   \n",
       "2           chr1:668394          1    668394             A           AG   \n",
       "3           chr1:676118          1    676118             T            C   \n",
       "4           chr1:693625          1    693625             C            T   \n",
       "...                 ...        ...       ...           ...          ...   \n",
       "8668488  chr22:51235959         22  51235959             C            T   \n",
       "8668489  chr22:51235979         22  51235979             A            G   \n",
       "8668490  chr22:51236013         22  51236013            AT            A   \n",
       "8668491  chr22:51237063         22  51237063             C            T   \n",
       "8668492  chr22:51238249         22  51238249             C            A   \n",
       "\n",
       "         effect_allele_frequency      beta   p_value  \n",
       "0                       0.065474 -0.187575  0.371967  \n",
       "1                       0.073317  0.266337  0.140982  \n",
       "2                       0.018808 -0.171700  0.726995  \n",
       "3                       0.038932  0.044314  0.875518  \n",
       "4                       0.021886 -0.310839  0.386173  \n",
       "...                          ...       ...       ...  \n",
       "8668488                 0.085745 -0.080132  0.643497  \n",
       "8668489                 0.281600 -0.046489  0.668448  \n",
       "8668490                 0.230838  0.031699  0.778325  \n",
       "8668491                 0.307650  0.125151  0.241345  \n",
       "8668492                 0.050723  0.077172  0.721062  \n",
       "\n",
       "[8668493 rows x 8 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Rename columns\n",
    "pd_lat = pd_lat.rename(columns={'CHROMOSOME':'Chromosome','POSITION':'Position','EA':'effect_allele','NEA':'other_allele','P':'p_value',\n",
    "                                'EAF':'effect_allele_frequency','BETA':'beta','MARKERNAME':'MarkerName'})\n",
    "pd_lat = pd_lat[['MarkerName','Chromosome','Position','effect_allele','other_allele','effect_allele_frequency','beta','p_value']]\n",
    "\n",
    "# Add ancestry column\n",
    "pd_lat['Ancestry'] = 'Latino'\n",
    "pd_lat.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a2f4e39-43a5-49a4-ab10-83ae97ea4a8b",
   "metadata": {},
   "source": [
    "### Create dataframe for calculations with summary statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "c42852d8-3b10-4b6c-982c-b7423dc8dcc2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Concatenate the summary statistics \n",
    "summary_stats = pd.concat([ad_eur,ad_afr,ad_eas,ad_lat,pd_eur,pd_afr,pd_eas,pd_lat], ignore_index=True)\n",
    "\n",
    "# Merge final_df with summ stats on MarkerName and Ancestry\n",
    "final_df = final_df.merge(summary_stats, on=['MarkerName','Ancestry'], how='left')\n",
    "\n",
    "# Drop unnecessary column\n",
    "final_df = final_df.drop(columns='beta_x')\n",
    "\n",
    "# Rename column\n",
    "final_df = final_df.rename(columns={'beta_y':'beta'})\n",
    "\n",
    "# Export to csv\n",
    "final_df.to_csv('{WORK_DIR}/PAR/reuslts/80_pp_par_sumstats.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0077c940-8f47-4a94-b0b7-6416f43f7451",
   "metadata": {},
   "source": [
    "### PAR calculations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "9301dd8a-b943-441a-bf83-6045be511bdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Negative beta -> flipped alleles, new maf = 1 - maf\n",
    "final_df['eaf_risk'] = final_df['effect_allele_frequency'].where(final_df['beta'] > 0, 1 - final_df['effect_allele_frequency'])\n",
    "\n",
    "# Absolute value of beta\n",
    "final_df['beta_abs'] = final_df['beta'].abs()\n",
    "\n",
    "# Risk allele\n",
    "final_df['risk_allele'] = np.where(final_df['beta'] < 0, final_df['other_allele'], final_df['effect_allele'])\n",
    "\n",
    "# Calculate the new odds ratio with beta_abs\n",
    "final_df['odds_ratio_new'] = np.exp(final_df['beta_abs'])\n",
    "\n",
    "#Calculate the population attributable risk (PAR = MAF(OR - 1)/1 + MAF(OR - 1))\n",
    "final_df['par'] = (final_df['eaf_risk']*(final_df['odds_ratio_new']-1)) / (1 + final_df['eaf_risk']*(final_df['odds_ratio_new']-1))\n",
    "\n",
    "# Replacing values -> CRHR1 is always MAPT, GBAP1 -> GBA1\n",
    "final_df['Gene'] = final_df['Gene'].replace({'CRHR1':'MAPT'})\n",
    "final_df['Gene'] = final_df['Gene'].replace({'GBAP1':'GBA1'})\n",
    "\n",
    "# Export results to csv\n",
    "final_df.to_csv('{WORK_DIR}/PAR/results/80_pp_par_results.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "python/3.10",
   "language": "python",
   "name": "py3.10"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
